
<!DOCTYPE html>
<html lang="zh-cn">
    
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="iuu&#39;s blog">
    <title>GWAS|EPACTS安装及使用 - iuu&#39;s blog</title>
    <meta name="author" content="Dr.iuu">
    
    
        <link rel="icon" href="https://iuu19.github.io/assets/images/coveriuu.jpg">
    
    
        
            <link rel="alternate" type="application/atom+xml" title="RSS" href="/atom.xml">
        
    
    <script type="application/ld+json">{"@context":"http://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg"},"articleBody":"\n\n\n\nGWAS|EPACTS安装及使用\n\n\n1. Installation DetailsIf you want to use EPACTS in an Ubuntu platform, following the step below\ngit clone https://github.com/statgen/EPACTS.git\nmv EPACTS/ epacts\ncd epacts\n./configure --prefix=~/you/biosoft/epacts\nmake\nmake install\n\nImportant Note: make sure to specify –prefix&#x3D;&#x2F;path&#x2F;to&#x2F;install。安装过程中有太多的坑：\n\n因为本服务器gcc是7.2.0，所以安装过程中总是安装不成功。去EPACTS Google group讨论区说gcc需为gcc-5. gcc-5安装过程中也会有很多问题，具体可以关注公众号Gene2You，进行留言咨询。\n.&#x2F;configure过程中还会遇到其他问题，如需要安装”groff”，”gnuplot”，”No man pages found in package ‘epactsR’”，”make[2]: *** No rule to make target single.b.spa.R’, needed by all-am’. Stop.”等，可以查询EPACTS Google group讨论区，或者关注公众号Gene2You，进行留言咨询。\n\n安装成功后：\n# Now $&#123;EPACTS_DIR&#125; represents the &#39;/path/to/install&#39; directory\nEPACTS_DIR=/WORK/Project/youguoling/biosoft/epacts\n\n2. Download the reference FASTA files\nUCSC reference\nNCBI reference\nEnsembl reference\n如果Download the reference FASTA files from 1000 Genomes FTP automatically by running the following commands：\n\n $&#123;EPACTS_DIR&#125;/bin/epacts download\n\n3. Perform a test runby running the following command:\n$&#123;EPACTS_DIR&#125;/bin/test_run_epacts.sh\n\n4. Getting Started With Examples4.1 All-in-one exampleTo get started with EPACTS, run the following command will perform an example run\n$&#123;EPACTS_DIR&#125;/bin/test_run_epacts.sh\n\nYou will find a series of lines in test_run_epacts.sh script commented out for each possible test.\nThe example phenotype (PED format) and genotype (VCF format) can be found at\n$&#123;EPACTS_DIR&#125;/share/EPACTS/\n\n4.2 Single Variant TestOr You can run EPACTS command yourself by running\n$&#123;EPACTS_DIR&#125;/epacts single \\\n  --vcf  $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.vcf.gz \\\n  --ped  $&#123;EPACTS_DIR&#125;/data/1000G_dummy_pheno.ped  \\\n  --min-maf 0.001 --chr 20 --pheno DISEASE --cov AGE --cov SEX --test b.score --anno \\ \n  --out out/test --run 2\n\nThe command above will perform single variant association test using a dummy case-control phenotype file and a subset of 1000 genomes exome VCF file (chr20) using score test statistic for all variants over 1% of higher MAF using 2 parallel runs.\nYou will see the 4 output files as the main outcome of the analysis\n4.2.1 Output Text of All Test StatisticsThe filename is out/test.single.b.score.epacts.gz and the contents will look like:\n$ zcat out/test.single.b.score.epacts.gz | head\n#CHROM\tBEGIN\tEND\tMARKER_ID\tNS\tAC\tCALLRATE\tMAF\tPVALUE\tSCORE\tN.CASE\tN.CTRL\tAF.CASE\tAF.CTRL\n20\t68303\t68303\t20:68303_A/G_Upstream:DEFB125\t266\t1\t1\t0.0018797\tNA\tNA\tNA\tNA\tNA\tNA\n20\t68319\t68319\t20:68319_C/A_Upstream:DEFB125\t266\t1.4467e-36\t1\t0\tNA\tNA\tNA\tNA\tNA\tNA\n20\t68396\t68396\t20:68396_C/T_Nonsynonymous:DEFB125\t266\t1\t1\t0.0018797\tNA\tNA\tNA\tNA\tNA\tNA\n20\t76635\t76635\t20:76635_A/T_Intron:DEFB125\t266\t1.534e-37\t1\t0\tNA\tNA\tNA\tNA\tNA\tNA\n20\t76689\t76689\t20:76689_T/C_Synonymous:DEFB125\t266\t0\t1\t0\tNA\tNA\tNA\tNA\tNA\tNA\n20\t76690\t76690\t20:76690_T/C_Nonsynonymous:DEFB125\t266\t1\t1\t0.0018797\tNA\tNA\tNA\tNA\tNA\tNA\n20\t76700\t76700\t20:76700_G/A_Nonsynonymous:DEFB125\t266\t0\t1\t0\tNA\tNA\tNA\tNA\tNA\tNA\n20\t76726\t76726\t20:76726_C/G_Nonsynonymous:DEFB125\t266\t0\t1\t0\tNA\tNA\tNA\tNA\tNA\tNA\n20\t76771\t76771\t20:76771_C/T_Nonsynonymous:DEFB125\t266\t3\t1\t0.0056391\t0.68484\t0.40587\t145\t121\t0.013793\t0.0082645\n\n4.2.2 Output Text of Top AssociationsSame type of file but containing top 5,000 association will be stored at out/test.epacts.top5000\n$ head out/test.single.b.score.epacts.top5000 \n#CHROM\tBEGIN\tEND\tMARKER_ID\tNS\tAC\tCALLRATE\tMAF\tPVALUE\tSCORE\tN.CASE\tN.CTRL\tAF.CASE\tAF.CTRL\n20\t1610894\t1610894\t20:1610894_G/A_Synonymous:SIRPG\t266\t138.64\t1\t0.26061\t6.9939e-05\t3.9765\t145\t121\t0.65177\t0.36476\n20\t4162411\t4162411\t20:4162411_T/C_Intron:SMOX\t266\t204\t1\t0.38346\t0.00055583\t-3.4523\t145\t121\t0.62759\t0.93388\n20\t34061918\t34061918\t20:34061918_T/C_Intron:CEP250\t266\t41.815\t1\t0.0786\t0.00095471\t3.3035\t145\t121\t0.22543\t0.075436\n20\t4155948\t4155948\t20:4155948_G/A_Intron:SMOX\t266\t215\t1\t0.40414\t0.0020792\t-3.0787\t145\t121\t0.68276\t0.95868\n20\t4680251\t4680251\t20:4680251_A/G_Nonsynonymous:PRNP\t266\t186\t1\t0.34962\t0.0025962\t3.0119\t145\t121\t0.8069\t0.57025\n20\t36668874\t36668874\t20:36668874_G/A_Synonymous:RPRD1B\t266\t96\t1\t0.18045\t0.003031\t2.9646\t145\t121\t0.44828\t0.2562\n20\t36641871\t36641871\t20:36641871_G/A_Synonymous:TTI1\t266\t10\t1\t0.018797\t0.004308\t-2.8547\t145\t121\t0.0068966\t0.07438\n20\t1616892\t1616892\t20:1616892_A/G_Synonymous:SIRPG\t266\t144\t1\t0.27068\t0.0051239\t2.7991\t145\t121\t0.63449\t0.42975\n20\t25038372\t25038372\t20:25038372_G/A_Intron:ACSS1\t266\t103.3\t1\t0.19418\t0.005748\t2.7618\t145\t121\t0.47201\t0.28813\n\nThe key columns represents:\n\nNS: Number of phenotyped samples with non-missing genotypes\nAC: Total Non-reference Allele Count\nCALLRATE: Fraction of non-missing genotypes.\nMAF: Minor allele frequencies\nPVALUE: P-value of single variant test\nAF.CASE: Non-reference allele frequencies for cases\nAF.CTRL: Non-reference allele frequencies for controls\n\n4.2.3 Q-Q plot of test statistics (stratified by MAF)The file out&#x2F;test.b.score.epacts.qq.pdf will be generated as shown below:\n4.2.4 Manhattan Plot of Test StatisticsThe file out&#x2F;test.b.score.epacts.mh.pdf will be generated for chr20 only.\nAn example Genome-wide manhattan plot (from a genome-wide run) will look like below\n4.3 Gene-wise or group-wise burden testGene-wise or group-wise burden test requires two steps. First, ‘group’ file containing the list of markers per group needs to be generated. Second, group-wise burden test needs to be run\n4.3.1 Creating marker group fileThe marker group file has the following format\n[GROUP_ID]  [MARKER_ID_1]   [MARKER_ID_2]  .... [MARKER_ID_N]\n\nwhere\n\n[GROUP_ID] is a string representing the group (e.g. gene name)\n[MARKER_ID_K] is a marker key as a format of [CHROM]:[POS]_[REF]&#x2F;[ALT] (NOTE THAT THIS IS DIFFERENT FROM TYPICAL VCF MARKER ID field)\n\nNote that [MARKER_ID_K] has to be sorted by increasing order of genomic coordinate\nIn order to create gene-level group file from typically formatted VCF file, one may use the following utility\n$&#123;EPACTS_DIR&#125;/epacts make-group --vcf [input-vcf] --out [output-group-file] --format [epacts, annovar, chaos or gatk] --nonsyn\n\nThe above command create a file [output-group-file] containing a list of missense and nonsense variants per each gene. To incorporate different types of functional annotations, use –type option as follows\n$&#123;EPACTS_DIR&#125;/epacts make-group --vcf [input-vcf] --out [output-group-file] --format [epacts, annovar, chaos or gatk] --type [function_type_1] --type [function_type_2] ...\n\nType &#39;epacts makegroup -man&#39; for the detailed documentation\n4.3.2 Annotating VCF file using EPACTSIf the VCF is not annotated, ‘epacts makegroup’ cannot be used. In order to annotate VCF, one can use the example VCF using ANNOVAR as follows:\n$&#123;EPACTS_DIR&#125;/epacts anno \\\n   --in $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.vcf.gz \\\n   --out $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.anno.vcf.gz\n\nThe epacts anno script will add “ANNO&#x3D;[function]:[genename]” entry into the INFO field based on gencodeV7 (default) or refGene database.\nIt is important to check whether the VCF file is already annotated or not in order to avoid no or redundant annotation.\n4.3.3 Running Groupwise TestTo perform a groupwise burden test on the example VCF (annotated as above), run the following command\n$&#123;EPACTS_DIR&#125;/epacts group --vcf $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.anno.vcf.gz \\\n  --groupf $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.anno.grp --out out/test.gene.skat \\\n  --ped $&#123;EPACTS_DIR&#125;/data/1000G_dummy_pheno.ped --maxAF 0.05 \\\n  --chr 20 --pheno QT --cov AGE --cov SEX --test skat --skat-o --run 2\n\n4.3.4 Example Output$ head out/test.gene.skat.epacts.top5000\n#CHROM BEGIN   END     MARKER_ID       NS      FRAC_WITH_RARE     NUM_ALL_VARS    NUM_PASS_VARS   NUM_SING_VARS   PVALUE  STATRHO\n20     62607037        62608720        20:62607037-62608720_SAMD10     266     0.14662 9       5       1       0.0020064       1\n20     2816211 2820493 20:2816211-2820493_FAM113A      266     0.011278        12      2       1       0.0032542       0\n20     47245987        47361692        20:47245987-47361692_PREX1      266     0.1391  54      9       6       0.0054849       1\n20     34761734        34810279        20:34761734-34810279_EPB41L1    266     0.071429        14      7       5       0.0068492       0.2\n20     61340671        61391602        20:61340671-61391602_NTSR1      266     0.11278 24      9       3       0.011063        1\n20     48561952        48568644        20:48561952-48568644_RNF114     266     0.011278        4       2       1       0.015175        0.2\n20     60962895        60963559        20:60962895-60963559_RPS21      266     0.06015 6       3       2       0.016409        0\n20     55904961        55917801        20:55904961-55917801_SPO11      266     0.011278        11      3       3       0.018031        0\n\nThe key columns represents:\n\nNS : Number of phenotyped samples with non-missing genotypes\nFRAC_WITH_RARE : Fraction of individual carrying rare variants below –max-maf (default : 0.05) threshold.\nNUM_ALL_VARS : Number of all variants defining the group.\nNUM_PASS_VARS : Number of variants passing the –min-maf, –min-mac, –max-maf, –min-callrate thresholds\nNUM_SING_VARS : Number of singletons among variants in NUM_PASS_VARSPVALUE : P-value of burden tests\nOther columns are test specific auxiliary columns. For example, in the VT test, the optimal MAF threshold is recorded as an auxiliary output column.\n\n4.4 Specialized Instruction for EMMAX testsEMMAX (Efficient Mixed Model Association eXpedited - Kang et al (2010) Nat Genet 42:348-54) is an efficient implementation of mixed model association accounting for sample structure including population structure and hidden relatedness. Currently EPACTS supports EMMAX association mapping in single variant test and CMC-like burden tests.\nBecause EMMAX is based on linear model, the method fits better to quantiative traits than binary traits. However, p-values for binary traits are expected to be valid in the spirit of Armitage trend test, although the estimated effect size may not be precise.\nIn order to run EMMAX analysis from sequence-based genotypes. We recommend running EPACTS multiple times using the following procedure.\n4.4.1 Single Variant EMMAX Association Analysis\nCreating Kinship Matrix : From VCF, we recommend to set a MAF (e.g. 0.01) and call rate (e.g. 0.95) threshold to select high-quality markers to generate kinship matrix as follows:\n\n$&#123;EPACTS_DIR&#125;/epacts make-kin \\\n --vcf [input.vcf.gz] --ped  [input.ped (Optional)] --min-maf 0.01 --minCallRate 0.95 \\\n --sepchr (if VCF is separated by chromosome) --out [outprefix.kinf] --run [# of parallel jobs]\n\nIf you provide [input.ped] file, then it will calculate the subset the individuals contained in the PED file.\nThe procedure above will create a file [outprefix.kinf] after splitting and merging the genomes into multiple pieces. If only a certain subset of SNPs needs to be considered due to target regions, LD-pruning, or any other reasons, a VCF containing the subset of markers must be created beforehand and should be used as input VCF file.\n\nPerform Single Variant Association : From VCF and PED, we recommend to use less stringent MAF threshold (e.g. 0.001) and call rate (e.g. 0.50) to perform single variant association\n\n$&#123;EPACTS_DIR&#125;/epacts single \\\n --vcf [input.vcf.gz] --ped  [input.ped] --min-maf 0.001 --kin [outputprefix.kinf] \\\n --sepchr --pheno [PHENO_NAME] --cov [COV1] --cov [COV2] --test q.emmax \\\n --out [outprefix] --run [# of parallel jobs]\n\nThe procedure above will perform single variant association analysis compatible to other types of single variant association analyses implemented in EPACTS\n4.4.2 Burden-style EMMAX Association AnalysisIn order to run EMMAX analysis from sequence-based genotypes. We recommend running EPACTS multiple times using the following procedure.\nCreating Kinship Matrix : See ‘Creating Kinship Matrix’ section in #Single Variant EMMAX Association AnalysisCreate Marker Group**\nBy annotating the VCF and extracting missense and nonsense variants: #Annotating VCF file using ANNOVAR - This step will be required to create marker group file #Creating marker group file - Assume that [group.grp] file is produced\nOr, by creating your own marker group information See #Creating marker group file for details\n\nRun CMC-style burden test by$&#123;EPACTS_DIR&#125;/epacts group --groupf [group.grp] \\\n --vcf [input.vcf.gz] --ped  [input.ped] --max-maf [max-MAF-for-rare-variants] \\\n --kin [outputprefix.kinf] --sepchr --pheno [PHENO_NAME] --cov [COV1] --cov [COV2] \\\n --test emmaxCMC --out [outprefix]\n\nRun Variable Threshold burden test by$&#123;EPACTS_DIR&#125;/epacts group --groupf [group.grp] \\\n --vcf [input.vcf.gz] --ped  [input.ped] --max-maf [max-MAF-for-rare-variants] \\\n --kin [outputprefix.kinf] --sepchr --pheno [PHENO_NAME] --cov [COV1] --cov [COV2] \\\n --test emmaxVT --out [outprefix]\n\n4.5 Preparing Your Own Input Data4.5.1 VCF file for GenotypesEPACTS support VCF files as input for association with the following requirement\nInput VCF file must be bgzipped and tabixed before running association to allow efficient random access of the file. Below is an example command to conver plain VCF into bgzipped and tabixed VCF\nbgzip input.vcf     ## this command will produce input.vcf.gz\ntabix -pvcf -f input.vcf.gz  ## this command will produce input.vcf.gz.tbi\n\n\nIf the VCF file is separated by chromosome, the VCF file specified in the input argument must contain the string “chr1” in the chromosome 1 file, and corresponding chromosome name for other chromosomes. Thus, the files names should be like [prefix]chr1[suffix].vcf.gz, [prefix]chr2[suffix].vcf.gz, …, [prefix]chr22[suffix].vcf.gz, [prefix]chrX[suffix].vcf.gz.\nSample IDs in the VCF file must be consistent to those from PED file\nCurrently EPACTS only support bi-allelic variants, but it handles SNPs, INDELs, snd SVs.\nCurrently, EPACTS only support VCF aligned with NCBI build 37 coordinates\nAn example VCF file from 1000 genome project is below.$ zcat example&#x2F;1000G_integrated_phase1_chr20.vcf.gz | cut -f 1-10 | head -50\n\n##fileformat=VCFv4.1\n##INFO=&lt;ID=LCSNP,Number=0,Type=Flag,Description=&quot;Genotype likelihood in Low coverage VCF in data integration&quot;&gt;\n##INFO=&lt;ID=EXSNP,Number=0,Type=Flag,Description=&quot;Genotype likelihood in Exome VCF in data integration&quot;&gt;\n##INFO=&lt;ID=INDEL,Number=0,Type=Flag,Description=&quot;Genotype likelihood in INDEL VCF in data integration&quot;&gt;\n##INFO=&lt;ID=SV,Number=0,Type=Flag,Description=&quot;Genotype likelihood in SV VCF in data integration&quot;&gt;\n##INFO=&lt;ID=BAVGPOST,Number=1,Type=Float,Description=&quot;Average posterior probability from beagle&quot;&gt;\n##INFO=&lt;ID=BRSQ,Number=1,Type=Float,Description=&quot;Genotype imputation quality estimate from beagle&quot;&gt;\n##INFO=&lt;ID=LDAF,Number=1,Type=Float,Description=&quot;MLE Allele Frequency Accounting for LD&quot;&gt;\n##INFO=&lt;ID=AVGPOST,Number=1,Type=Float,Description=&quot;Average posterior probability from MaCH/Thunder&quot;&gt;\n##INFO=&lt;ID=RSQ,Number=1,Type=Float,Description=&quot;Genotype imputation quality from MaCH/Thunder&quot;&gt;\n##INFO=&lt;ID=ERATE,Number=1,Type=Float,Description=&quot;Per-marker Mutation rate from MaCH/Thunder&quot;&gt;\n##INFO=&lt;ID=THETA,Number=1,Type=Float,Description=&quot;Per-marker Transition rate from MaCH/Thunder&quot;&gt;\n##INFO=&lt;ID=CIEND,Number=2,Type=Integer,Description=&quot;Confidence interval around END for imprecise variants&quot;&gt;\n##INFO=&lt;ID=CIPOS,Number=2,Type=Integer,Description=&quot;Confidence interval around POS for imprecise variants&quot;&gt;\n##INFO=&lt;ID=END,Number=1,Type=Integer,Description=&quot;End position of the variant described in this record&quot;&gt;\n##INFO=&lt;ID=HOMLEN,Number=.,Type=Integer,Description=&quot;Length of base pair identical micro-homology at event breakpoints&quot;&gt;\n##INFO=&lt;ID=HOMSEQ,Number=.,Type=String,Description=&quot;Sequence of base pair identical micro-homology at event breakpoints&quot;&gt;\n##INFO=&lt;ID=SOURCE,Number=.,Type=String,Description=&quot;Source of deletion call&quot;&gt;\n##INFO=&lt;ID=SVLEN,Number=1,Type=Integer,Description=&quot;Difference in length between REF and ALT alleles&quot;&gt;\n##INFO=&lt;ID=SVTYPE,Number=1,Type=String,Description=&quot;Type of structural variant&quot;&gt;\n##INFO=&lt;ID=AC,Number=.,Type=Integer,Description=&quot;Alternate Allele Count&quot;&gt;\n##INFO=&lt;ID=AN,Number=1,Type=Integer,Description=&quot;Total Allele Count&quot;&gt;\n##ALT=&lt;ID=DEL,Description=&quot;Deletion&quot;&gt;\n##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;\n##FORMAT=&lt;ID=DS,Number=1,Type=Float,Description=&quot;Genotype dosage from MaCH/Thunder&quot;&gt;\n##FORMAT=&lt;ID=GL,Number=.,Type=Float,Description=&quot;Genotype Likelihoods&quot;&gt;\n##FORMAT=&lt;ID=BD,Number=1,Type=Float,Description=&quot;Genotype dosage from beagle&quot;&gt;\n#CHROM POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  HG00096\n20 60479   .   C   T   100 PASS    LCSNP;EXSNP;BAVGPOST=1.000;BRSQ=0.894;LDAF=0.0020;AVGPOST=0.9995;RSQ=0.8779;ERATE=0.0005;THETA=0.0008;AC=4;AN=2184  GT:DS:GL:BD 0|0:0.000:-0.19,-0.46,-2.68:0.0022\n20 60522   .   T   TC  1588    PASS    INDEL;BAVGPOST=1.000;BRSQ=0.994;LDAF=0.0116;AVGPOST=0.9980;RSQ=0.9327;ERATE=0.0004;THETA=0.0167;AC=24;AN=2184   GT:DS:GL:BD 0|0:0.000:0.00,-0.90,-9.20:0\n20 60571   .   C   A   100 PASS    LCSNP;EXSNP;BAVGPOST=0.999;BRSQ=0.813;LDAF=0.0029;AVGPOST=0.9986;RSQ=0.8085;ERATE=0.0014;THETA=0.0014;AC=5;AN=2184  GT:DS:GL:BD 0|0:0.000:-0.05,-0.96,-5.00:0.0008\n20 60795   .   G   C   100 PASS    LCSNP;EXSNP;BAVGPOST=1.000;BRSQ=0.930;LDAF=0.0006;AVGPOST=0.9996;RSQ=0.7205;ERATE=0.0003;THETA=0.0041;AC=1;AN=2184  GT:DS:GL:BD 0|0:0.000:-0.03,-1.21,-5.00:0.0001\n20 60810   .   G   GA  127 PASS    INDEL;BAVGPOST=1.000;BRSQ=0.862;LDAF=0.0013;AVGPOST=0.9987;RSQ=0.5684;ERATE=0.0004;THETA=0.0061;AC=2;AN=2184    GT:DS:GL:BD 0|0:0.000:0.00,-1.80,-18.80:0\n\n4.5.2 PED file for Phenotypes and CovariatesEPACTS accepts a PED format supported by MERLIN or PLINK software to represent phenotypes. For example, the example.ped file and example.dat file can represent the phenotypes and corresponding column name (from 6th column and after).\n$ head example.ped\n13281  NA12344 NA12347 NA12348 1   1   94.17   66.1\n13281  NA12347 0   0   1   1   109.54  44.0\n13281  NA12348 0   0   2   2   119.40  46.6\n1328   NA06984 0   0   1   2   87.72   39.3\n1328   NA06989 0   0   2   1   100.60  41.7\n1328   NA12329 NA06984 NA06989 2   1   100.85  46.4\n13291  NA06986 0   0   1   2   91.94   61.9\n13291  NA06995 NA07435 NA07037 1   2   104.36  57.4\n13291  NA06997 NA06986 NA07045 2   2   107.53  53.1\n\n$ cat example.dat\nA DISEASE\nT QT\nT AGE\n\nEPACTS also accept a PED format with header information. The above file can be combined into one file as follows\n$ head data/1000G_dummy_pheno.ped\n#FAM_ID    IND_ID  FAT_ID  MOT_ID  SEX DISEASE QT  AGE\n13281  NA12344 NA12347 NA12348 1   1   94.17   66.1\n13281  NA12347 0   0   1   1   109.54  44.0\n13281  NA12348 0   0   2   2   119.40  46.6\n1328   NA06984 0   0   1   2   87.72   39.3\n1328   NA06989 0   0   2   1   100.60  41.7\n1328   NA12329 NA06984 NA06989 2   1   100.85  46.4\n13291  NA06986 0   0   1   2   91.94   61.9\n13291  NA06995 NA07435 NA07037 1   2   104.36  57.4\n13291  NA06997 NA06986 NA07045 2   2   107.53  53.1\nThe column names can be used to identify the names of phenotypes and covariates in the analysis.\n\n5. Detailed OptionsThe detailed options can viewed by running the following commands\n$&#123;EPACTS_DIR&#125;/bin/epacts -man           (for overall structure) \n$&#123;EPACTS_DIR&#125;/bin/epacts single -man    (for single variant test)\n$&#123;EPACTS_DIR&#125;/bin/epacts group -man     (for groupwise test)\n$&#123;EPACTS_DIR&#125;/bin/epacts anno -man      (for annotation)\n$&#123;EPACTS_DIR&#125;/bin/epacts plot -man      (for QQ and Manhattan plot)\n$&#123;EPACTS_DIR&#125;/bin/epacts zoom -man      (for zoom plot)\n$&#123;EPACTS_DIR&#125;/bin/epacts meta -man      (for meta-analysis)\n$&#123;EPACTS_DIR&#125;/bin/epacts make-group -man (for creating gene group)\n\n6. Implementing Additional Statistical TestsIn order to add additional statistical test to EPACTS, the following procedure are recommended\n\nCreate a file named ‘single.[testname].R’ for single variant test or ‘gene.[testname].R’ for gene-level test under ${EPACTS_DIR}&#x2F;share&#x2F;EPACTS&#x2F;\nTest your implementation using –test [testname] option to perform sanity check and debugging\nIf you want to add your test in the official in-house version, please send your code to Hyun\n\nBelow is an example of a single variant test implementation ( single.q.lm.R )\n## Core functions of EPACTS to perform association\n\n##################################################################\n## SINGLE VARIANT TEST\n## INPUT VARIABLES:\n##   n        : total # of individuals\n##   NS       : number of called samples\n##   AC       : allele count\n##   MAF      : minor allele frequency\n##   vids     : indices from 1:nrow(NS) after AF/AC threshold\n##   genos    : genotype matrix (after AF/AC threshold)\n## EXPECTED OUTPUT : list(p, addcols, addnames) for each genos row\n##   p        : p-value\n##   add      : additional columns to add\n##   cname    : column names for additional columns\n##################################################################  \n\n## single.lm() : Use built-in lm() function to perform association\n## KEY FEATURES : SIMPLE, BUT MAY BE SLOW\n##                GOOD SNIPPLET TO START A NEW FUNCTION\n## TRAITS  : QUANTITATIVE\n## RETURNS : PVALUE, BETA, SEBETA, TSTAT\n## MISSING VALUES : IGNORED\nsingle.q.lm &lt;- function() &#123;\n  cname &lt;- c(&quot;BETA&quot;,&quot;SEBETA&quot;,&quot;TSTAT&quot;) # column names for additional variables in the EPACTS output\n  m &lt;- nrow(genos)\n  p &lt;- rep(NA,m)\n  add &lt;- matrix(NA,m,3) ## BETA, SEBETA, TSTAT\n  if ( m &gt; 0 ) &#123;\n   for(i in 1:m) &#123;\n     r &lt;- summary(lm(pheno~genos[i,]+cov-1))$coefficients[1,]  # run simple linear regression\n     p[i] &lt;- r[4]   # store p-value to p[i]\n     add[i,] &lt;- r[1:3] # store additional variables to add[i,]\n   &#125;\n &#125;\n return(list(p=p,add=add,cname=cname))\n&#125;\n\nAs described in the comment, you may assume that the following variables are available for use for testing association across m markers\n\nn (scalar) : total number of individuals\nNS (M * 1 vector) : Number of called samples for each marker\nAC (M * 1 vector) : Non-reference allele count for each marker\nMAF (M * 1 vector) : Minor allele frequency\nvids (m * 1 vector) : indices of markers passing the inclusion criteria (e.g. MAF threshold) among 1:M\ngenos (m * n matrix) : genotype matrix as a input for association test\n\nThe output variables to generate is as follows\n\np (m * 1 vector) : p-value matrix as output\nadd (m * c matrix) : additional columns as output of test (such as SCORE, BETA, etc)\ncname (c * 1 vector) : column names of add\n\nIn the output files, the following columns will be displayed\n\nMARKER : Marker ID\nNS : Number of called samples\nAC : Non-ref allele count\nCALLRATE : Call rate &#x3D; NS&#x2F;n\nMAF : Minor allele frequency\nPVALUE : P-values\nAdditional columns specified by return values ‘add’\n\nBelow is an example of a gene-lvel variant test implementation ( single.q.lm.R )\n##################################################################\n## GENE-LEVEL BURDEN TEST\n## INPUT VARIABLES: \n##   n        : total # of individuals\n##   genos    : genotype matrix for each gene\n##   NS       : number of called samples for each marker\n##   AC       : allele count for each marker\n##   MAC      : minor allele count for each marker\n##   MAF      : minor allele frequency\n##   vids     : indices from 1:n after AF/AC threshold\n## EXPECTED OUTPUT : list(p, addcols, addnames) for each genos row\n##   p        : p-value\n##   add      : additional column values\n##   cname    : additional column names\n##################################################################      \n\n## gene.q.reverse() : Reverse logistic regression\n## KEY FEATURES : 0/1 collapsing variable ~ rare variants\n## TRAITS  : QUANTITATIVE (GAUSSIAN)\n## RETURNS : PVALUE, BETA, SEBETA, ZSTAT\n## MISSING VALUE : IMPUTED AS MAJOR ALLELES\ngene.q.reverse &lt;- function() &#123;\n  cname &lt;- c(&quot;BETA&quot;,&quot;SEBETA&quot;,&quot;ZSTAT&quot;)\n  m &lt;- nrow(genos)\n  if ( m &gt; 0 ) &#123;\n    g &lt;- as.double(colSums(genos,na.rm=T) &gt; 0)\n    sg &lt;- sum(g)\n    if ( ( sg &gt; 0 ) &amp;&amp; ( sg &lt; n ) ) &#123;\n      r &lt;- glm(g~pheno+cov-1,family=binomial)\n       if ( ( r$converged ) &amp;&amp; ( ! r$boundary ) ) &#123;\n        return(list(p=summary(r)$coefficients[1,4],\n                    add=summary(r)$coefficients[1,1:3],\n                    cname=cname))\n      &#125;\n    &#125;\n  &#125;\n  return(list(p=NA,add=rep(NA,3),cname=cname))\n&#125;\n\nSimilar to gene-level test, you may assume the following variables exist for testing A SINGLE GENE. Note that M is the number of markers spanning the gene region\n\nn (scalar) : total number of individuals\nNS (M * 1 vector) : Number of called samples for each marker\nAC (M * 1 vector) : Non-reference allele count for each marker\nMAC (M * 1 vector) : Minor allele count\nMAF (M * 1 vector) : Minor allele frequency\nvids (m * 1 vector) : indices of markers passing the inclusion criteria (e.g. MAF threshold) among 1:M\ngenos (m * n matrix) : genotype matrix as a input for association test\n\nThe output variables to generate is as follows\n\np (scalar) : p-value matrix as output\nadd (c * 1 vector) : additional columns as output of test (such as SCORE, BETA, etc)\ncname (c * 1 vector) : column names of add\nIn the output files, the following columns will be displayed\n\n\nMARKER : Marker ID\nNS : Number of called samples\nMAF_BURDEN : MAF of 0&#x2F;1 collapsing variables (existence of rare variants)\nNUM_ALL_VARS : Number of all variants within the gene\nNUM_RARE_VARS : Number of rare variants below the max-MAF threshold\nNUM_SING_VARS : Number of singleton variants\nPVALUE : P-value from the test\nAdditional columns specified by return values ‘add’\n\n","dateCreated":"2020-06-15T18:30:50+08:00","dateModified":"2020-06-15T18:30:50+08:00","datePublished":"2020-06-15T18:30:50+08:00","description":"","headline":"GWAS|EPACTS安装及使用","image":["qqplot.png"],"mainEntityOfPage":{"@type":"WebPage","@id":"https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"},"publisher":{"@type":"Organization","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg","logo":{"@type":"ImageObject","url":"coveriuu.jpg"}},"url":"https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/","thumbnailUrl":"qqplot.png"}</script>
    <meta property="og:type" content="blog">
<meta property="og:title" content="GWAS|EPACTS安装及使用">
<meta property="og:url" content="https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/index.html">
<meta property="og:site_name" content="iuu&#39;s blog">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://iuu19.github.io/GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8/qqplot.png">
<meta property="og:image" content="https://iuu19.github.io/GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8/mh.png">
<meta property="og:image" content="https://iuu19.github.io/GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8/Manhattan.png">
<meta property="article:published_time" content="2020-06-15T10:30:50.000Z">
<meta property="article:modified_time" content="2020-06-15T10:30:50.000Z">
<meta property="article:author" content="Dr.iuu">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://iuu19.github.io/GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8/qqplot.png">
    
    
        
    
    
        <meta property="og:image" content="https://iuu19.github.io/assets/images/coveriuu.jpg"/>
    
    
        <meta property="og:image" content="https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/qqplot.png"/>
        <meta class="swiftype" name="image" data-type="enum" content="https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/qqplot.png"/>
    
    
    
    <!--STYLES-->
    
<link rel="stylesheet" href="/assets/css/style-gf8wcm29a1hig4pgasvkpmpwcvjyqc9ufzpbo404wwnc2qcamao8rlll1csq.min.css">

    <!--STYLES END-->
    

    

    
<link rel="stylesheet" href="/css/prism-vsc-dark-plus.css" type="text/css">
<link rel="stylesheet" href="/css/prism-line-numbers.css" type="text/css"></head>

    <body>
        <div id="blog">
            <!-- Define author's picture -->


    
        
            
        
    

<header id="header" data-behavior="4">
    <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
    <div class="header-title">
        <a
            class="header-title-link"
            href="/"
            aria-label=""
        >
            iuu&#39;s blog
        </a>
    </div>
    
        
            <a
                class="header-right-picture "
                href="/all-about"
                aria-label="打开链接: /all-about"
            >
        
        
            <img class="header-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
        </a>
    
</header>

            <!-- Define author's picture -->



        
    

<nav id="sidebar" data-behavior="4">
    <div class="sidebar-container">
        
            <div class="sidebar-profile">
                <a
                    href="/#about"
                    aria-label="阅读有关作者的更多信息"
                >
                    <img class="sidebar-profile-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
                </a>
                <h4 class="sidebar-profile-name">Dr.iuu</h4>
                
                    <h5 class="sidebar-profile-bio"><p>NCMC-SCMC-SWCH</p>
</h5>
                
            </div>
        
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://iuu19.github.io"
                            
                            rel="noopener"
                            title="首页"
                        >
                        <i class="sidebar-button-icon fa fa-home" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">首页</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-categories"
                            
                            rel="noopener"
                            title="分类"
                        >
                        <i class="sidebar-button-icon fa fa-bookmark" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">分类</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-tags"
                            
                            rel="noopener"
                            title="标签"
                        >
                        <i class="sidebar-button-icon fa fa-tags" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">标签</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-archives"
                            
                            rel="noopener"
                            title="归档"
                        >
                        <i class="sidebar-button-icon fa fa-archive" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">归档</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-about"
                            
                            rel="noopener"
                            title="关于"
                        >
                        <i class="sidebar-button-icon fa fa-question" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">关于</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://github.com/iuu19/iuu19.github.io"
                            
                                target="_blank"
                            
                            rel="noopener"
                            title="GitHub"
                        >
                        <i class="sidebar-button-icon fab fa-github" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">GitHub</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/atom.xml"
                            
                            rel="noopener"
                            title="RSS"
                        >
                        <i class="sidebar-button-icon fa fa-rss" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">RSS</span>
                    </a>
            </li>
            
        </ul>
        
    </div>
</nav>

            
            <div id="main" data-behavior="4"
                 class="
                        hasCoverMetaIn
                        ">
                
<article class="post">
    
    
        <div class="post-header main-content-wrap text-center">
    
        <h1 class="post-title">
            GWAS|EPACTS安装及使用
        </h1>
    
    
        <div class="post-meta">
    <time datetime="2020-06-15T18:30:50+08:00">
	
		    6月 15, 2020
    	
    </time>
    
</div>

    
</div>

    
    <div class="post-content markdown">
        <div class="main-content-wrap">
            <font face="times new roman" size=4>

<span id="more"></span>

<p>GWAS|EPACTS安装及使用</p>
<h1 id="table-of-contents">目录</h1><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#1-Installation-Details"><span class="toc-text">1. Installation Details</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#2-Download-the-reference-FASTA-files"><span class="toc-text">2. Download the reference FASTA files</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#3-Perform-a-test-run"><span class="toc-text">3. Perform a test run</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#4-Getting-Started-With-Examples"><span class="toc-text">4. Getting Started With Examples</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#4-1-All-in-one-example"><span class="toc-text">4.1 All-in-one example</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-2-Single-Variant-Test"><span class="toc-text">4.2 Single Variant Test</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#4-2-1-Output-Text-of-All-Test-Statistics"><span class="toc-text">4.2.1 Output Text of All Test Statistics</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-2-2-Output-Text-of-Top-Associations"><span class="toc-text">4.2.2 Output Text of Top Associations</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-2-3-Q-Q-plot-of-test-statistics-stratified-by-MAF"><span class="toc-text">4.2.3 Q-Q plot of test statistics (stratified by MAF)</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-2-4-Manhattan-Plot-of-Test-Statistics"><span class="toc-text">4.2.4 Manhattan Plot of Test Statistics</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-3-Gene-wise-or-group-wise-burden-test"><span class="toc-text">4.3 Gene-wise or group-wise burden test</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#4-3-1-Creating-marker-group-file"><span class="toc-text">4.3.1 Creating marker group file</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-3-2-Annotating-VCF-file-using-EPACTS"><span class="toc-text">4.3.2 Annotating VCF file using EPACTS</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-3-3-Running-Groupwise-Test"><span class="toc-text">4.3.3 Running Groupwise Test</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-3-4-Example-Output"><span class="toc-text">4.3.4 Example Output</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-4-Specialized-Instruction-for-EMMAX-tests"><span class="toc-text">4.4 Specialized Instruction for EMMAX tests</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#4-4-1-Single-Variant-EMMAX-Association-Analysis"><span class="toc-text">4.4.1 Single Variant EMMAX Association Analysis</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-4-2-Burden-style-EMMAX-Association-Analysis"><span class="toc-text">4.4.2 Burden-style EMMAX Association Analysis</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Creating-Kinship-Matrix-See-%E2%80%98Creating-Kinship-Matrix%E2%80%99-section-in-Single-Variant-EMMAX-Association-Analysis"><span class="toc-text">Creating Kinship Matrix : See ‘Creating Kinship Matrix’ section in #Single Variant EMMAX Association Analysis</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Create-Marker-Group"><span class="toc-text">Create Marker Group**</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Run-CMC-style-burden-test-by"><span class="toc-text">Run CMC-style burden test by</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Run-Variable-Threshold-burden-test-by"><span class="toc-text">Run Variable Threshold burden test by</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-5-Preparing-Your-Own-Input-Data"><span class="toc-text">4.5 Preparing Your Own Input Data</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#4-5-1-VCF-file-for-Genotypes"><span class="toc-text">4.5.1 VCF file for Genotypes</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#4-5-2-PED-file-for-Phenotypes-and-Covariates"><span class="toc-text">4.5.2 PED file for Phenotypes and Covariates</span></a></li></ol></li></ol></li></ol></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#5-Detailed-Options"><span class="toc-text">5. Detailed Options</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#6-Implementing-Additional-Statistical-Tests"><span class="toc-text">6. Implementing Additional Statistical Tests</span></a></li></ol>
<hr>
<h1 id="1-Installation-Details"><a href="#1-Installation-Details" class="headerlink" title="1. Installation Details"></a>1. Installation Details</h1><p>If you want to use EPACTS in an Ubuntu platform, following the step below</p>
<pre><code>git clone https://github.com/statgen/EPACTS.git
mv EPACTS/ epacts
cd epacts
./configure --prefix=~/you/biosoft/epacts
make
make install
</code></pre>
<p><strong>Important Note</strong>: make sure to specify –prefix&#x3D;&#x2F;path&#x2F;to&#x2F;install。安装过程中有太多的坑：</p>
<ul>
<li>因为本服务器gcc是7.2.0，所以安装过程中总是安装不成功。去<a target="_blank" rel="noopener" href="http://groups.google.com/group/epacts">EPACTS Google group讨论区</a>说gcc需为gcc-5. gcc-5安装过程中也会有很多问题，具体可以关注公众号Gene2You，进行留言咨询。</li>
<li><strong>.&#x2F;configure</strong>过程中还会遇到其他问题，如需要安装”groff”，”gnuplot”，”No man pages found in package ‘epactsR’”，”make[2]: *** No rule to make target single.b.spa.R’, needed by all-am’. Stop.”等，可以查询<a target="_blank" rel="noopener" href="http://groups.google.com/group/epacts">EPACTS Google group讨论区</a>，或者关注公众号Gene2You，进行留言咨询。</li>
</ul>
<p>安装成功后：</p>
<pre><code># Now $&#123;EPACTS_DIR&#125; represents the &#39;/path/to/install&#39; directory
EPACTS_DIR=/WORK/Project/youguoling/biosoft/epacts
</code></pre>
<h1 id="2-Download-the-reference-FASTA-files"><a href="#2-Download-the-reference-FASTA-files" class="headerlink" title="2. Download the reference FASTA files"></a>2. Download the reference FASTA files</h1><ul>
<li><a target="_blank" rel="noopener" href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz">UCSC reference</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/projects/r_gencoll/ftp_service/nph-gc-ftp-service.cgi/?HistoryId=NCID_1_43150472_130.14.22.33_5555_1592229715_618157301_0MetA0_S_HStore&QueryKey=2&ReleaseType=RefSeq&FileType=GENOME_FASTA&Flat=true">NCBI reference</a></li>
<li><a href="ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz">Ensembl reference</a></li>
<li>如果Download the reference FASTA files from 1000 Genomes FTP automatically by running the following commands：</li>
</ul>
<pre><code> $&#123;EPACTS_DIR&#125;/bin/epacts download
</code></pre>
<h1 id="3-Perform-a-test-run"><a href="#3-Perform-a-test-run" class="headerlink" title="3. Perform a test run"></a>3. Perform a test run</h1><p>by running the following command:</p>
<pre><code>$&#123;EPACTS_DIR&#125;/bin/test_run_epacts.sh
</code></pre>
<h1 id="4-Getting-Started-With-Examples"><a href="#4-Getting-Started-With-Examples" class="headerlink" title="4. Getting Started With Examples"></a>4. Getting Started With Examples</h1><h2 id="4-1-All-in-one-example"><a href="#4-1-All-in-one-example" class="headerlink" title="4.1 All-in-one example"></a>4.1 All-in-one example</h2><p>To get started with EPACTS, run the following command will perform an example run</p>
<pre><code>$&#123;EPACTS_DIR&#125;/bin/test_run_epacts.sh
</code></pre>
<p>You will find a series of lines in <code>test_run_epacts.sh</code> script commented out for each possible test.</p>
<p>The example <strong>phenotype</strong> (<em>PED format</em>) and <strong>genotype</strong> (<em>VCF format</em>) can be found at</p>
<pre><code>$&#123;EPACTS_DIR&#125;/share/EPACTS/
</code></pre>
<h2 id="4-2-Single-Variant-Test"><a href="#4-2-Single-Variant-Test" class="headerlink" title="4.2 Single Variant Test"></a>4.2 Single Variant Test</h2><p>Or You can run EPACTS command yourself by running</p>
<pre><code>$&#123;EPACTS_DIR&#125;/epacts single \
  --vcf  $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.vcf.gz \
  --ped  $&#123;EPACTS_DIR&#125;/data/1000G_dummy_pheno.ped  \
  --min-maf 0.001 --chr 20 --pheno DISEASE --cov AGE --cov SEX --test b.score --anno \ 
  --out out/test --run 2
</code></pre>
<p>The command above will perform single variant association test using a dummy case-control phenotype file and a subset of 1000 genomes exome VCF file (chr20) using score test statistic for all variants over 1% of higher MAF using 2 parallel runs.</p>
<p><strong>You will see the 4 output files as the main outcome of the analysis</strong></p>
<h4 id="4-2-1-Output-Text-of-All-Test-Statistics"><a href="#4-2-1-Output-Text-of-All-Test-Statistics" class="headerlink" title="4.2.1 Output Text of All Test Statistics"></a>4.2.1 Output Text of All Test Statistics</h4><p>The filename is <code>out/test.single.b.score.epacts.gz</code> and the contents will look like:</p>
<pre><code>$ zcat out/test.single.b.score.epacts.gz | head
#CHROM	BEGIN	END	MARKER_ID	NS	AC	CALLRATE	MAF	PVALUE	SCORE	N.CASE	N.CTRL	AF.CASE	AF.CTRL
20	68303	68303	20:68303_A/G_Upstream:DEFB125	266	1	1	0.0018797	NA	NA	NA	NA	NA	NA
20	68319	68319	20:68319_C/A_Upstream:DEFB125	266	1.4467e-36	1	0	NA	NA	NA	NA	NA	NA
20	68396	68396	20:68396_C/T_Nonsynonymous:DEFB125	266	1	1	0.0018797	NA	NA	NA	NA	NA	NA
20	76635	76635	20:76635_A/T_Intron:DEFB125	266	1.534e-37	1	0	NA	NA	NA	NA	NA	NA
20	76689	76689	20:76689_T/C_Synonymous:DEFB125	266	0	1	0	NA	NA	NA	NA	NA	NA
20	76690	76690	20:76690_T/C_Nonsynonymous:DEFB125	266	1	1	0.0018797	NA	NA	NA	NA	NA	NA
20	76700	76700	20:76700_G/A_Nonsynonymous:DEFB125	266	0	1	0	NA	NA	NA	NA	NA	NA
20	76726	76726	20:76726_C/G_Nonsynonymous:DEFB125	266	0	1	0	NA	NA	NA	NA	NA	NA
20	76771	76771	20:76771_C/T_Nonsynonymous:DEFB125	266	3	1	0.0056391	0.68484	0.40587	145	121	0.013793	0.0082645
</code></pre>
<h4 id="4-2-2-Output-Text-of-Top-Associations"><a href="#4-2-2-Output-Text-of-Top-Associations" class="headerlink" title="4.2.2 Output Text of Top Associations"></a>4.2.2 Output Text of Top Associations</h4><p>Same type of file but containing top 5,000 association will be stored at <code>out/test.epacts.top5000</code></p>
<pre><code>$ head out/test.single.b.score.epacts.top5000 
#CHROM	BEGIN	END	MARKER_ID	NS	AC	CALLRATE	MAF	PVALUE	SCORE	N.CASE	N.CTRL	AF.CASE	AF.CTRL
20	1610894	1610894	20:1610894_G/A_Synonymous:SIRPG	266	138.64	1	0.26061	6.9939e-05	3.9765	145	121	0.65177	0.36476
20	4162411	4162411	20:4162411_T/C_Intron:SMOX	266	204	1	0.38346	0.00055583	-3.4523	145	121	0.62759	0.93388
20	34061918	34061918	20:34061918_T/C_Intron:CEP250	266	41.815	1	0.0786	0.00095471	3.3035	145	121	0.22543	0.075436
20	4155948	4155948	20:4155948_G/A_Intron:SMOX	266	215	1	0.40414	0.0020792	-3.0787	145	121	0.68276	0.95868
20	4680251	4680251	20:4680251_A/G_Nonsynonymous:PRNP	266	186	1	0.34962	0.0025962	3.0119	145	121	0.8069	0.57025
20	36668874	36668874	20:36668874_G/A_Synonymous:RPRD1B	266	96	1	0.18045	0.003031	2.9646	145	121	0.44828	0.2562
20	36641871	36641871	20:36641871_G/A_Synonymous:TTI1	266	10	1	0.018797	0.004308	-2.8547	145	121	0.0068966	0.07438
20	1616892	1616892	20:1616892_A/G_Synonymous:SIRPG	266	144	1	0.27068	0.0051239	2.7991	145	121	0.63449	0.42975
20	25038372	25038372	20:25038372_G/A_Intron:ACSS1	266	103.3	1	0.19418	0.005748	2.7618	145	121	0.47201	0.28813
</code></pre>
<p>The key columns represents:</p>
<ul>
<li><strong>NS:</strong> Number of phenotyped samples with non-missing genotypes</li>
<li><strong>AC:</strong> Total Non-reference Allele Count</li>
<li><strong>CALLRATE:</strong> Fraction of non-missing genotypes.</li>
<li><strong>MAF:</strong> Minor allele frequencies</li>
<li><strong>PVALUE:</strong> P-value of single variant test</li>
<li><strong>AF.CASE:</strong> Non-reference allele frequencies for cases</li>
<li><strong>AF.CTRL:</strong> Non-reference allele frequencies for controls</li>
</ul>
<h4 id="4-2-3-Q-Q-plot-of-test-statistics-stratified-by-MAF"><a href="#4-2-3-Q-Q-plot-of-test-statistics-stratified-by-MAF" class="headerlink" title="4.2.3 Q-Q plot of test statistics (stratified by MAF)"></a>4.2.3 Q-Q plot of test statistics (stratified by MAF)</h4><p>The file out&#x2F;test.b.score.epacts.qq.pdf will be generated as shown below:<br><img src="/GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8/qqplot.png" alt="Q-Q Plot"></p>
<h4 id="4-2-4-Manhattan-Plot-of-Test-Statistics"><a href="#4-2-4-Manhattan-Plot-of-Test-Statistics" class="headerlink" title="4.2.4 Manhattan Plot of Test Statistics"></a>4.2.4 Manhattan Plot of Test Statistics</h4><p>The file out&#x2F;test.b.score.epacts.mh.pdf will be generated for chr20 only.<br><img src="/GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8/mh.png" alt="Manhattan Plot"></p>
<p>An example Genome-wide manhattan plot (from a genome-wide run) will look like below<br><img src="/GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8/Manhattan.png" alt="Manhattan Plot"></p>
<h2 id="4-3-Gene-wise-or-group-wise-burden-test"><a href="#4-3-Gene-wise-or-group-wise-burden-test" class="headerlink" title="4.3 Gene-wise or group-wise burden test"></a>4.3 Gene-wise or group-wise burden test</h2><p>Gene-wise or group-wise burden test requires two steps. First, ‘group’ file containing the list of markers per group needs to be generated. Second, group-wise burden test needs to be run</p>
<h4 id="4-3-1-Creating-marker-group-file"><a href="#4-3-1-Creating-marker-group-file" class="headerlink" title="4.3.1 Creating marker group file"></a>4.3.1 Creating marker group file</h4><p>The marker group file has the following format</p>
<pre><code>[GROUP_ID]  [MARKER_ID_1]   [MARKER_ID_2]  .... [MARKER_ID_N]
</code></pre>
<p>where</p>
<ul>
<li><strong>[GROUP_ID]</strong> is a string representing the group (e.g. gene name)</li>
<li><strong>[MARKER_ID_K]</strong> is a marker key as a format of [CHROM]:[POS]_[REF]&#x2F;[ALT] (NOTE THAT THIS IS DIFFERENT FROM TYPICAL VCF MARKER ID field)</li>
</ul>
<p>Note that <strong>[MARKER_ID_K]</strong> has to be sorted by increasing order of genomic coordinate</p>
<p>In order to create gene-level group file from typically formatted VCF file, one may use the following utility</p>
<pre><code>$&#123;EPACTS_DIR&#125;/epacts make-group --vcf [input-vcf] --out [output-group-file] --format [epacts, annovar, chaos or gatk] --nonsyn
</code></pre>
<p>The above command create a file [output-group-file] containing a list of missense and nonsense variants per each gene. To incorporate different types of functional annotations, use –type option as follows</p>
<pre><code>$&#123;EPACTS_DIR&#125;/epacts make-group --vcf [input-vcf] --out [output-group-file] --format [epacts, annovar, chaos or gatk] --type [function_type_1] --type [function_type_2] ...
</code></pre>
<p>Type <code>&#39;epacts makegroup -man&#39;</code> for the detailed documentation</p>
<h4 id="4-3-2-Annotating-VCF-file-using-EPACTS"><a href="#4-3-2-Annotating-VCF-file-using-EPACTS" class="headerlink" title="4.3.2 Annotating VCF file using EPACTS"></a>4.3.2 Annotating VCF file using EPACTS</h4><p>If the VCF is not annotated, ‘epacts makegroup’ cannot be used. In order to annotate VCF, one can use the example VCF using ANNOVAR as follows:</p>
<pre><code>$&#123;EPACTS_DIR&#125;/epacts anno \
   --in $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.vcf.gz \
   --out $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.anno.vcf.gz
</code></pre>
<p>The epacts anno script will add “ANNO&#x3D;[function]:[genename]” entry into the INFO field based on gencodeV7 (default) or refGene database.</p>
<p>It is important to check whether the VCF file is already annotated or not in order to avoid no or redundant annotation.</p>
<h4 id="4-3-3-Running-Groupwise-Test"><a href="#4-3-3-Running-Groupwise-Test" class="headerlink" title="4.3.3 Running Groupwise Test"></a>4.3.3 Running Groupwise Test</h4><p>To perform a groupwise burden test on the example VCF (annotated as above), run the following command</p>
<pre><code>$&#123;EPACTS_DIR&#125;/epacts group --vcf $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.anno.vcf.gz \
  --groupf $&#123;EPACTS_DIR&#125;/data/1000G_exome_chr20_example_softFiltered.calls.anno.grp --out out/test.gene.skat \
  --ped $&#123;EPACTS_DIR&#125;/data/1000G_dummy_pheno.ped --maxAF 0.05 \
  --chr 20 --pheno QT --cov AGE --cov SEX --test skat --skat-o --run 2
</code></pre>
<h4 id="4-3-4-Example-Output"><a href="#4-3-4-Example-Output" class="headerlink" title="4.3.4 Example Output"></a>4.3.4 Example Output</h4><pre><code>$ head out/test.gene.skat.epacts.top5000
#CHROM BEGIN   END     MARKER_ID       NS      FRAC_WITH_RARE     NUM_ALL_VARS    NUM_PASS_VARS   NUM_SING_VARS   PVALUE  STATRHO
20     62607037        62608720        20:62607037-62608720_SAMD10     266     0.14662 9       5       1       0.0020064       1
20     2816211 2820493 20:2816211-2820493_FAM113A      266     0.011278        12      2       1       0.0032542       0
20     47245987        47361692        20:47245987-47361692_PREX1      266     0.1391  54      9       6       0.0054849       1
20     34761734        34810279        20:34761734-34810279_EPB41L1    266     0.071429        14      7       5       0.0068492       0.2
20     61340671        61391602        20:61340671-61391602_NTSR1      266     0.11278 24      9       3       0.011063        1
20     48561952        48568644        20:48561952-48568644_RNF114     266     0.011278        4       2       1       0.015175        0.2
20     60962895        60963559        20:60962895-60963559_RPS21      266     0.06015 6       3       2       0.016409        0
20     55904961        55917801        20:55904961-55917801_SPO11      266     0.011278        11      3       3       0.018031        0
</code></pre>
<p>The key columns represents:</p>
<ul>
<li><strong>NS :</strong> Number of phenotyped samples with non-missing genotypes</li>
<li><strong>FRAC_WITH_RARE :</strong> Fraction of individual carrying rare variants below –max-maf (default : 0.05) threshold.</li>
<li><strong>NUM_ALL_VARS :</strong> Number of all variants defining the group.</li>
<li><strong>NUM_PASS_VARS :</strong> Number of variants passing the –min-maf, –min-mac, –max-maf, –min-callrate thresholds</li>
<li><strong>NUM_SING_VARS :</strong> Number of singletons among variants in NUM_PASS_VARS<br>PVALUE : P-value of burden tests</li>
<li>Other columns are test specific auxiliary columns. For example, in the VT test, the optimal MAF threshold is recorded as an auxiliary output column.</li>
</ul>
<h2 id="4-4-Specialized-Instruction-for-EMMAX-tests"><a href="#4-4-Specialized-Instruction-for-EMMAX-tests" class="headerlink" title="4.4 Specialized Instruction for EMMAX tests"></a>4.4 Specialized Instruction for EMMAX tests</h2><p>EMMAX (Efficient Mixed Model Association eXpedited - Kang et al (2010) Nat Genet 42:348-54) is an efficient implementation of mixed model association accounting for sample structure including population structure and hidden relatedness. Currently EPACTS supports EMMAX association mapping in single variant test and CMC-like burden tests.</p>
<p>Because EMMAX is based on linear model, the method fits better to quantiative traits than binary traits. However, p-values for binary traits are expected to be valid in the spirit of Armitage trend test, although the estimated effect size may not be precise.</p>
<p>In order to run EMMAX analysis from sequence-based genotypes. We recommend running EPACTS multiple times using the following procedure.</p>
<h4 id="4-4-1-Single-Variant-EMMAX-Association-Analysis"><a href="#4-4-1-Single-Variant-EMMAX-Association-Analysis" class="headerlink" title="4.4.1 Single Variant EMMAX Association Analysis"></a>4.4.1 Single Variant EMMAX Association Analysis</h4><ul>
<li><strong>Creating Kinship Matrix :</strong> From VCF, we recommend to set a MAF (e.g. 0.01) and call rate (e.g. 0.95) threshold to select high-quality markers to generate kinship matrix as follows:</li>
</ul>
<pre><code>$&#123;EPACTS_DIR&#125;/epacts make-kin \
 --vcf [input.vcf.gz] --ped  [input.ped (Optional)] --min-maf 0.01 --minCallRate 0.95 \
 --sepchr (if VCF is separated by chromosome) --out [outprefix.kinf] --run [# of parallel jobs]
</code></pre>
<p>If you provide [input.ped] file, then it will calculate the subset the individuals contained in the PED file.</p>
<p>The procedure above will create a file [outprefix.kinf] after splitting and merging the genomes into multiple pieces. If only a certain subset of SNPs needs to be considered due to target regions, LD-pruning, or any other reasons, a VCF containing the subset of markers must be created beforehand and should be used as input VCF file.</p>
<ul>
<li><strong>Perform Single Variant Association :</strong> From VCF and PED, we recommend to use less stringent MAF threshold (e.g. 0.001) and call rate (e.g. 0.50) to perform single variant association</li>
</ul>
<pre><code>$&#123;EPACTS_DIR&#125;/epacts single \
 --vcf [input.vcf.gz] --ped  [input.ped] --min-maf 0.001 --kin [outputprefix.kinf] \
 --sepchr --pheno [PHENO_NAME] --cov [COV1] --cov [COV2] --test q.emmax \
 --out [outprefix] --run [# of parallel jobs]
</code></pre>
<p>The procedure above will perform single variant association analysis compatible to other types of single variant association analyses implemented in EPACTS</p>
<h4 id="4-4-2-Burden-style-EMMAX-Association-Analysis"><a href="#4-4-2-Burden-style-EMMAX-Association-Analysis" class="headerlink" title="4.4.2 Burden-style EMMAX Association Analysis"></a>4.4.2 Burden-style EMMAX Association Analysis</h4><p>In order to run EMMAX analysis from sequence-based genotypes. We recommend running EPACTS multiple times using the following procedure.</p>
<h5 id="Creating-Kinship-Matrix-See-‘Creating-Kinship-Matrix’-section-in-Single-Variant-EMMAX-Association-Analysis"><a href="#Creating-Kinship-Matrix-See-‘Creating-Kinship-Matrix’-section-in-Single-Variant-EMMAX-Association-Analysis" class="headerlink" title="Creating Kinship Matrix : See ‘Creating Kinship Matrix’ section in #Single Variant EMMAX Association Analysis"></a>Creating Kinship Matrix : See ‘Creating Kinship Matrix’ section in #Single Variant EMMAX Association Analysis</h5><h5 id="Create-Marker-Group"><a href="#Create-Marker-Group" class="headerlink" title="Create Marker Group**"></a>Create Marker Group**</h5><ul>
<li>By annotating the VCF and extracting missense and nonsense variants:<br> #Annotating VCF file using ANNOVAR - This step will be required to create marker group file<br> #Creating marker group file - Assume that [group.grp] file is produced</li>
<li>Or, by creating your own marker group information<br> See #Creating marker group file for details</li>
</ul>
<h5 id="Run-CMC-style-burden-test-by"><a href="#Run-CMC-style-burden-test-by" class="headerlink" title="Run CMC-style burden test by"></a>Run CMC-style burden test by</h5><pre><code>$&#123;EPACTS_DIR&#125;/epacts group --groupf [group.grp] \
 --vcf [input.vcf.gz] --ped  [input.ped] --max-maf [max-MAF-for-rare-variants] \
 --kin [outputprefix.kinf] --sepchr --pheno [PHENO_NAME] --cov [COV1] --cov [COV2] \
 --test emmaxCMC --out [outprefix]
</code></pre>
<h5 id="Run-Variable-Threshold-burden-test-by"><a href="#Run-Variable-Threshold-burden-test-by" class="headerlink" title="Run Variable Threshold burden test by"></a>Run Variable Threshold burden test by</h5><pre><code>$&#123;EPACTS_DIR&#125;/epacts group --groupf [group.grp] \
 --vcf [input.vcf.gz] --ped  [input.ped] --max-maf [max-MAF-for-rare-variants] \
 --kin [outputprefix.kinf] --sepchr --pheno [PHENO_NAME] --cov [COV1] --cov [COV2] \
 --test emmaxVT --out [outprefix]
</code></pre>
<h4 id="4-5-Preparing-Your-Own-Input-Data"><a href="#4-5-Preparing-Your-Own-Input-Data" class="headerlink" title="4.5 Preparing Your Own Input Data"></a>4.5 Preparing Your Own Input Data</h4><h5 id="4-5-1-VCF-file-for-Genotypes"><a href="#4-5-1-VCF-file-for-Genotypes" class="headerlink" title="4.5.1 VCF file for Genotypes"></a>4.5.1 VCF file for Genotypes</h5><p>EPACTS support VCF files as input for association with the following requirement</p>
<p>Input VCF file must be bgzipped and tabixed before running association to allow efficient random access of the file. Below is an example command to conver plain VCF into bgzipped and tabixed VCF</p>
<pre><code>bgzip input.vcf     ## this command will produce input.vcf.gz
tabix -pvcf -f input.vcf.gz  ## this command will produce input.vcf.gz.tbi
</code></pre>
<ul>
<li>If the VCF file is separated by chromosome, the VCF file specified in the input argument must contain the string “chr1” in the chromosome 1 file, and corresponding chromosome name for other chromosomes. Thus, the files names should be like [prefix]chr1[suffix].vcf.gz, [prefix]chr2[suffix].vcf.gz, …, [prefix]chr22[suffix].vcf.gz, [prefix]chrX[suffix].vcf.gz.</li>
<li>Sample IDs in the VCF file must be consistent to those from PED file</li>
<li>Currently EPACTS only support bi-allelic variants, but it handles SNPs, INDELs, snd SVs.</li>
<li>Currently, EPACTS only support VCF aligned with NCBI build 37 coordinates</li>
<li>An example VCF file from 1000 genome project is below.<br>$ zcat example&#x2F;1000G_integrated_phase1_chr20.vcf.gz | cut -f 1-10 | head -50</li>
</ul>
<pre><code>##fileformat=VCFv4.1
##INFO=&lt;ID=LCSNP,Number=0,Type=Flag,Description=&quot;Genotype likelihood in Low coverage VCF in data integration&quot;&gt;
##INFO=&lt;ID=EXSNP,Number=0,Type=Flag,Description=&quot;Genotype likelihood in Exome VCF in data integration&quot;&gt;
##INFO=&lt;ID=INDEL,Number=0,Type=Flag,Description=&quot;Genotype likelihood in INDEL VCF in data integration&quot;&gt;
##INFO=&lt;ID=SV,Number=0,Type=Flag,Description=&quot;Genotype likelihood in SV VCF in data integration&quot;&gt;
##INFO=&lt;ID=BAVGPOST,Number=1,Type=Float,Description=&quot;Average posterior probability from beagle&quot;&gt;
##INFO=&lt;ID=BRSQ,Number=1,Type=Float,Description=&quot;Genotype imputation quality estimate from beagle&quot;&gt;
##INFO=&lt;ID=LDAF,Number=1,Type=Float,Description=&quot;MLE Allele Frequency Accounting for LD&quot;&gt;
##INFO=&lt;ID=AVGPOST,Number=1,Type=Float,Description=&quot;Average posterior probability from MaCH/Thunder&quot;&gt;
##INFO=&lt;ID=RSQ,Number=1,Type=Float,Description=&quot;Genotype imputation quality from MaCH/Thunder&quot;&gt;
##INFO=&lt;ID=ERATE,Number=1,Type=Float,Description=&quot;Per-marker Mutation rate from MaCH/Thunder&quot;&gt;
##INFO=&lt;ID=THETA,Number=1,Type=Float,Description=&quot;Per-marker Transition rate from MaCH/Thunder&quot;&gt;
##INFO=&lt;ID=CIEND,Number=2,Type=Integer,Description=&quot;Confidence interval around END for imprecise variants&quot;&gt;
##INFO=&lt;ID=CIPOS,Number=2,Type=Integer,Description=&quot;Confidence interval around POS for imprecise variants&quot;&gt;
##INFO=&lt;ID=END,Number=1,Type=Integer,Description=&quot;End position of the variant described in this record&quot;&gt;
##INFO=&lt;ID=HOMLEN,Number=.,Type=Integer,Description=&quot;Length of base pair identical micro-homology at event breakpoints&quot;&gt;
##INFO=&lt;ID=HOMSEQ,Number=.,Type=String,Description=&quot;Sequence of base pair identical micro-homology at event breakpoints&quot;&gt;
##INFO=&lt;ID=SOURCE,Number=.,Type=String,Description=&quot;Source of deletion call&quot;&gt;
##INFO=&lt;ID=SVLEN,Number=1,Type=Integer,Description=&quot;Difference in length between REF and ALT alleles&quot;&gt;
##INFO=&lt;ID=SVTYPE,Number=1,Type=String,Description=&quot;Type of structural variant&quot;&gt;
##INFO=&lt;ID=AC,Number=.,Type=Integer,Description=&quot;Alternate Allele Count&quot;&gt;
##INFO=&lt;ID=AN,Number=1,Type=Integer,Description=&quot;Total Allele Count&quot;&gt;
##ALT=&lt;ID=DEL,Description=&quot;Deletion&quot;&gt;
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
##FORMAT=&lt;ID=DS,Number=1,Type=Float,Description=&quot;Genotype dosage from MaCH/Thunder&quot;&gt;
##FORMAT=&lt;ID=GL,Number=.,Type=Float,Description=&quot;Genotype Likelihoods&quot;&gt;
##FORMAT=&lt;ID=BD,Number=1,Type=Float,Description=&quot;Genotype dosage from beagle&quot;&gt;
#CHROM POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  HG00096
20 60479   .   C   T   100 PASS    LCSNP;EXSNP;BAVGPOST=1.000;BRSQ=0.894;LDAF=0.0020;AVGPOST=0.9995;RSQ=0.8779;ERATE=0.0005;THETA=0.0008;AC=4;AN=2184  GT:DS:GL:BD 0|0:0.000:-0.19,-0.46,-2.68:0.0022
20 60522   .   T   TC  1588    PASS    INDEL;BAVGPOST=1.000;BRSQ=0.994;LDAF=0.0116;AVGPOST=0.9980;RSQ=0.9327;ERATE=0.0004;THETA=0.0167;AC=24;AN=2184   GT:DS:GL:BD 0|0:0.000:0.00,-0.90,-9.20:0
20 60571   .   C   A   100 PASS    LCSNP;EXSNP;BAVGPOST=0.999;BRSQ=0.813;LDAF=0.0029;AVGPOST=0.9986;RSQ=0.8085;ERATE=0.0014;THETA=0.0014;AC=5;AN=2184  GT:DS:GL:BD 0|0:0.000:-0.05,-0.96,-5.00:0.0008
20 60795   .   G   C   100 PASS    LCSNP;EXSNP;BAVGPOST=1.000;BRSQ=0.930;LDAF=0.0006;AVGPOST=0.9996;RSQ=0.7205;ERATE=0.0003;THETA=0.0041;AC=1;AN=2184  GT:DS:GL:BD 0|0:0.000:-0.03,-1.21,-5.00:0.0001
20 60810   .   G   GA  127 PASS    INDEL;BAVGPOST=1.000;BRSQ=0.862;LDAF=0.0013;AVGPOST=0.9987;RSQ=0.5684;ERATE=0.0004;THETA=0.0061;AC=2;AN=2184    GT:DS:GL:BD 0|0:0.000:0.00,-1.80,-18.80:0
</code></pre>
<h5 id="4-5-2-PED-file-for-Phenotypes-and-Covariates"><a href="#4-5-2-PED-file-for-Phenotypes-and-Covariates" class="headerlink" title="4.5.2 PED file for Phenotypes and Covariates"></a>4.5.2 PED file for Phenotypes and Covariates</h5><p>EPACTS accepts a PED format supported by MERLIN or PLINK software to represent phenotypes. For example, the example.ped file and example.dat file can represent the phenotypes and corresponding column name (from 6th column and after).</p>
<pre><code>$ head example.ped
13281  NA12344 NA12347 NA12348 1   1   94.17   66.1
13281  NA12347 0   0   1   1   109.54  44.0
13281  NA12348 0   0   2   2   119.40  46.6
1328   NA06984 0   0   1   2   87.72   39.3
1328   NA06989 0   0   2   1   100.60  41.7
1328   NA12329 NA06984 NA06989 2   1   100.85  46.4
13291  NA06986 0   0   1   2   91.94   61.9
13291  NA06995 NA07435 NA07037 1   2   104.36  57.4
13291  NA06997 NA06986 NA07045 2   2   107.53  53.1
</code></pre>
<pre><code>$ cat example.dat
A DISEASE
T QT
T AGE
</code></pre>
<p>EPACTS also accept a PED format with header information. The above file can be combined into one file as follows</p>
<pre><code>$ head data/1000G_dummy_pheno.ped
#FAM_ID    IND_ID  FAT_ID  MOT_ID  SEX DISEASE QT  AGE
13281  NA12344 NA12347 NA12348 1   1   94.17   66.1
13281  NA12347 0   0   1   1   109.54  44.0
13281  NA12348 0   0   2   2   119.40  46.6
1328   NA06984 0   0   1   2   87.72   39.3
1328   NA06989 0   0   2   1   100.60  41.7
1328   NA12329 NA06984 NA06989 2   1   100.85  46.4
13291  NA06986 0   0   1   2   91.94   61.9
13291  NA06995 NA07435 NA07037 1   2   104.36  57.4
13291  NA06997 NA06986 NA07045 2   2   107.53  53.1
The column names can be used to identify the names of phenotypes and covariates in the analysis.
</code></pre>
<h1 id="5-Detailed-Options"><a href="#5-Detailed-Options" class="headerlink" title="5. Detailed Options"></a>5. Detailed Options</h1><p>The detailed options can viewed by running the following commands</p>
<pre><code>$&#123;EPACTS_DIR&#125;/bin/epacts -man           (for overall structure) 
$&#123;EPACTS_DIR&#125;/bin/epacts single -man    (for single variant test)
$&#123;EPACTS_DIR&#125;/bin/epacts group -man     (for groupwise test)
$&#123;EPACTS_DIR&#125;/bin/epacts anno -man      (for annotation)
$&#123;EPACTS_DIR&#125;/bin/epacts plot -man      (for QQ and Manhattan plot)
$&#123;EPACTS_DIR&#125;/bin/epacts zoom -man      (for zoom plot)
$&#123;EPACTS_DIR&#125;/bin/epacts meta -man      (for meta-analysis)
$&#123;EPACTS_DIR&#125;/bin/epacts make-group -man (for creating gene group)
</code></pre>
<h1 id="6-Implementing-Additional-Statistical-Tests"><a href="#6-Implementing-Additional-Statistical-Tests" class="headerlink" title="6. Implementing Additional Statistical Tests"></a>6. Implementing Additional Statistical Tests</h1><p>In order to add additional statistical test to EPACTS, the following procedure are recommended</p>
<ol>
<li>Create a file named ‘single.[testname].R’ for single variant test or ‘gene.[testname].R’ for gene-level test under ${EPACTS_DIR}&#x2F;share&#x2F;EPACTS&#x2F;</li>
<li>Test your implementation using –test [testname] option to perform sanity check and debugging</li>
<li>If you want to add your test in the official in-house version, please send your code to Hyun</li>
</ol>
<p>Below is an example of a single variant test implementation ( single.q.lm.R )</p>
<pre><code>## Core functions of EPACTS to perform association

##################################################################
## SINGLE VARIANT TEST
## INPUT VARIABLES:
##   n        : total # of individuals
##   NS       : number of called samples
##   AC       : allele count
##   MAF      : minor allele frequency
##   vids     : indices from 1:nrow(NS) after AF/AC threshold
##   genos    : genotype matrix (after AF/AC threshold)
## EXPECTED OUTPUT : list(p, addcols, addnames) for each genos row
##   p        : p-value
##   add      : additional columns to add
##   cname    : column names for additional columns
##################################################################  

## single.lm() : Use built-in lm() function to perform association
## KEY FEATURES : SIMPLE, BUT MAY BE SLOW
##                GOOD SNIPPLET TO START A NEW FUNCTION
## TRAITS  : QUANTITATIVE
## RETURNS : PVALUE, BETA, SEBETA, TSTAT
## MISSING VALUES : IGNORED
single.q.lm &lt;- function() &#123;
  cname &lt;- c(&quot;BETA&quot;,&quot;SEBETA&quot;,&quot;TSTAT&quot;) # column names for additional variables in the EPACTS output
  m &lt;- nrow(genos)
  p &lt;- rep(NA,m)
  add &lt;- matrix(NA,m,3) ## BETA, SEBETA, TSTAT
  if ( m &gt; 0 ) &#123;
   for(i in 1:m) &#123;
     r &lt;- summary(lm(pheno~genos[i,]+cov-1))$coefficients[1,]  # run simple linear regression
     p[i] &lt;- r[4]   # store p-value to p[i]
     add[i,] &lt;- r[1:3] # store additional variables to add[i,]
   &#125;
 &#125;
 return(list(p=p,add=add,cname=cname))
&#125;
</code></pre>
<p>As described in the comment, you may assume that the following variables are available for use for testing association across m markers</p>
<ul>
<li>n (scalar) : total number of individuals</li>
<li>NS (M * 1 vector) : Number of called samples for each marker</li>
<li>AC (M * 1 vector) : Non-reference allele count for each marker</li>
<li>MAF (M * 1 vector) : Minor allele frequency</li>
<li>vids (m * 1 vector) : indices of markers passing the inclusion criteria (e.g. MAF threshold) among 1:M</li>
<li>genos (m * n matrix) : genotype matrix as a input for association test</li>
</ul>
<p>The output variables to generate is as follows</p>
<ul>
<li>p (m * 1 vector) : p-value matrix as output</li>
<li>add (m * c matrix) : additional columns as output of test (such as SCORE, BETA, etc)</li>
<li>cname (c * 1 vector) : column names of add</li>
</ul>
<p>In the output files, the following columns will be displayed</p>
<ol>
<li>MARKER : Marker ID</li>
<li>NS : Number of called samples</li>
<li>AC : Non-ref allele count</li>
<li>CALLRATE : Call rate &#x3D; NS&#x2F;n</li>
<li>MAF : Minor allele frequency</li>
<li>PVALUE : P-values</li>
<li>Additional columns specified by return values ‘add’</li>
</ol>
<p>Below is an example of a gene-lvel variant test implementation ( single.q.lm.R )</p>
<pre><code>##################################################################
## GENE-LEVEL BURDEN TEST
## INPUT VARIABLES: 
##   n        : total # of individuals
##   genos    : genotype matrix for each gene
##   NS       : number of called samples for each marker
##   AC       : allele count for each marker
##   MAC      : minor allele count for each marker
##   MAF      : minor allele frequency
##   vids     : indices from 1:n after AF/AC threshold
## EXPECTED OUTPUT : list(p, addcols, addnames) for each genos row
##   p        : p-value
##   add      : additional column values
##   cname    : additional column names
##################################################################      

## gene.q.reverse() : Reverse logistic regression
## KEY FEATURES : 0/1 collapsing variable ~ rare variants
## TRAITS  : QUANTITATIVE (GAUSSIAN)
## RETURNS : PVALUE, BETA, SEBETA, ZSTAT
## MISSING VALUE : IMPUTED AS MAJOR ALLELES
gene.q.reverse &lt;- function() &#123;
  cname &lt;- c(&quot;BETA&quot;,&quot;SEBETA&quot;,&quot;ZSTAT&quot;)
  m &lt;- nrow(genos)
  if ( m &gt; 0 ) &#123;
    g &lt;- as.double(colSums(genos,na.rm=T) &gt; 0)
    sg &lt;- sum(g)
    if ( ( sg &gt; 0 ) &amp;&amp; ( sg &lt; n ) ) &#123;
      r &lt;- glm(g~pheno+cov-1,family=binomial)
       if ( ( r$converged ) &amp;&amp; ( ! r$boundary ) ) &#123;
        return(list(p=summary(r)$coefficients[1,4],
                    add=summary(r)$coefficients[1,1:3],
                    cname=cname))
      &#125;
    &#125;
  &#125;
  return(list(p=NA,add=rep(NA,3),cname=cname))
&#125;
</code></pre>
<p>Similar to gene-level test, you may assume the following variables exist for testing A SINGLE GENE. Note that M is the number of markers spanning the gene region</p>
<ul>
<li>n (scalar) : total number of individuals</li>
<li>NS (M * 1 vector) : Number of called samples for each marker</li>
<li>AC (M * 1 vector) : Non-reference allele count for each marker</li>
<li>MAC (M * 1 vector) : Minor allele count</li>
<li>MAF (M * 1 vector) : Minor allele frequency</li>
<li>vids (m * 1 vector) : indices of markers passing the inclusion criteria (e.g. MAF threshold) among 1:M</li>
<li>genos (m * n matrix) : genotype matrix as a input for association test</li>
</ul>
<p>The output variables to generate is as follows</p>
<ul>
<li>p (scalar) : p-value matrix as output</li>
<li>add (c * 1 vector) : additional columns as output of test (such as SCORE, BETA, etc)</li>
<li>cname (c * 1 vector) : column names of add</li>
<li>In the output files, the following columns will be displayed</li>
</ul>
<ol>
<li>MARKER : Marker ID</li>
<li>NS : Number of called samples</li>
<li>MAF_BURDEN : MAF of 0&#x2F;1 collapsing variables (existence of rare variants)</li>
<li>NUM_ALL_VARS : Number of all variants within the gene</li>
<li>NUM_RARE_VARS : Number of rare variants below the max-MAF threshold</li>
<li>NUM_SING_VARS : Number of singleton variants</li>
<li>PVALUE : P-value from the test</li>
<li>Additional columns specified by return values ‘add’</li>
</ol>

            


        </div>
    </div>
    <div id="post-footer" class="post-footer main-content-wrap">
        
        
            <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-20-GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    data-tooltip="GWAS|Rvtests安装及使用教程"
                    aria-label="上一篇: GWAS|Rvtests安装及使用教程"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    data-tooltip="GWAS|GMMAT安装及使用指南"
                    aria-label="下一篇: GWAS|GMMAT安装及使用指南"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


        
        
    </div>
</article>



                <footer id="footer" class="main-content-wrap">
    <span class="copyrights">
        Copyrights &copy; 2022 Dr.iuu. All Rights Reserved.
    </span>
</footer>

            </div>
            
                <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
                    <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-20-GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    data-tooltip="GWAS|Rvtests安装及使用教程"
                    aria-label="上一篇: GWAS|Rvtests安装及使用教程"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    data-tooltip="GWAS|GMMAT安装及使用指南"
                    aria-label="下一篇: GWAS|GMMAT安装及使用指南"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


                </div>
                
    <div id="share-options-bar" class="share-options-bar" data-behavior="4">
        <i id="btn-close-shareoptions" class="fa fa-times"></i>
        <ul class="share-options">
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                        aria-label="分享到 Facebook"
                    >
                        <i class="fab fa-facebook" aria-hidden="true"></i><span>分享到 Facebook</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                        aria-label="分享到 Twitter"
                    >
                        <i class="fab fa-twitter" aria-hidden="true"></i><span>分享到 Twitter</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                        aria-label="分享到 Google+"
                    >
                        <i class="fab fa-google-plus" aria-hidden="true"></i><span>分享到 Google+</span>
                    </a>
                </li>
            
        </ul>
    </div>


            
        </div>
        


    
        
    

<div id="about">
    <div id="about-card">
        <div id="about-btn-close">
            <i class="fa fa-times"></i>
        </div>
        
            <img id="about-card-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
            <h4 id="about-card-name">Dr.iuu</h4>
        
            <div id="about-card-bio"><p>NCMC-SCMC-SWCH</p>
</div>
        
        
            <div id="about-card-job">
                <i class="fa fa-briefcase"></i>
                <br/>
                <p>Bioinfo-CHD</p>

            </div>
        
        
            <div id="about-card-location">
                <i class="fa fa-map-marker-alt"></i>
                <br/>
                Pudong, Shanghai
            </div>
        
    </div>
</div>

        
        
<div id="cover" style="background-image:url('/assets/images/cover2.jpg');"></div>
        <!--SCRIPTS-->

<script src="/assets/js/script-3nlvflbbvekvxx0g6h2clfqhhxs61cvvbofv7nmctigkdbbhu5htlvmrqg80.min.js"></script>

<!--SCRIPTS END-->





    </body>
</html>
