
<!DOCTYPE html>
<html lang="zh-cn">
    
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="iuu&#39;s blog">
    <title>Ancestry祖源分析|LASER - iuu&#39;s blog</title>
    <meta name="author" content="Dr.iuu">
    
    
        <link rel="icon" href="https://iuu19.github.io/assets/images/coveriuu.jpg">
    
    
        
            <link rel="alternate" type="application/atom+xml" title="RSS" href="/atom.xml">
        
    
    <script type="application/ld+json">{"@context":"http://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg"},"articleBody":"\n\n\n\nAncestry祖源分析|LASER\n\n\n1. 介绍LASER (Locating Ancestry using SEquencing Reads)，是C++编译的软件包，可以直接从全基因组测序读取估算祖先，而无需调用基因型。该方法依赖于一组参考个体，这些参考个体的全基因组SNP基因型和祖先信息是已知的。我们首先通过将主成分分析（PCA）应用于参考个体的基因型数据来构建参考坐标系。\n2. 下载LASER Package: Version 2.03\nwget -c https://sph.umich.edu/csg/software_pages/softwaredownloads/LASER-2.03.tar.gz\ntar xzvf LASER-2.03.tar.gz\nmv LASER-2.03 laser\n\n# version-2.04: http://csg.sph.umich.edu/chaolong/LASER/LASER-2.04.tar.gz\n\n3. LASER workflow\n4. HGDP reference panelHGDP data: Include 632,958 autosomal SNPs for 938 unrelated individuals from the Human Genome Diversity Project (HGDP).\n\n5. Preparing input files for LASER5.1 liftOver HGDP hg19tohg385.1.1 download liftOverliftOver file:http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/\nwget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver\n\n5.1.2 download HGDP reference file#wget -c http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/hg19ToHg38.over.chain.gz\n\nLIFTOVER=/WORK/Project/youguoling/biosoft/liftover\n\n$LIFTOVER/liftOver.txt $LASERF/HGDP/HGDP_938chr.site hg19ToHg38.over.chain.gz HGDP_938chrhg38.site unlifted.bed\n\n$LIFTOVER/liftOver.txt $LASERF/HGDP/HGDP_938chr.bed hg19ToHg38.over.chain.gz HGDP_938chrhg38.bed unlifted.bed\n\n5.1.3 或者 vcf –&gt; genoThis step prepares the reference panel by converting a VCF genotype file to a GENO file. We will skip this step and use a ready-to-use HGDP reference panel. A typical command to run the vcf2geno tool is given in the file “$LASERF&#x2F;vcf2geno&#x2F;cmd.sh”:\n\n# $LASERF/vcf2geno/vcf2geno --inVcf exampleVCF/example.vcf.gz --updateID test.updateId --out test\n\n5.2 : bam –&gt; pileupThis step uses samtools to generate pileup files from bam files.\nsamtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101035.recal.bam &gt; 121101035.recal.pileup\nsamtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101043.recal.bam &gt; 121101043.recal.pileup \nsamtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101050.recal.bam &gt; 121101050.recal.pileup\nsamtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101052.recal.bam &gt; 121101052.recal.pileup\nsamtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101415.recal.bam &gt; 121101415.recal.pileup\nsamtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101861.recal.bam &gt; 121101861.recal.pileup\n\nWe use -q 30 and -Q 20 to exclude reads that have mapping quality score lower than 30 or base quality score lower than 20.\n5.3 : pileup –&gt; seq这一步生成hapmap_trios.seq.\npython $LASER/pileup2seq/pileup2seq.py \\\n-m $HGDP/HGDP_938.site \\\n-b $BAM/AMD_roi_1-based.bed \\\n-i $BAM/AMD_hapmap_trios_id.txt \\\n-o hapmap_trios \\\n$BAM/121101035.recal.pileup \\\n$BAM/121101043.recal.pileup \\\n$BAM/121101050.recal.pileup \\\n$BAM/121101052.recal.pileup \\\n$BAM/121101415.recal.pileup \\\n$BAM/121101861.recal.pileup\n\n-b provides the targeted regions to exclude and -i specifies alternative IDs for the BAM files to be used in the .seq file (including popID and indivID). -b and -i are optional.\n6. Estimating ancestry coordinates6.1 Generate the reference ancestry spaceLASER can perform principal components analysis (PCA) on genotype data of the reference panel to generate a reference ancestry space.\n# $LASER/laser -g $HGDP/HGDP_938.geno -pca 1 -k 30 -o HGDP_938\n\nThe above command takes ~20 minutes to finish. We will skip this step, and use a set of reference ancestry coordinates that have been generated in the file $HGDP/HGDP_938.RefPC.coord.View the reference coordinates:\nless -S $HGDP/HGDP_938.RefPC.coord\n\n6.2 Estimate ancestry for sequenced samplesSubmit two jobs to place sequenced samples into the reference ancestry space:\n$LASER/laser -g $HGDP/HGDP_938.geno -c $HGDP/HGDP_938.RefPC.coord -s hapmap_trios.seq -K 20 -k 4 -x 1 -y 3 -o hapmap_trios.1-3 &amp;\n$LASER/laser -g $HGDP/HGDP_938.geno -c $HGDP/HGDP_938.RefPC.coord -s hapmap_trios.seq -K 20 -k 4 -x 4 -y 6 -o hapmap_trios.4-6 &amp;\n\nThe first job will process samples 1 to 3 and the second job will processed samples 4 to 6. Each sequenced sample will be projected from a 20-dimensional PCA space onto a 4-dimensional reference ancestry space. The running time is ~10 minutes for processing 3 samples in each job.\n6.3 Combine resultsResults from previous step will be output to two files hapmap_trios.1-3.SeqPC.coord and hapmap_trios.4-6.SeqPC.coord. Here we simply concatenate the two files while skipping the header line of the second file.\ncp hapmap_trios.1-3.SeqPC.coord hapmap_trios.SeqPC.coord\nmore +2 hapmap_trios.4-6.SeqPC.coord &gt;&gt; hapmap_trios.SeqPC.coord\n\nView the results:\nless -S hapmap_trios.SeqPC.coord\n\nThe results should look like below (results will vary slightly): \npopID  indivID  L1      Ci         K     t          PC1        PC2         PC3         PC4\nYRI    NA19238  78386   0.170864   20    0.999688   467.989    -210.294    -14.1729    -14.4204\nCEU    NA12892  85486   0.185973   20    0.999723   10.796     199.095     -9.90387    -21.4534\nCEU    NA12891  87588   0.190442   20    0.99973    2.04224    196.07      -19.5705    -12.8022\nCEU    NA12878  83213   0.181748   20    0.999711   4.34591    199.861     -12.4825    -22.6281\nYRI    NA19239  87564   0.193424   20    0.999734   474.464    -215.96     -9.02921    -19.7372\nYRI    NA19240  95866   0.213874   20    0.999748   469.914    -214.94     -14.9923    -13.6559\n\n7. Visualizing resultsExample R codes are available in .&#x2F;LASER-2.01&#x2F;plot&#x2F;. Let’s copy the folder to current working directory:\ncp -r $LASER/plot/ ./\n\nGo to the plot folder and run the script to plot results:\ncd plot\nRscript plotHGDP.r $HGDP/HGDP_938.RefPC.coord ../hapmap_trios.SeqPC.coord\n\nA figure named Results_on_HGDP.pdf will be generated. Visualize the figure:\nevince Results_on_HGDP.pdf &amp;\n\nWe expect to see the following figure, in which 3 CEU samples cluster with HGDP Europeans and 3 YRI samples cluster with HGDP Africans:\n\n8. File format8.1 Geno fileGeno file are from reference samples. LASER use genotype of these samples as a reference panel. You can obtain geno file from VCF files using vcf2geno.\nAn example geno file for the HGDP data set (resource&#x2F;HGDP&#x2F;HGDP_938.geno):\nBrahui\tHGDP00001\t1\t2\t1\t1\t0\t2\t0\t2\t1\t2\t2\t2\t1\t1\t2\t1\t0\nBrahui\tHGDP00003\t0\t0\t2\t0\t0\t2\t0\t2\t0\t2\t2\t2\t2\t0\t2\t2\t0\nBrahui\tHGDP00005\t0\t2\t2\t0\t0\t1\t0\t2\t1\t2\t2\t2\t2\t1\t2\t2\t1\nBrahui\tHGDP00007\t0\t2\t2\t0\t0\t2\t0\t2\t0\t2\t2\t2\t1\t1\t2\t2\t1\nBrahui\tHGDP00009\t0\t1\t0\t1\t0\t2\t0\t2\t0\t2\t2\t2\t2\t0\t2\t2\t0\nBrahui\tHGDP00011\t1\t1\t2\t1\t1\t2\t1\t1\t1\t2\t2\t2\t1\t1\t2\t2\t0\nBrahui\tHGDP00013\t1\t2\t2\t1\t1\t2\t1\t2\t0\t2\t2\t2\t2\t0\t2\t2\t0\nBrahui\tHGDP00015\t1\t1\t2\t0\t0\t2\t0\t2\t0\t2\t2\t2\t2\t0\t2\t2\t0\nBrahui\tHGDP00017\t1\t1\t2\t0\t0\t1\t0\t0\t0\t2\t0\t1\t1\t2\t2\t2\t0\nBrahui\tHGDP00019\t0\t2\t2\t0\t0\t1\t0\t1\t0\t2\t1\t2\t2\t1\t2\t2\t0\n\nThe first and second columns represent the population id and individual id. From the third column, each number represents a genotype. To be consistent with the sequence data, genotypes should be given on the forward strand. Genotypes are coded by 0, 1, or 2, representing copies of the reference allele at a locus in one individual.\nIn this geno file, we have 632,960 columns which contains 632,958 markers from column 3 to the last column.\n8.2 Seq fileSeq file is generated from pileup files. It contains sequencing information and organize it in a LASER readable format. The first two columns represent population id and individual id. Subsequent columns are total read depths and reference base counts. For example, column 3 and 4 are 0, 0 in the following example. That means at first marker, the sequence read depth is 0 and thus none of the reads has reference base. We enforce tab delimiters between markers and space delimiters between each read depths and reference base counts. On line of seq file looks like below:\nNA12878.chrom22\tNA12878.chrom22\t0 0\t0 0\t0 0\t0 0\t0 \n\n8.3 Pileup filePileup file are generated using samtools. An example pileup file is shown below:\n22\t17094749\tA\t1\tc\tD\n22\t17202602\tT\t1\t.\tD\n22\t17411899\tA\t1\t.\tC\n22\t17450515\tG\t2\t.,\t9&lt;\n22\t17452966\tT\t1\tc\t5\n22\t17470779\tC\t1\t,\tA\n22\t17492203\tG\t1\t,\tB\n22\t17504945\tC\t3\t,..\tBCA\n22\t17529814\tT\t3\t..,\tCCC\n\nThe columns are chromosome, position (1-based), reference base, depth, bases and base qualities.\n8.4 BED fileBED file represents genomic regions and it follows UCSC conventions:\n1 752565 752566\n1 768447 768448\n1 1005805 1005806\n1 1018703 1018704\n1 1021414 1021415\n\nThe columns are: chromosome, start position (0-based) and end position (1-based).\n8.5 Coord fileCoord files represent the ancestries of both reference samples and sequence samples. An example coord file looks like below:\npopID  indivID  L1    Ci        t         PC1       PC2\nYRI    NA19238  1409  0.304122  0.98933   52.7634   -39.7924\nCEU    NA12892  1552  0.330037  0.989709  9.82674   25.2898\nCEU    NA12891  1609  0.362198  0.988082  0.439573  26.8872\nCEU    NA12878  1579  0.334825  0.988677  8.83775   28.1342\nYRI    NA19239  1558  0.34898   0.988302  53.9104   -39.1727\nYRI    NA19240  1735  0.404142  0.990264  59.8379   -45.2765\n\nThe columns are: popID means “population ID”, indivID means “individual ID”, L1 means number of loci has been covered, Ci means “average coverage”, t means Procrustes similarity. PC1, PC2 means coordinates of first and second principal components. You may notice L1, Ci, and t are omitted in the coord files of reference samples. The reason is that reference samples use genotypes and do not have coverage information.\n8.6 Site fileSite file is equivalent to BED file and it is used here to represent marker positions. An example site file looks like below:\nCHR  POS      ID          REF  ALT\n1    752566   rs3094315   G    A\n1    768448   rs12562034  G    A\n1    1005806  rs3934834   C    T\n1    1018704  rs9442372   A    G\n1    1021415  rs3737728   A    G\n\nThe site file has header line, and it contains chromosome, position(1-based), id (usually marker name), ref (reference allele) and alt (alternative allele).\n","dateCreated":"2020-06-24T21:25:22+08:00","dateModified":"2020-06-24T21:25:22+08:00","datePublished":"2020-06-24T21:25:22+08:00","description":"","headline":"Ancestry祖源分析|LASER","image":["1.png"],"mainEntityOfPage":{"@type":"WebPage","@id":"https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"},"publisher":{"@type":"Organization","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg","logo":{"@type":"ImageObject","url":"coveriuu.jpg"}},"url":"https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/","thumbnailUrl":"1.png"}</script>
    <meta property="og:type" content="blog">
<meta property="og:title" content="Ancestry祖源分析|LASER">
<meta property="og:url" content="https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/index.html">
<meta property="og:site_name" content="iuu&#39;s blog">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://iuu19.github.io/Ancestry%E7%A5%96%E6%BA%90%E5%88%86%E6%9E%90-LASER/LASER-workflow.png">
<meta property="og:image" content="https://iuu19.github.io/Ancestry%E7%A5%96%E6%BA%90%E5%88%86%E6%9E%90-LASER/HGDP_Popualtions.png">
<meta property="og:image" content="https://iuu19.github.io/Ancestry%E7%A5%96%E6%BA%90%E5%88%86%E6%9E%90-LASER/Results_on_HGDP.png">
<meta property="article:published_time" content="2020-06-24T13:25:22.000Z">
<meta property="article:modified_time" content="2020-06-24T13:25:22.000Z">
<meta property="article:author" content="Dr.iuu">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://iuu19.github.io/Ancestry%E7%A5%96%E6%BA%90%E5%88%86%E6%9E%90-LASER/LASER-workflow.png">
    
    
        
    
    
        <meta property="og:image" content="https://iuu19.github.io/assets/images/coveriuu.jpg"/>
    
    
        <meta property="og:image" content="https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/1.png"/>
        <meta class="swiftype" name="image" data-type="enum" content="https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/1.png"/>
    
    
    
    <!--STYLES-->
    
<link rel="stylesheet" href="/assets/css/style-gf8wcm29a1hig4pgasvkpmpwcvjyqc9ufzpbo404wwnc2qcamao8rlll1csq.min.css">

    <!--STYLES END-->
    

    

    
<link rel="stylesheet" href="/css/prism-vsc-dark-plus.css" type="text/css">
<link rel="stylesheet" href="/css/prism-line-numbers.css" type="text/css"></head>

    <body>
        <div id="blog">
            <!-- Define author's picture -->


    
        
            
        
    

<header id="header" data-behavior="4">
    <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
    <div class="header-title">
        <a
            class="header-title-link"
            href="/"
            aria-label=""
        >
            iuu&#39;s blog
        </a>
    </div>
    
        
            <a
                class="header-right-picture "
                href="/all-about"
                aria-label="打开链接: /all-about"
            >
        
        
            <img class="header-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
        </a>
    
</header>

            <!-- Define author's picture -->



        
    

<nav id="sidebar" data-behavior="4">
    <div class="sidebar-container">
        
            <div class="sidebar-profile">
                <a
                    href="/#about"
                    aria-label="阅读有关作者的更多信息"
                >
                    <img class="sidebar-profile-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
                </a>
                <h4 class="sidebar-profile-name">Dr.iuu</h4>
                
                    <h5 class="sidebar-profile-bio"><p>NCMC-SCMC-SWCH</p>
</h5>
                
            </div>
        
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://iuu19.github.io"
                            
                            rel="noopener"
                            title="首页"
                        >
                        <i class="sidebar-button-icon fa fa-home" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">首页</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-categories"
                            
                            rel="noopener"
                            title="分类"
                        >
                        <i class="sidebar-button-icon fa fa-bookmark" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">分类</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-tags"
                            
                            rel="noopener"
                            title="标签"
                        >
                        <i class="sidebar-button-icon fa fa-tags" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">标签</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-archives"
                            
                            rel="noopener"
                            title="归档"
                        >
                        <i class="sidebar-button-icon fa fa-archive" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">归档</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-about"
                            
                            rel="noopener"
                            title="关于"
                        >
                        <i class="sidebar-button-icon fa fa-question" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">关于</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://github.com/iuu19/iuu19.github.io"
                            
                                target="_blank"
                            
                            rel="noopener"
                            title="GitHub"
                        >
                        <i class="sidebar-button-icon fab fa-github" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">GitHub</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/atom.xml"
                            
                            rel="noopener"
                            title="RSS"
                        >
                        <i class="sidebar-button-icon fa fa-rss" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">RSS</span>
                    </a>
            </li>
            
        </ul>
        
    </div>
</nav>

            
            <div id="main" data-behavior="4"
                 class="
                        hasCoverMetaIn
                        ">
                
<article class="post">
    
    
        <div class="post-header main-content-wrap text-center">
    
        <h1 class="post-title">
            Ancestry祖源分析|LASER
        </h1>
    
    
        <div class="post-meta">
    <time datetime="2020-06-24T21:25:22+08:00">
	
		    6月 24, 2020
    	
    </time>
    
</div>

    
</div>

    
    <div class="post-content markdown">
        <div class="main-content-wrap">
            <font face="times new roman" size=4>

<span id="more"></span>

<p>Ancestry祖源分析|LASER</p>
<h1 id="table-of-contents">目录</h1><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-%E4%BB%8B%E7%BB%8D"><span class="toc-text">1. 介绍</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E4%B8%8B%E8%BD%BD"><span class="toc-text">2. 下载</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-LASER-workflow"><span class="toc-text">3. LASER workflow</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-HGDP-reference-panel"><span class="toc-text">4. HGDP reference panel</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-Preparing-input-files-for-LASER"><span class="toc-text">5. Preparing input files for LASER</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-1-liftOver-HGDP-hg19tohg38"><span class="toc-text">5.1 liftOver HGDP hg19tohg38</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-1-1-download-liftOver"><span class="toc-text">5.1.1 download liftOver</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-1-2-download-HGDP-reference-file"><span class="toc-text">5.1.2 download HGDP reference file</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-1-3-%E6%88%96%E8%80%85-vcf-%E2%80%93-gt-geno"><span class="toc-text">5.1.3 或者 vcf –&gt; geno</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-2-bam-%E2%80%93-gt-pileup"><span class="toc-text">5.2 : bam –&gt; pileup</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-3-pileup-%E2%80%93-gt-seq"><span class="toc-text">5.3 : pileup –&gt; seq</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#6-Estimating-ancestry-coordinates"><span class="toc-text">6. Estimating ancestry coordinates</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#6-1-Generate-the-reference-ancestry-space"><span class="toc-text">6.1 Generate the reference ancestry space</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#6-2-Estimate-ancestry-for-sequenced-samples"><span class="toc-text">6.2 Estimate ancestry for sequenced samples</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#6-3-Combine-results"><span class="toc-text">6.3 Combine results</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-Visualizing-results"><span class="toc-text">7. Visualizing results</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-File-format"><span class="toc-text">8. File format</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-1-Geno-file"><span class="toc-text">8.1 Geno file</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-2-Seq-file"><span class="toc-text">8.2 Seq file</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-3-Pileup-file"><span class="toc-text">8.3 Pileup file</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-4-BED-file"><span class="toc-text">8.4 BED file</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-5-Coord-file"><span class="toc-text">8.5 Coord file</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-6-Site-file"><span class="toc-text">8.6 Site file</span></a></li></ol>
<hr>
<h3 id="1-介绍"><a href="#1-介绍" class="headerlink" title="1. 介绍"></a>1. 介绍</h3><p>LASER (Locating Ancestry using SEquencing Reads)，是C++编译的软件包，可以直接从全基因组测序读取估算祖先，而无需调用基因型。该方法依赖于一组参考个体，这些参考个体的全基因组SNP基因型和祖先信息是已知的。我们首先通过将主成分分析（PCA）应用于参考个体的基因型数据来构建参考坐标系。</p>
<h3 id="2-下载"><a href="#2-下载" class="headerlink" title="2. 下载"></a>2. 下载</h3><p><a target="_blank" rel="noopener" href="http://csg.sph.umich.edu//chaolong/LASER/">LASER Package</a>: Version 2.03</p>
<pre><code>wget -c https://sph.umich.edu/csg/software_pages/softwaredownloads/LASER-2.03.tar.gz
tar xzvf LASER-2.03.tar.gz
mv LASER-2.03 laser

# version-2.04: http://csg.sph.umich.edu/chaolong/LASER/LASER-2.04.tar.gz
</code></pre>
<h3 id="3-LASER-workflow"><a href="#3-LASER-workflow" class="headerlink" title="3. LASER workflow"></a>3. LASER workflow</h3><p><img src="/Ancestry%E7%A5%96%E6%BA%90%E5%88%86%E6%9E%90-LASER/LASER-workflow.png" alt="LASER workflow"></p>
<h3 id="4-HGDP-reference-panel"><a href="#4-HGDP-reference-panel" class="headerlink" title="4. HGDP reference panel"></a>4. HGDP reference panel</h3><p><a target="_blank" rel="noopener" href="http://csg.sph.umich.edu/chaolong/LASER/HGDP-938-632958.tar.gz">HGDP data</a>: Include 632,958 autosomal SNPs for 938 unrelated individuals from the Human Genome Diversity Project (HGDP).</p>
<p><img src="/Ancestry%E7%A5%96%E6%BA%90%E5%88%86%E6%9E%90-LASER/HGDP_Popualtions.png" alt="Human Genome Diversity Project"></p>
<h3 id="5-Preparing-input-files-for-LASER"><a href="#5-Preparing-input-files-for-LASER" class="headerlink" title="5. Preparing input files for LASER"></a>5. Preparing input files for LASER</h3><h3 id="5-1-liftOver-HGDP-hg19tohg38"><a href="#5-1-liftOver-HGDP-hg19tohg38" class="headerlink" title="5.1 liftOver HGDP hg19tohg38"></a>5.1 liftOver HGDP hg19tohg38</h3><h3 id="5-1-1-download-liftOver"><a href="#5-1-1-download-liftOver" class="headerlink" title="5.1.1 download liftOver"></a>5.1.1 download liftOver</h3><p>liftOver file:<a target="_blank" rel="noopener" href="http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/">http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/</a></p>
<pre><code>wget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver
</code></pre>
<h3 id="5-1-2-download-HGDP-reference-file"><a href="#5-1-2-download-HGDP-reference-file" class="headerlink" title="5.1.2 download HGDP reference file"></a>5.1.2 download HGDP reference file</h3><pre><code>#wget -c http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/hg19ToHg38.over.chain.gz

LIFTOVER=/WORK/Project/youguoling/biosoft/liftover

$LIFTOVER/liftOver.txt $LASERF/HGDP/HGDP_938chr.site hg19ToHg38.over.chain.gz HGDP_938chrhg38.site unlifted.bed

$LIFTOVER/liftOver.txt $LASERF/HGDP/HGDP_938chr.bed hg19ToHg38.over.chain.gz HGDP_938chrhg38.bed unlifted.bed
</code></pre>
<h3 id="5-1-3-或者-vcf-–-gt-geno"><a href="#5-1-3-或者-vcf-–-gt-geno" class="headerlink" title="5.1.3 或者 vcf –&gt; geno"></a>5.1.3 或者 vcf –&gt; geno</h3><p>This step prepares the reference panel by converting a VCF genotype file to a GENO file. We will skip this step and use a ready-to-use HGDP reference panel. A typical command to run the vcf2geno tool is given in the file “$LASERF&#x2F;vcf2geno&#x2F;cmd.sh”:</p>
<pre><code>
# $LASERF/vcf2geno/vcf2geno --inVcf exampleVCF/example.vcf.gz --updateID test.updateId --out test
</code></pre>
<h3 id="5-2-bam-–-gt-pileup"><a href="#5-2-bam-–-gt-pileup" class="headerlink" title="5.2 : bam –&gt; pileup"></a>5.2 : bam –&gt; pileup</h3><p>This step uses samtools to generate pileup files from bam files.</p>
<pre><code>samtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101035.recal.bam &gt; 121101035.recal.pileup
samtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101043.recal.bam &gt; 121101043.recal.pileup 
samtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101050.recal.bam &gt; 121101050.recal.pileup
samtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101052.recal.bam &gt; 121101052.recal.pileup
samtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101415.recal.bam &gt; 121101415.recal.pileup
samtools mpileup -q 30 -Q 20 -f $REF -l $HGDP/HGDP_938.bed $BAM/121101861.recal.bam &gt; 121101861.recal.pileup
</code></pre>
<p>We use -q 30 and -Q 20 to exclude reads that have mapping quality score lower than 30 or base quality score lower than 20.</p>
<h3 id="5-3-pileup-–-gt-seq"><a href="#5-3-pileup-–-gt-seq" class="headerlink" title="5.3 : pileup –&gt; seq"></a>5.3 : pileup –&gt; seq</h3><p>这一步生成<code>hapmap_trios.seq</code>.</p>
<pre><code>python $LASER/pileup2seq/pileup2seq.py \
-m $HGDP/HGDP_938.site \
-b $BAM/AMD_roi_1-based.bed \
-i $BAM/AMD_hapmap_trios_id.txt \
-o hapmap_trios \
$BAM/121101035.recal.pileup \
$BAM/121101043.recal.pileup \
$BAM/121101050.recal.pileup \
$BAM/121101052.recal.pileup \
$BAM/121101415.recal.pileup \
$BAM/121101861.recal.pileup
</code></pre>
<p>-b provides the targeted regions to exclude and -i specifies alternative IDs for the BAM files to be used in the .seq file (including popID and indivID). -b and -i are optional.</p>
<h3 id="6-Estimating-ancestry-coordinates"><a href="#6-Estimating-ancestry-coordinates" class="headerlink" title="6. Estimating ancestry coordinates"></a>6. Estimating ancestry coordinates</h3><h3 id="6-1-Generate-the-reference-ancestry-space"><a href="#6-1-Generate-the-reference-ancestry-space" class="headerlink" title="6.1 Generate the reference ancestry space"></a>6.1 Generate the reference ancestry space</h3><p>LASER can perform principal components analysis (PCA) on genotype data of the reference panel to generate a reference ancestry space.</p>
<pre><code># $LASER/laser -g $HGDP/HGDP_938.geno -pca 1 -k 30 -o HGDP_938
</code></pre>
<p>The above command takes ~20 minutes to finish. We will skip this step, and use a set of reference ancestry coordinates that have been generated in the file <code>$HGDP/HGDP_938.RefPC</code>.coord.<br>View the reference coordinates:</p>
<pre><code>less -S $HGDP/HGDP_938.RefPC.coord
</code></pre>
<h3 id="6-2-Estimate-ancestry-for-sequenced-samples"><a href="#6-2-Estimate-ancestry-for-sequenced-samples" class="headerlink" title="6.2 Estimate ancestry for sequenced samples"></a>6.2 Estimate ancestry for sequenced samples</h3><p>Submit two jobs to place sequenced samples into the reference ancestry space:</p>
<pre><code>$LASER/laser -g $HGDP/HGDP_938.geno -c $HGDP/HGDP_938.RefPC.coord -s hapmap_trios.seq -K 20 -k 4 -x 1 -y 3 -o hapmap_trios.1-3 &amp;
$LASER/laser -g $HGDP/HGDP_938.geno -c $HGDP/HGDP_938.RefPC.coord -s hapmap_trios.seq -K 20 -k 4 -x 4 -y 6 -o hapmap_trios.4-6 &amp;
</code></pre>
<p>The first job will process samples 1 to 3 and the second job will processed samples 4 to 6. Each sequenced sample will be projected from a 20-dimensional PCA space onto a 4-dimensional reference ancestry space. The running time is ~10 minutes for processing 3 samples in each job.</p>
<h3 id="6-3-Combine-results"><a href="#6-3-Combine-results" class="headerlink" title="6.3 Combine results"></a>6.3 Combine results</h3><p>Results from previous step will be output to two files <code>hapmap_trios.1-3.SeqPC.coord</code> and <code>hapmap_trios.4-6.SeqPC.coord</code>. Here we simply concatenate the two files while skipping the header line of the second file.</p>
<pre><code>cp hapmap_trios.1-3.SeqPC.coord hapmap_trios.SeqPC.coord
more +2 hapmap_trios.4-6.SeqPC.coord &gt;&gt; hapmap_trios.SeqPC.coord
</code></pre>
<p><strong>View the results:</strong></p>
<pre><code>less -S hapmap_trios.SeqPC.coord
</code></pre>
<p><strong>The results should look like below (results will vary slightly):</strong> </p>
<pre><code>popID  indivID  L1      Ci         K     t          PC1        PC2         PC3         PC4
YRI    NA19238  78386   0.170864   20    0.999688   467.989    -210.294    -14.1729    -14.4204
CEU    NA12892  85486   0.185973   20    0.999723   10.796     199.095     -9.90387    -21.4534
CEU    NA12891  87588   0.190442   20    0.99973    2.04224    196.07      -19.5705    -12.8022
CEU    NA12878  83213   0.181748   20    0.999711   4.34591    199.861     -12.4825    -22.6281
YRI    NA19239  87564   0.193424   20    0.999734   474.464    -215.96     -9.02921    -19.7372
YRI    NA19240  95866   0.213874   20    0.999748   469.914    -214.94     -14.9923    -13.6559
</code></pre>
<h3 id="7-Visualizing-results"><a href="#7-Visualizing-results" class="headerlink" title="7. Visualizing results"></a>7. Visualizing results</h3><p>Example R codes are available in .&#x2F;LASER-2.01&#x2F;plot&#x2F;. Let’s copy the folder to current working directory:</p>
<pre><code>cp -r $LASER/plot/ ./
</code></pre>
<p>Go to the plot folder and run the script to plot results:</p>
<pre><code>cd plot
Rscript plotHGDP.r $HGDP/HGDP_938.RefPC.coord ../hapmap_trios.SeqPC.coord
</code></pre>
<p>A figure named <code>Results_on_HGDP.pdf</code> will be generated. Visualize the figure:</p>
<pre><code>evince Results_on_HGDP.pdf &amp;
</code></pre>
<p>We expect to see the following figure, in which 3 CEU samples cluster with HGDP Europeans and 3 YRI samples cluster with HGDP Africans:</p>
<p><img src="/Ancestry%E7%A5%96%E6%BA%90%E5%88%86%E6%9E%90-LASER/Results_on_HGDP.png" alt="Human Genome Diversity Project"></p>
<h3 id="8-File-format"><a href="#8-File-format" class="headerlink" title="8. File format"></a>8. File format</h3><h3 id="8-1-Geno-file"><a href="#8-1-Geno-file" class="headerlink" title="8.1 Geno file"></a>8.1 Geno file</h3><p>Geno file are from reference samples. LASER use genotype of these samples as a reference panel. You can obtain geno file from VCF files using vcf2geno.</p>
<p>An example geno file for the HGDP data set (resource&#x2F;HGDP&#x2F;HGDP_938.geno):</p>
<pre><code>Brahui	HGDP00001	1	2	1	1	0	2	0	2	1	2	2	2	1	1	2	1	0
Brahui	HGDP00003	0	0	2	0	0	2	0	2	0	2	2	2	2	0	2	2	0
Brahui	HGDP00005	0	2	2	0	0	1	0	2	1	2	2	2	2	1	2	2	1
Brahui	HGDP00007	0	2	2	0	0	2	0	2	0	2	2	2	1	1	2	2	1
Brahui	HGDP00009	0	1	0	1	0	2	0	2	0	2	2	2	2	0	2	2	0
Brahui	HGDP00011	1	1	2	1	1	2	1	1	1	2	2	2	1	1	2	2	0
Brahui	HGDP00013	1	2	2	1	1	2	1	2	0	2	2	2	2	0	2	2	0
Brahui	HGDP00015	1	1	2	0	0	2	0	2	0	2	2	2	2	0	2	2	0
Brahui	HGDP00017	1	1	2	0	0	1	0	0	0	2	0	1	1	2	2	2	0
Brahui	HGDP00019	0	2	2	0	0	1	0	1	0	2	1	2	2	1	2	2	0
</code></pre>
<p>The first and second columns represent the population id and individual id. From the third column, each number represents a genotype. To be consistent with the sequence data, genotypes should be given on the forward strand. Genotypes are coded by 0, 1, or 2, representing copies of the reference allele at a locus in one individual.</p>
<p>In this geno file, we have 632,960 columns which contains 632,958 markers from column 3 to the last column.</p>
<h3 id="8-2-Seq-file"><a href="#8-2-Seq-file" class="headerlink" title="8.2 Seq file"></a>8.2 Seq file</h3><p>Seq file is generated from pileup files. It contains sequencing information and organize it in a LASER readable format. The first two columns represent population id and individual id. Subsequent columns are total read depths and reference base counts. For example, column 3 and 4 are 0, 0 in the following example. That means at first marker, the sequence read depth is 0 and thus none of the reads has reference base. We enforce tab delimiters between markers and space delimiters between each read depths and reference base counts. On line of seq file looks like below:</p>
<pre><code>NA12878.chrom22	NA12878.chrom22	0 0	0 0	0 0	0 0	0 
</code></pre>
<h3 id="8-3-Pileup-file"><a href="#8-3-Pileup-file" class="headerlink" title="8.3 Pileup file"></a>8.3 Pileup file</h3><p>Pileup file are generated using samtools. An example pileup file is shown below:</p>
<pre><code>22	17094749	A	1	c	D
22	17202602	T	1	.	D
22	17411899	A	1	.	C
22	17450515	G	2	.,	9&lt;
22	17452966	T	1	c	5
22	17470779	C	1	,	A
22	17492203	G	1	,	B
22	17504945	C	3	,..	BCA
22	17529814	T	3	..,	CCC
</code></pre>
<p>The columns are chromosome, position (1-based), reference base, depth, bases and base qualities.</p>
<h3 id="8-4-BED-file"><a href="#8-4-BED-file" class="headerlink" title="8.4 BED file"></a>8.4 BED file</h3><p>BED file represents genomic regions and it follows UCSC conventions:</p>
<pre><code>1 752565 752566
1 768447 768448
1 1005805 1005806
1 1018703 1018704
1 1021414 1021415
</code></pre>
<p>The columns are: chromosome, start position (0-based) and end position (1-based).</p>
<h3 id="8-5-Coord-file"><a href="#8-5-Coord-file" class="headerlink" title="8.5 Coord file"></a>8.5 Coord file</h3><p>Coord files represent the ancestries of both reference samples and sequence samples. An example coord file looks like below:</p>
<pre><code>popID  indivID  L1    Ci        t         PC1       PC2
YRI    NA19238  1409  0.304122  0.98933   52.7634   -39.7924
CEU    NA12892  1552  0.330037  0.989709  9.82674   25.2898
CEU    NA12891  1609  0.362198  0.988082  0.439573  26.8872
CEU    NA12878  1579  0.334825  0.988677  8.83775   28.1342
YRI    NA19239  1558  0.34898   0.988302  53.9104   -39.1727
YRI    NA19240  1735  0.404142  0.990264  59.8379   -45.2765
</code></pre>
<p>The columns are: popID means “population ID”, indivID means “individual ID”, L1 means number of loci has been covered, Ci means “average coverage”, t means Procrustes similarity. PC1, PC2 means coordinates of first and second principal components. You may notice L1, Ci, and t are omitted in the coord files of reference samples. The reason is that reference samples use genotypes and do not have coverage information.</p>
<h3 id="8-6-Site-file"><a href="#8-6-Site-file" class="headerlink" title="8.6 Site file"></a>8.6 Site file</h3><p>Site file is equivalent to BED file and it is used here to represent marker positions. An example site file looks like below:</p>
<pre><code>CHR  POS      ID          REF  ALT
1    752566   rs3094315   G    A
1    768448   rs12562034  G    A
1    1005806  rs3934834   C    T
1    1018704  rs9442372   A    G
1    1021415  rs3737728   A    G
</code></pre>
<p>The site file has header line, and it contains chromosome, position(1-based), id (usually marker name), ref (reference allele) and alt (alternative allele).</p>

            


        </div>
    </div>
    <div id="post-footer" class="post-footer main-content-wrap">
        
        
            <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    <a
                        class="post-action-btn btn btn--disabled"
                        aria-hidden="true"
                    >
                        
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-23-ancestry-zu-yuan-fen-xi-peddy/"
                    data-tooltip="Ancestry祖源分析|peddy"
                    aria-label="下一篇: Ancestry祖源分析|peddy"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


        
        
    </div>
</article>



                <footer id="footer" class="main-content-wrap">
    <span class="copyrights">
        Copyrights &copy; 2022 Dr.iuu. All Rights Reserved.
    </span>
</footer>

            </div>
            
                <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
                    <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    <a
                        class="post-action-btn btn btn--disabled"
                        aria-hidden="true"
                    >
                        
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-23-ancestry-zu-yuan-fen-xi-peddy/"
                    data-tooltip="Ancestry祖源分析|peddy"
                    aria-label="下一篇: Ancestry祖源分析|peddy"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


                </div>
                
    <div id="share-options-bar" class="share-options-bar" data-behavior="4">
        <i id="btn-close-shareoptions" class="fa fa-times"></i>
        <ul class="share-options">
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                        aria-label="分享到 Facebook"
                    >
                        <i class="fab fa-facebook" aria-hidden="true"></i><span>分享到 Facebook</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                        aria-label="分享到 Twitter"
                    >
                        <i class="fab fa-twitter" aria-hidden="true"></i><span>分享到 Twitter</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-24-ancestry-zu-yuan-fen-xi-laser/"
                        aria-label="分享到 Google+"
                    >
                        <i class="fab fa-google-plus" aria-hidden="true"></i><span>分享到 Google+</span>
                    </a>
                </li>
            
        </ul>
    </div>


            
        </div>
        


    
        
    

<div id="about">
    <div id="about-card">
        <div id="about-btn-close">
            <i class="fa fa-times"></i>
        </div>
        
            <img id="about-card-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
            <h4 id="about-card-name">Dr.iuu</h4>
        
            <div id="about-card-bio"><p>NCMC-SCMC-SWCH</p>
</div>
        
        
            <div id="about-card-job">
                <i class="fa fa-briefcase"></i>
                <br/>
                <p>Bioinfo-CHD</p>

            </div>
        
        
            <div id="about-card-location">
                <i class="fa fa-map-marker-alt"></i>
                <br/>
                Pudong, Shanghai
            </div>
        
    </div>
</div>

        
        
<div id="cover" style="background-image:url('/assets/images/cover2.jpg');"></div>
        <!--SCRIPTS-->

<script src="/assets/js/script-3nlvflbbvekvxx0g6h2clfqhhxs61cvvbofv7nmctigkdbbhu5htlvmrqg80.min.js"></script>

<!--SCRIPTS END-->





    </body>
</html>
