
<!DOCTYPE html>
<html lang="zh-cn">
    
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="iuu&#39;s blog">
    <title>GWAS|Rvtests安装及使用教程 - iuu&#39;s blog</title>
    <meta name="author" content="Dr.iuu">
    
    
        <link rel="icon" href="https://iuu19.github.io/assets/images/coveriuu.jpg">
    
    
        
            <link rel="alternate" type="application/atom+xml" title="RSS" href="/atom.xml">
        
    
    <script type="application/ld+json">{"@context":"http://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg"},"articleBody":"\n\n\n\nGWAS|Rvtests安装及使用教程\n\n\n1.介绍Rvtests，全称（Rare Variant tests，稀有变异），rvtests已开发为支持遗传关联分析和Meta分析的综合工具。它可以分析unrelated individual和family-based individuals的quantitative 和binary结果。它包括各种关联分析（例如，single variant score test, burden test, variable threshold test, SKAT test, fast linear mixed model score test）。它采用 VCF&#x2F;BGEN&#x2F;PLINK作为基因型输入文件，并采用PLINK格式表型文件和协变量文件。\n此软件包能够在计算机工作站上的线性混合模型中分析多达1,000,000个人的数据集。RVTESTS支持single variant and gene-level tests。它还允许以RAREMETAL格式在得分统计之间高效生成协方差矩阵，可用于支持下一波整合了大型生物库数据集的Meta分析。\n可以使用线性回归或逻辑回归模型处理更大的样本量。\n2.引用RVTESTS: An Efficient and Comprehensive Tool for Rare Variant Association Analysis Using Sequence Data. Bioinformatics, 2016, 32:1423-1426.\n3.下载和安装源代码可以从github或github页面下载最新版本。\ngit clone https://github.com/zhanxw/rvtests\ncd retests\nmake\n\n编译成功后，可执行文件将位于executable文件夹下。只需键入executable&#x2F;rvtests即可开始。\n或者，可以从此处下载二进制可执行文件。\n4.快速教程4.1 Single variant testsrvtest --inVcf input.vcf --pheno phenotype.ped --out output --single wald,score\n\nphenotype.ped中推荐controls as 1, cases as 2, missing phenotypes as -9 or 0.\n对于其他类型的关联分析，可以参考如下Models：\n\n4.2 Groupwise testsGroupwise tests主要包括三种类型的tests：\n\nBurden tests：用于group variants，通常少于1％或5％稀有变异关联分析，包括：CMC test, Zeggini test, Madsen-Browning test, CMAT test和rare-cover test\n\nVariable threshold tests：group variants under different frequency thresholds.\n\nKernel methods:包括SKAT test和KBAC test\n\n\n如果按基因进行稀有变异分析，需要使用--geneFilerefFlat格式指定参考基因。也可以--gene用来指定要测试的基因。例如，指定--gene CFH,ARMS2将对CFH和ARMS2基因执行关联分析。\nrvtest --inVcf input.vcf --pheno phenotype.ped --out output --geneFile refFlat_hg19.txt.gz --burden cmc --vt price --kernel skat,kbac\n\n4.3 Related individual tests首先需要创建一个亲属关系矩阵：\nvcf2kinship --inVcf input.vcf --bn --out output\n\n该选项--bn意味着使用Balding-Nicols方法计算经验亲属关系。您可以指定--ibs获取IBS亲属关系或者--pedigree input.ped用于根据已知的家系信息计算血统。\n然后，您可以使用基于线性混合模型的关联分析，例如Fast-LMM score test，Fast-LMM LRT test和Grammar-gamma tests：\nrvtest --inVcf input.vcf --pheno phenotype.ped --out output --kinship output.kinship --single famScore,famLRT,famGrammarGamma\n\n4.4 Meta-analysis testsThe meta-analysis models outputs association test results and genotype covariance matrix. These calculated summary statistics can be used in rare variant association analysis (details). We provide single variant score test and generate a genotype covariance matrix. You can use this command:\nrvtest --inVcf input.vcf --pheno phenotype.ped --meta score,cov --out output\n\nIn a more realistic scenario, you may want to adjust for covariates and want to inverse normalized residuals obtained in null model (link to our methodology paper), then this command will work:\nrvtest --inVcf input.vcf --pheno phenotype.ped --covar example.covar --covar-name age,bmi --inverseNormal --useResidualAsPhenotype  --meta score,cov --out output\n\n--covar指定协变量文件，--covar-name指定在分析中使用的协变量。协变量文件格式可以在这里找到。--inverseNormal --useResidualAsPhenotype指定trait transformation method。这意味着首先在协变量上拟合表型的回归模型（自动添加截距），然后对残差进行逆归一化。trait transformation的详细信息可以在这里找到。\nWe support both unrelated individuals and related individuals (e.g. family data)。需要--kinship input.kinship：\nrvtest --inVcf input.vcf --pheno phenotype.ped --meta score,cov --out output --kinship input.kinship\n\n其中input.kinship是由vcf2kinship计算。\n注意：默认情况下，协方差矩阵是在一百万个基对的滑动窗口中计算的。您可以通过选项更改此设置windowSize。例如，--meta cov[windowSize=500000]指定一个500k-bp的滑动窗口。\n\nThe above models are suitable to generate summary statistics which can be later meta-analyzed (see Dajiang Liu (2014) Nature Genetics). Rvtests implemented the above methods and the results can be further analyzed by RareMetals (link) for quantitative trait and RareMetals2 (link). It also worth to mention that our group offers another toolset for meta analysis (link).\n4.4.1 Explanation of outputsThe meta-analysis results come as two files: summary score statistics files (prefix.MetaScore.assoc.gz) and covariance files (prefix.MetaCov.assoc.gz).\nIn summary score statistics files, you will obtain these columns:\n\nN_INFORMATIVE: Number of samples that are analyzed for association.\nAF: allele frequency. For related individuals, we use BLUE estimator. For case-control study, we list overall frequency (adjusted by relatedness if possible), case frequency and control frequency separated by a colon.\nINFORMATIVE_ALT_AC: The number of alternative alleles in the analyzed samples.\nCALL_RATE: The fraction of non-missing alleles. For case-control study, we calculate call rate for all samples, case samples and control samples separated by a colon.\nHWE_PVALUE: Hardy-Weinberg equilibrium. For related individuals, this statistic can be inflated. For case-control study, we calculate HWE pvalues for all samples, case samples and controls samples separated by a colon.\nN_REF&#x2F;N_HET&#x2F;N_ALT: Number of samples carrying homozygous reference&#x2F;heterozygous&#x2F;homozygous alternative alleles. For case-control study, we calculate these three statistics for all samples, case samples and controls samples separated by a colon.\nU_STAT, SQRT_V_STAT: U and V statistics are score statistics and their covariance matrix. Details can be found in Dajiang Liu (2014) Nature Genetics.\nALT_EFFSIZE: for continuous outcome, this is the estimated effect size; for binary outcome, this is the estimated log odds-ratio. We apply a new correction method when binary trait associations for related individuals are analyzed in standard linear mixed models. The log odds ratio is approximately correct for related individual analysis as well.\n\nThe covariance files stores covariances in sliding windows. You will obtain these columns:\n\nCHROM: the chromosome name\nSTART_POS&#x2F;END_POS: within one sliding window, the first&#x2F;last variant chromosomal position.\nNUM_MARKER: number of markers in one sliding window.\nMARKER_POS: all chromosomal positions in the sliding window.\nCOV: covariances between markers. When the sliding window has markers G.1, G.2, …, G.N, this column will store “covariances” of (G.1, G.1), (G.1, G.2), …, (G.1, G.N); for binary outcomes, this column uses colons as separators and additionally stores covariances between N genotypes and K covariates (G.1, C.1), (G.1, C.2), …, (G.1, C.K), as well as covariances between K covariates (C.1, C.1), (C.1, C.2), … (C.1, C.K), (C.2, C.2), …, (C.2, C.K), …, (C.K, C.K).\n\n4.4.2 Dominant models and recessive models通过在 –meta选项之后附加dominant和&#x2F;或recessive，可以支持显性和隐性疾病模型。例如，使用–meta dominant,recessive将生成两组文件。对于dominant模型，分别是prefix.MetaDominant.assoc和prefix.MetaDominantCov.assoc.gz；对于recessive模型，分别是prefix.MetaRecessive.assoc和prefix.MetaRecessiveCov.assoc.gz。在dominant模型中，基因型0&#x2F;1&#x2F;2被编码为0&#x2F;1&#x2F;1；在recessive模型中，基因型0&#x2F;1&#x2F;2被编码为0&#x2F;0&#x2F;1。基因型缺失将被推算为均值。\n5.输入文件5.1 Genotype files（VCF，BCF，BGEN，KGG）Rvtests支持VCF（变体调用格式）文件。支持未压缩格式或压缩格式的vcf文件。To use group-based rare variant tests，需要使用tabix为VCF文件建立索引。\n以下是将未压缩VCF格式转换为bgzipped VCF格式的命令：\n(grep ^&quot;#&quot; $your_old_vcf; grep -v ^&quot;#&quot; $your_old_vcf | sed &#39;s:^chr::ig&#39; | sort -k1,1n -k2,2n) | bgzip -c &gt; $your_vcf_file \ntabix -f -p vcf $your_vcf_file\n\n上述命令将（1）chr从染色体名称中删除前缀；（2）首先按染色体，然后按染色体位置对VCF文件进行排序；（3）使用bgzip压缩；（4）创建tabix索引。\nRvtests支持genotype dosages。使用--dosage DosageTag指定dosage tag。例如，如果VCF格式字段为GT：EC，单个基因型字段为0/0：0.02，则可以使用--dosage EC，并且rvtests将在回归模型中使用剂量0.02。\nRvtests支持BGEN输入格式 v1.0到v1.3。使用--inBgen指定一个BGEN文件,--inBgenSample指定SAMPLE文件。\nRvtests支持KGGSeq输入格式。此格式是对二进制PLINK格式的扩展。使用–inKgg来代替–inVcf。\n5.2 Phenotype files可以使用--mpheno $phenotypeColumnNumber或--pheno-name指定特定表型。\n表型文件示例（example.pheno）具有以下格式：\nfid iid fatid matid sex y1 y2 y3 y4\nP1 P1 0 0 0 1.7642934435605 -0.733862638327895 -0.980843608339726 2\nP2 P2 0 0 0 0.457111744989746 0.623297281416372 -2.24266162284447 1\nP3 P3 0 0 0 0.566689682543218 1.44136462889459 -1.6490100777089 1\nP4 P4 0 0 0 0.350528353203767 -1.79533911725537 -1.11916876241804 1\nP5 P5 0 0 1 2.72675074738545 -1.05487747371158 -0.33586430010589 2\n\n表型文件由--pheno example.pheno 指定。默认表型列标题为y1。如果要使用替代列作为关联分析的表型（例如标题为y2的列），则可以使用列或名称来指定表型\n–mpheno 2\n–pheno-name y2\n\n注意：要使用--pheno-name，标题行必须以fid iid开头。\n在表型文件中，缺失值可以用NA或任何非数字值表示。缺少表型的个体将自动从后续关联分析中删除。对于每个缺失的表型值，将生成一条警告并将其记录在日志文件中。\n当表型值分别为0、1和2时，rvtests会自动将其视为binary traits。但是，如果要将其视为连续性状，需使用--qtl选项。\n5.3 Covariate files可以使用--covar和--covar-name指定将用于单变量关联分析的协变量。这是一个可选参数。如果数据中没有协变量，则可以忽略此选项。\n协变量文件（例如example.covar）与表型文件具有相似的格式：\nfid iid fatid matid sex y1 y2 y3 y4\nP1 P1 0 0 0 1.911 -1.465 -0.817 1\nP2 P2 0 0 0 2.146 -2.451 -0.178 2\nP3 P3 0 0 0 1.086 -1.194 -0.899 1\nP4 P4 0 0 0 0.704 -1.052 -0.237 1\nP5 P5 0 0 1 2.512 -3.085 -2.579 1\n\n协变量文件由--covar选项指定（例如--covar example.covar）。要指定将在关联分析中使用的协变量，可以使用该选项--covar-name。例如，当使用age，bmi和3个PCAs进行关联分析时，可以为rvtests程序指定以下选项，即 --covar example.covar --covar-name age,bmi,pc1,pc2,pc3。\n注意：协变量文件中缺少的数据可以用任何非数字值（例如NA）标记。它们将自动推算为数据文件中的平均值。\n5.4 Trait transformationIn this meta-analysis, we use inverse normal transformed residuals in the association analysis, which is achieved by using a combination of --inverseNormal and --useResidualAsPhenotype. Specifically, we first fit the null model by regressing phenotype on covariates. The residuals are then inverse normal transformed (see Appendix A more detailed formula for transformation). Transformed residuals will be used to obtain score statistics.\n./rvtest --inVcf $vcf --pheno $example.pheno --covar example.covar --covar-name age,bmi --inverseNormal --useResidualAsPhenotype  --meta score,cov --out $output_prefix\n\n6. Models6.1 Utility modelsRvtests has an convenient option –outputRaw. When specifying this, rvtests can output genotypes, phenotype, covariates (if any) and collapsed genotype to tabular files. These files can be imported into other software (e.g. R) for further analyses.\n7. Association test options7.1 Sample inclusion&#x2F;exclusionRvtests可以灵活地指定要包括或排除的样本：\n--peopleIncludeID : give IDs of people that will be included in study\n--peopleIncludeFile : from given file, set IDs of people that will be included in study\n--peopleExcludeID : give IDs of people that will be included in study\n--peopleExcludeFile : from given file, set IDs of people that will be included in study\n\n7.2 Variant site filters通常在稀有变异分析中使用不同的frequency cut-off。因此，rvtests指定frequency cut-offs。Frequency Cut-off\n--freqUpper: Specify upper minor allele frequency bound to be included in analysis\n--freqLower: Specify lower minor allele frequency bound to be included in analysis\n\n如果指定--freqLower 0.01 --freqUpper 0.05，则仅分析次要等位基因频率在0.01到0.05（含边界）之间的变体。\n与Sample inclusion&#x2F;exclusion类似，可以通过--rangeList选项来指定要包含的变异范围。例如·–rangeList 1:100-200·将1号染色体的位置包括100bp至200bp的区域。或者，使用单独的文件，range.txt --rangeFile range.txt指定关联分析范围。\n--rangeList: Specify some ranges to use, please use chr:begin-end format.\n--rangeFile: Specify the file containing ranges, please use chr:begin-end format.\n--siteFile: Specify the file containing sites to include, please use &quot;chr pos&quot; format.\n\nIt is supported to filter variant site by site depth, minor allele count or annotation (annotated VCF file is needed).\n--siteDepthMin : Specify minimum depth(inclusive) to be included in analysis\n--siteDepthMax : Specify maximum depth(inclusive) to be included in analysis\n--siteMACMin : Specify minimum Minor Allele Count(inclusive) to be included in analysis\n--annoType : Specify annotation type that is followed by ANNO= in the VCF INFO field, regular expression is allowed\n\n注意：--annoType可以基于正则表达式过滤变异。例如，--annoType Nonsynonymous将仅分析ANNO&#x3D;Nonsynonymous在INFO字段中存在的非同义变异。要提取一种以上的注释类型，请使用--annoType &#39;Start_Gain|Stop_Loss|Start_Loss|Essential_Splice_Site|Stop_Gain|Normal_Splice_Site|Synonymous|Nonsynonymous提取LOF（功能丧失）突变。要生成带注释的VCF文件，请阅读下一节Annotation。\n7.2.1 Annotation在此，我们将带有注释信息的VCF文件定义为带注释的VCF。可以使用ANNO（一种快速准确的注释软件）来完成注释步骤。最少有两个步骤来注释VCF文件：\n\n安装ANNO及其资源文件\n\ngit clone https://github.com/zhanxw/anno.git; cd anno; 使; cd资源；./download.sh\n\n\n运行以下脚本：\n\nanno -i input.vcf -o output.vcf.gz -r anno / resources / hs37d5.fa -g anno / resources / refFlat_hg19.txt.gz -p anno / priority.txt -c anno / codon.txt –indexOutput\n\n然后，您将获得带注释的VCF文件output.vcf.gz及其tabix索引output.vcf.gz.tbi。\n7.3 Genotype filters可以通过以下选项过滤掉深度较浅或质量较差的基因型：\n-indvDepthMin : Specify minimum depth(inclusive) of a sample to be included in analysis\n--indvDepthMax : Specify maximum depth(inclusive) of a sample to be included in analysis\n--indvQualMin : Specify minimum depth(inclusive) of a sample to be included in analysis\n\n7.4 处理缺失的基因型和表型When genotypes are missing (e.g. genotype &#x3D; “.&#x2F;.”) or genotypes are filtered out, there are three options to handle them: \n\n(1) impute to its mean(default option); \n\n(2) impute by HWE equilibrium; \n\n(3) remove from the model. \nUse –impute [mean|hwe|drop] to specify which option to use.\n\n\n例如，当缺少定量表型时，某些样本具有基因型文件，但没有表型，rvtest可以将缺失表型推算为其平均值。\n注意：请勿将--imputePheno用于binary trait。\n总之，可以使用以下两个选项：\n--impute : Specify either of mean, hwe, and drop\n--imputePheno : Impute phenotype to mean by those have genotypes but no phenotypes\n\n7.5 Specify groups (e.g burden unit)使用--geneFile选项指定基因文件名。例如，--geneFile refFlat_hg19.txt.gz将refFlat_hg19.txt.gz用作基因定义文件，然后对每个基因执行关联分析。使用--gene指定的基因的一个子集的测试。例如，--gene CFH将仅测试CFH基因。\nAlternative grouping unit can be specified as set. These sets are treated similar to gene. You can thus use --setFile to define sets (similar to --geneFile option), and use --set to define a specific set (similar to --gene option). Additionally, use --setList can specify a set to test from command line.\nThe format of a set file is: (1) set names; (2) ranges (e.g. chrom:begin-end); For example, you have a set file, example.set, like this:\nset1 1:100-200,1:250-300\nset2 2:500-600\n\n您可以指定--setFile example.set --set set2将2号染色体内500至600bp的变体分组。如果要测试特定区域，例如2号染色体，位置500到550bp，但又不想制作另一个文件，则可以使用--setList 2:500-600。\n概括来说，与分组单位有关的选项如下：\n--geneFile : specify a gene file (for burden tests)\n--gene : specify which genes to test\n--setList : specify a list to test (for burden tests)\n--setFile : specify a list file (for burden tests, first two columns:setName chr:beg-end)\n--set : specify which set to test (1st column)\n\n8.Sex chromosome analysisRvtests support X chromosome analysis. In human X chromosome, there is PAR (pseudoautosomal region) and non-PAR region. For males, there are two X allele in PAR region and one allele in non-PAR region. While the PAR region is treated in the same way as autosomes, rvtests treat non-PAR region differently. Below we will describe the details about how rvtests handles non-PAR region.\nPrepare data. According to VCF standard, male genotype needs to coded as 0 or 1. For compatibility, rvtests also support 0&#x2F;0 or 1&#x2F;1 coding. In VCF files, male genotypes can be written as “0”, “1”, “0|0”, “0/0”, “1|1”, “1/1”. All other genotypes will be treated as missing.\nGenotype in regression models. For consistence, male genotypes are converted to 0 or 2. When male dosages are provided, we expect the values in the VCF file are between 0.0 and 1.0, and then will model them as twice the value.\nMetaScore results. If –meta score is specified, the output file prefix.MetaScore.assoc.gz includes both PAR-region and non-PAR region analysis. However, in the non-PAR region, the difference is that Hardy-Weinberg P-value and homozygous-reference&#x2F;heterzygous&#x2F;homozygous-alternative sample sizes are calculated using female samples only.\nRelated individuals. Just append –xHemi to the vcf2kinship (more details in Kinship generation) and rvtest command lines. Rvtests can recognize non-PAR region kinship file and use it in the analysis.\nPAR region. PAR region is defined as two regions X:60001-2699520 and X:154931044-155260560. Use --xLabel can specify which chromosome has PAR region (default: 23 or X) and use --xParRegion to specify PAR region (default: hg19, meaning ‘60001-2699520,154931044-155260560’ in the UCSC build hg19, specify “hg18” will use PAR region definition in the UCSC build hg18, or specify “hg38” will use UCSC build 38).\n9.Kinship generation分析相关个人通常需要估计亲属关系。可以使用vcf2kinship。\nvcf2kinship can calculate pedigree kinship using a pedigree input file (PED format, see Phenotype file, use option --ped). The output file name is specified by --prefix option. If you use --prefix output then the output files will include output.kinship.\nIt can also calculate empirical kinship using genotype input file (VCF format, see Genotype file (VCF), use option --inVcf). For empirical kinship, you also need to specify the kinship model, either Balding-Nicols model (use option --bn) or Identity-by-state model (use option --ibs).\nIn sex chromosome analysis, it is often required to generate kinship on X chromosome regions, then you need to specify --xHemi. If your input VCF file has different X chromosome label (e.g. chromosome name is ‘23’ instead of ‘X’, you can use --xLabel 23.\nIf principal component decomposition (PCA) results are needed, you can use option --pca. Then output files with suffix ‘.pca’ include PCA results.\nWhen dealing with large input files, it is often preferred to use multiple CPU to speed up calculation using the option --thread N in which N is the number of CPU.\nFor example, to generate pedigree-based kinship (--ped) on both autosomal region and X chromosome (--xHemi) region, the command line is:\nvcf2kinship --ped input.ped --xHemi --out output\n\nTo generate empirical kinship (--inVcf) on both autosomal region and X chromosome (--xHemi) region using Balding-Nicols model, the command line is:\nvcf2kinship --inVcf input.vcf.gz --ped input.ped --bn --xHemi --out output\n\nNOTE: you need to provide a pedigree file (PED) in the above case, as vcf2kinship needs the sex information of samples to construct kinship for sex chromosome.\nFor modern genetic datasets, genotype data is often stored by chromosomes, with each chromosome stored in a separate VCF file. In this case, kinship matrix can be first calcualted separately for each chromosome, and then combined using the python script combineKinship.py. Specifically, if you generated two kinship matrices for chr1 chr2 (chr1.kinship, and chr2.kinship), you can run the python script to combine them, i.e.\npython combineKinship.py -o prefix chr1.kinship chr2.kinship\n\n所得的亲属关系矩阵等效于使用合并的vcf文件计算的亲属关系矩阵。\n10.Resources10.1 UCSC RefFlat GenesrefFlat_hg19.txt.gz\n\nUCSC基因组版本中refFlat格式的UCSC基因定义文件（详细信息）。\nFile link：http : &#x2F;&#x2F;qbrc.swmed.edu&#x2F;zhanxw&#x2F;seqminer&#x2F;data&#x2F;refFlat_hg19.txt.gz\n10.2 Gencode GenesrefFlat.gencode.v19.gz\n\nGencode基因定义版本19，采用refFlat格式（详细信息）。\nFile link：http : &#x2F;&#x2F;qbrc.swmed.edu&#x2F;zhanxw&#x2F;seqminer&#x2F;data&#x2F;refFlat.gencode.v19.gz\n","dateCreated":"2020-06-20T22:48:21+08:00","dateModified":"2020-06-20T22:48:21+08:00","datePublished":"2020-06-20T22:48:21+08:00","description":"","headline":"GWAS|Rvtests安装及使用教程","image":["1.png"],"mainEntityOfPage":{"@type":"WebPage","@id":"https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"},"publisher":{"@type":"Organization","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg","logo":{"@type":"ImageObject","url":"coveriuu.jpg"}},"url":"https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/","thumbnailUrl":"1.png"}</script>
    <meta property="og:type" content="blog">
<meta property="og:title" content="GWAS|Rvtests安装及使用教程">
<meta property="og:url" content="https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/index.html">
<meta property="og:site_name" content="iuu&#39;s blog">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://iuu19.github.io/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/single.jpg">
<meta property="og:image" content="https://iuu19.github.io/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/burden.jpg">
<meta property="og:image" content="https://iuu19.github.io/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/variablethreshold.jpg">
<meta property="og:image" content="https://iuu19.github.io/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/kernel.jpg">
<meta property="og:image" content="https://iuu19.github.io/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/meta.jpg">
<meta property="article:published_time" content="2020-06-20T14:48:21.000Z">
<meta property="article:modified_time" content="2020-06-20T14:48:21.000Z">
<meta property="article:author" content="Dr.iuu">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://iuu19.github.io/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/single.jpg">
    
    
        
    
    
        <meta property="og:image" content="https://iuu19.github.io/assets/images/coveriuu.jpg"/>
    
    
        <meta property="og:image" content="https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/1.png"/>
        <meta class="swiftype" name="image" data-type="enum" content="https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/1.png"/>
    
    
    
    <!--STYLES-->
    
<link rel="stylesheet" href="/assets/css/style-gf8wcm29a1hig4pgasvkpmpwcvjyqc9ufzpbo404wwnc2qcamao8rlll1csq.min.css">

    <!--STYLES END-->
    

    

    
<link rel="stylesheet" href="/css/prism-vsc-dark-plus.css" type="text/css">
<link rel="stylesheet" href="/css/prism-line-numbers.css" type="text/css"></head>

    <body>
        <div id="blog">
            <!-- Define author's picture -->


    
        
            
        
    

<header id="header" data-behavior="4">
    <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
    <div class="header-title">
        <a
            class="header-title-link"
            href="/"
            aria-label=""
        >
            iuu&#39;s blog
        </a>
    </div>
    
        
            <a
                class="header-right-picture "
                href="/all-about"
                aria-label="打开链接: /all-about"
            >
        
        
            <img class="header-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
        </a>
    
</header>

            <!-- Define author's picture -->



        
    

<nav id="sidebar" data-behavior="4">
    <div class="sidebar-container">
        
            <div class="sidebar-profile">
                <a
                    href="/#about"
                    aria-label="阅读有关作者的更多信息"
                >
                    <img class="sidebar-profile-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
                </a>
                <h4 class="sidebar-profile-name">Dr.iuu</h4>
                
                    <h5 class="sidebar-profile-bio"><p>NCMC-SCMC-SWCH</p>
</h5>
                
            </div>
        
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://iuu19.github.io"
                            
                            rel="noopener"
                            title="首页"
                        >
                        <i class="sidebar-button-icon fa fa-home" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">首页</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-categories"
                            
                            rel="noopener"
                            title="分类"
                        >
                        <i class="sidebar-button-icon fa fa-bookmark" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">分类</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-tags"
                            
                            rel="noopener"
                            title="标签"
                        >
                        <i class="sidebar-button-icon fa fa-tags" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">标签</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-archives"
                            
                            rel="noopener"
                            title="归档"
                        >
                        <i class="sidebar-button-icon fa fa-archive" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">归档</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-about"
                            
                            rel="noopener"
                            title="关于"
                        >
                        <i class="sidebar-button-icon fa fa-question" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">关于</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://github.com/iuu19/iuu19.github.io"
                            
                                target="_blank"
                            
                            rel="noopener"
                            title="GitHub"
                        >
                        <i class="sidebar-button-icon fab fa-github" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">GitHub</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/atom.xml"
                            
                            rel="noopener"
                            title="RSS"
                        >
                        <i class="sidebar-button-icon fa fa-rss" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">RSS</span>
                    </a>
            </li>
            
        </ul>
        
    </div>
</nav>

            
            <div id="main" data-behavior="4"
                 class="
                        hasCoverMetaIn
                        ">
                
<article class="post">
    
    
        <div class="post-header main-content-wrap text-center">
    
        <h1 class="post-title">
            GWAS|Rvtests安装及使用教程
        </h1>
    
    
        <div class="post-meta">
    <time datetime="2020-06-20T22:48:21+08:00">
	
		    6月 20, 2020
    	
    </time>
    
</div>

    
</div>

    
    <div class="post-content markdown">
        <div class="main-content-wrap">
            <font face="times new roman" size=4>

<span id="more"></span>

<p>GWAS|Rvtests安装及使用教程</p>
<h1 id="table-of-contents">目录</h1><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-%E4%BB%8B%E7%BB%8D"><span class="toc-text">1.介绍</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E5%BC%95%E7%94%A8"><span class="toc-text">2.引用</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-%E4%B8%8B%E8%BD%BD%E5%92%8C%E5%AE%89%E8%A3%85"><span class="toc-text">3.下载和安装</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-%E5%BF%AB%E9%80%9F%E6%95%99%E7%A8%8B"><span class="toc-text">4.快速教程</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-1-Single-variant-tests"><span class="toc-text">4.1 Single variant tests</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-2-Groupwise-tests"><span class="toc-text">4.2 Groupwise tests</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-3-Related-individual-tests"><span class="toc-text">4.3 Related individual tests</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-4-Meta-analysis-tests"><span class="toc-text">4.4 Meta-analysis tests</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-4-1-Explanation-of-outputs"><span class="toc-text">4.4.1 Explanation of outputs</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-4-2-Dominant-models-and-recessive-models"><span class="toc-text">4.4.2 Dominant models and recessive models</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6"><span class="toc-text">5.输入文件</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-1-Genotype-files%EF%BC%88VCF%EF%BC%8CBCF%EF%BC%8CBGEN%EF%BC%8CKGG%EF%BC%89"><span class="toc-text">5.1 Genotype files（VCF，BCF，BGEN，KGG）</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-2-Phenotype-files"><span class="toc-text">5.2 Phenotype files</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-3-Covariate-files"><span class="toc-text">5.3 Covariate files</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-4-Trait-transformation"><span class="toc-text">5.4 Trait transformation</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#6-Models"><span class="toc-text">6. Models</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#6-1-Utility-models"><span class="toc-text">6.1 Utility models</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-Association-test-options"><span class="toc-text">7. Association test options</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-1-Sample-inclusion-x2F-exclusion"><span class="toc-text">7.1 Sample inclusion&#x2F;exclusion</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-2-Variant-site-filters"><span class="toc-text">7.2 Variant site filters</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-2-1-Annotation"><span class="toc-text">7.2.1 Annotation</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-3-Genotype-filters"><span class="toc-text">7.3 Genotype filters</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-4-%E5%A4%84%E7%90%86%E7%BC%BA%E5%A4%B1%E7%9A%84%E5%9F%BA%E5%9B%A0%E5%9E%8B%E5%92%8C%E8%A1%A8%E5%9E%8B"><span class="toc-text">7.4 处理缺失的基因型和表型</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#7-5-Specify-groups-e-g-burden-unit"><span class="toc-text">7.5 Specify groups (e.g burden unit)</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#8-Sex-chromosome-analysis"><span class="toc-text">8.Sex chromosome analysis</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#9-Kinship-generation"><span class="toc-text">9.Kinship generation</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#10-Resources"><span class="toc-text">10.Resources</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#10-1-UCSC-RefFlat-Genes"><span class="toc-text">10.1 UCSC RefFlat Genes</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#10-2-Gencode-Genes"><span class="toc-text">10.2 Gencode Genes</span></a></li></ol>
<hr>
<h3 id="1-介绍"><a href="#1-介绍" class="headerlink" title="1.介绍"></a>1.介绍</h3><p><strong><a target="_blank" rel="noopener" href="http://zhanxw.github.io/rvtests/">Rvtests</a><strong>，全称（Rare Variant tests，稀有变异），rvtests已开发为支持遗传关联分析和Meta分析的综合工具。它可以分析unrelated individual和family-based individuals的quantitative 和binary结果。它包括各种关联分析（例如，</strong>single variant score test, burden test, variable threshold test, SKAT test, fast linear mixed model score test</strong>）。它采用 VCF&#x2F;BGEN&#x2F;PLINK作为基因型输入文件，并采用PLINK格式表型文件和协变量文件。</p>
<p>此软件包能够在计算机工作站上的线性混合模型中分析多达1,000,000个人的数据集。RVTESTS支持single variant and gene-level tests。它还允许以RAREMETAL格式在得分统计之间高效生成协方差矩阵，可用于支持下一波整合了大型生物库数据集的Meta分析。</p>
<p>可以使用线性回归或逻辑回归模型处理更大的样本量。</p>
<h3 id="2-引用"><a href="#2-引用" class="headerlink" title="2.引用"></a>2.引用</h3><p><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4848408/">RVTESTS</a>: An Efficient and Comprehensive Tool for Rare Variant Association Analysis Using Sequence Data. Bioinformatics, 2016, 32:1423-1426.</p>
<h3 id="3-下载和安装"><a href="#3-下载和安装" class="headerlink" title="3.下载和安装"></a>3.下载和安装</h3><p>源代码可以从github或github页面下载最新版本。</p>
<pre><code>git clone https://github.com/zhanxw/rvtests
cd retests
make
</code></pre>
<p>编译成功后，可执行文件将位于executable文件夹下。只需键入executable&#x2F;rvtests即可开始。</p>
<p>或者，可以从<a target="_blank" rel="noopener" href="https://github.com/zhanxw/rvtests/releases">此处</a>下载二进制可执行文件。</p>
<h3 id="4-快速教程"><a href="#4-快速教程" class="headerlink" title="4.快速教程"></a>4.快速教程</h3><h3 id="4-1-Single-variant-tests"><a href="#4-1-Single-variant-tests" class="headerlink" title="4.1 Single variant tests"></a>4.1 Single variant tests</h3><pre><code>rvtest --inVcf input.vcf --pheno phenotype.ped --out output --single wald,score
</code></pre>
<p>phenotype.ped中推荐controls as 1, cases as 2, missing phenotypes as -9 or 0.</p>
<p>对于其他类型的关联分析，可以参考如下<strong>Models</strong>：</p>
<p><img src="/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/single.jpg" alt="Single variant tests Models"></p>
<h3 id="4-2-Groupwise-tests"><a href="#4-2-Groupwise-tests" class="headerlink" title="4.2 Groupwise tests"></a>4.2 Groupwise tests</h3><p>Groupwise tests主要包括三种类型的tests：</p>
<ul>
<li><p><strong>Burden tests</strong>：用于group variants，通常少于1％或5％稀有变异关联分析，包括：CMC test, Zeggini test, Madsen-Browning test, CMAT test和rare-cover test<br><img src="/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/burden.jpg" alt="Burden tests"></p>
</li>
<li><p><strong>Variable threshold tests</strong>：group variants under different frequency thresholds.<br><img src="/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/variablethreshold.jpg" alt="Variable threshold tests"></p>
</li>
<li><p><strong>Kernel methods</strong>:包括SKAT test和KBAC test<br><img src="/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/kernel.jpg" alt="Kernel methods"></p>
</li>
</ul>
<p>如果按基因进行稀有变异分析，需要使用<code>--geneFilerefFlat</code>格式指定参考基因。也可以<code>--gene</code>用来指定要测试的基因。例如，指定<code>--gene CFH,ARMS2</code>将对CFH和ARMS2基因执行关联分析。</p>
<pre><code>rvtest --inVcf input.vcf --pheno phenotype.ped --out output --geneFile refFlat_hg19.txt.gz --burden cmc --vt price --kernel skat,kbac
</code></pre>
<h3 id="4-3-Related-individual-tests"><a href="#4-3-Related-individual-tests" class="headerlink" title="4.3 Related individual tests"></a>4.3 Related individual tests</h3><p>首先需要创建一个亲属关系矩阵：</p>
<pre><code>vcf2kinship --inVcf input.vcf --bn --out output
</code></pre>
<p>该选项<code>--bn</code>意味着使用<code>Balding-Nicols</code>方法计算经验亲属关系。您可以指定<code>--ibs</code>获取IBS亲属关系或者<code>--pedigree input.ped</code>用于根据已知的家系信息计算血统。</p>
<p>然后，您可以使用基于线性混合模型的关联分析，例如<strong>Fast-LMM score test，Fast-LMM LRT test和Grammar-gamma tests</strong>：</p>
<pre><code>rvtest --inVcf input.vcf --pheno phenotype.ped --out output --kinship output.kinship --single famScore,famLRT,famGrammarGamma
</code></pre>
<h3 id="4-4-Meta-analysis-tests"><a href="#4-4-Meta-analysis-tests" class="headerlink" title="4.4 Meta-analysis tests"></a>4.4 Meta-analysis tests</h3><p>The meta-analysis models outputs association test results and genotype covariance matrix. These calculated summary statistics can be used in rare variant association analysis (<a target="_blank" rel="noopener" href="https://github.com/zhanxw/rvtests#meta-analysis-models">details</a>). We provide single variant score test and generate a genotype covariance matrix. You can use this command:</p>
<pre><code>rvtest --inVcf input.vcf --pheno phenotype.ped --meta score,cov --out output
</code></pre>
<p>In a more realistic scenario, you may want to adjust for covariates and want to inverse normalized residuals obtained in null model (<a target="_blank" rel="noopener" href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.2852.html">link</a> to our methodology paper), then this command will work:</p>
<pre><code>rvtest --inVcf input.vcf --pheno phenotype.ped --covar example.covar --covar-name age,bmi --inverseNormal --useResidualAsPhenotype  --meta score,cov --out output
</code></pre>
<p><code>--covar</code>指定协变量文件，<code>--covar-name</code>指定在分析中使用的协变量。协变量文件格式可以在<a target="_blank" rel="noopener" href="http://zhanxw.github.io/rvtests/#Covariate%20file">这里</a>找到。<code>--inverseNormal --useResidualAsPhenotype</code>指定trait transformation method。这意味着首先在协变量上拟合表型的回归模型（自动添加截距），然后对残差进行逆归一化。<code>trait transformation</code>的详细信息可以在<a target="_blank" rel="noopener" href="http://zhanxw.github.io/rvtests/#Trait%20transformation">这里</a>找到。</p>
<p>We support both unrelated individuals and related individuals (e.g. family data)。需要<code>--kinship input.kinship</code>：</p>
<pre><code>rvtest --inVcf input.vcf --pheno phenotype.ped --meta score,cov --out output --kinship input.kinship
</code></pre>
<p>其中<code>input.kinship</code>是由<code>vcf2kinship</code>计算。</p>
<p><strong>注意</strong>：默认情况下，协方差矩阵是在一百万个基对的滑动窗口中计算的。您可以通过选项更改此设置<code>windowSize</code>。例如，<code>--meta cov[windowSize=500000]</code>指定一个500k-bp的滑动窗口。</p>
<p><img src="/GWAS-Rvtests%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/meta.jpg" alt="Meta-analysis models"></p>
<p>The above models are suitable to generate summary statistics which can be later meta-analyzed (see <a target="_blank" rel="noopener" href="http://www.nature.com/ng/journal/v46/n2/abs/ng.2852.html">Dajiang Liu (2014) Nature Genetics</a>). Rvtests implemented the above methods and the results can be further analyzed by RareMetals (<a target="_blank" rel="noopener" href="http://genome.sph.umich.edu/wiki/RareMETALS">link</a>) for quantitative trait and RareMetals2 (<a target="_blank" rel="noopener" href="http://genome.sph.umich.edu/wiki/RareMETALS2">link</a>). It also worth to mention that our group offers another toolset for meta analysis (<a target="_blank" rel="noopener" href="http://genome.sph.umich.edu/wiki/Rare-Metal">link</a>).</p>
<h3 id="4-4-1-Explanation-of-outputs"><a href="#4-4-1-Explanation-of-outputs" class="headerlink" title="4.4.1 Explanation of outputs"></a>4.4.1 Explanation of outputs</h3><p>The meta-analysis results come as two files: summary score statistics files (prefix.MetaScore.assoc.gz) and covariance files (prefix.MetaCov.assoc.gz).</p>
<p>In summary score statistics files, you will obtain these columns:</p>
<ul>
<li>N_INFORMATIVE: Number of samples that are analyzed for association.</li>
<li>AF: allele frequency. For related individuals, we use BLUE estimator. For case-control study, we list overall frequency (adjusted by relatedness if possible), case frequency and control frequency separated by a colon.</li>
<li>INFORMATIVE_ALT_AC: The number of alternative alleles in the analyzed samples.</li>
<li>CALL_RATE: The fraction of non-missing alleles. For case-control study, we calculate call rate for all samples, case samples and control samples separated by a colon.</li>
<li>HWE_PVALUE: Hardy-Weinberg equilibrium. For related individuals, this statistic can be inflated. For case-control study, we calculate HWE pvalues for all samples, case samples and controls samples separated by a colon.</li>
<li>N_REF&#x2F;N_HET&#x2F;N_ALT: Number of samples carrying homozygous reference&#x2F;heterozygous&#x2F;homozygous alternative alleles. For case-control study, we calculate these three statistics for all samples, case samples and controls samples separated by a colon.</li>
<li>U_STAT, SQRT_V_STAT: U and V statistics are score statistics and their covariance matrix. Details can be found in Dajiang Liu (2014) Nature Genetics.</li>
<li>ALT_EFFSIZE: for continuous outcome, this is the estimated effect size; for binary outcome, this is the estimated log odds-ratio. We apply a new correction method when binary trait associations for related individuals are analyzed in standard linear mixed models. The log odds ratio is approximately correct for related individual analysis as well.</li>
</ul>
<p>The covariance files stores covariances in sliding windows. You will obtain these columns:</p>
<ul>
<li>CHROM: the chromosome name</li>
<li>START_POS&#x2F;END_POS: within one sliding window, the first&#x2F;last variant chromosomal position.</li>
<li>NUM_MARKER: number of markers in one sliding window.</li>
<li>MARKER_POS: all chromosomal positions in the sliding window.</li>
<li>COV: covariances between markers. When the sliding window has markers G.1, G.2, …, G.N, this column will store “covariances” of (G.1, G.1), (G.1, G.2), …, (G.1, G.N); for binary outcomes, this column uses colons as separators and additionally stores covariances between N genotypes and K covariates (G.1, C.1), (G.1, C.2), …, (G.1, C.K), as well as covariances between K covariates (C.1, C.1), (C.1, C.2), … (C.1, C.K), (C.2, C.2), …, (C.2, C.K), …, (C.K, C.K).</li>
</ul>
<h3 id="4-4-2-Dominant-models-and-recessive-models"><a href="#4-4-2-Dominant-models-and-recessive-models" class="headerlink" title="4.4.2 Dominant models and recessive models"></a>4.4.2 Dominant models and recessive models</h3><p>通过在<code> –meta</code>选项之后附加<code>dominant</code>和&#x2F;或<code>recessive</code>，可以支持显性和隐性疾病模型。例如，使用<code>–meta dominant,recessive</code>将生成两组文件。对于dominant模型，分别是<code>prefix.MetaDominant.assoc</code>和<code>prefix.MetaDominantCov.assoc.gz</code>；对于recessive模型，分别是<code>prefix.MetaRecessive.assoc</code>和<code>prefix.MetaRecessiveCov.assoc.gz</code>。在dominant模型中，基因型0&#x2F;1&#x2F;2被编码为0&#x2F;1&#x2F;1；在recessive模型中，基因型0&#x2F;1&#x2F;2被编码为0&#x2F;0&#x2F;1。基因型缺失将被推算为均值。</p>
<h3 id="5-输入文件"><a href="#5-输入文件" class="headerlink" title="5.输入文件"></a>5.输入文件</h3><h3 id="5-1-Genotype-files（VCF，BCF，BGEN，KGG）"><a href="#5-1-Genotype-files（VCF，BCF，BGEN，KGG）" class="headerlink" title="5.1 Genotype files（VCF，BCF，BGEN，KGG）"></a>5.1 Genotype files（VCF，BCF，BGEN，KGG）</h3><p>Rvtests支持VCF（变体调用格式）文件。支持未压缩格式或压缩格式的vcf文件。To use group-based rare variant tests，需要使用<a target="_blank" rel="noopener" href="http://samtools.sourceforge.net/tabix.shtml">tabix</a>为VCF文件建立索引。</p>
<p>以下是将未压缩VCF格式转换为bgzipped VCF格式的命令：</p>
<pre><code>(grep ^&quot;#&quot; $your_old_vcf; grep -v ^&quot;#&quot; $your_old_vcf | sed &#39;s:^chr::ig&#39; | sort -k1,1n -k2,2n) | bgzip -c &gt; $your_vcf_file 
tabix -f -p vcf $your_vcf_file
</code></pre>
<p>上述命令将（1）chr从染色体名称中删除前缀；（2）首先按染色体，然后按染色体位置对VCF文件进行排序；（3）使用bgzip压缩；（4）创建tabix索引。</p>
<p>Rvtests支持<code>genotype dosages</code>。使用<code>--dosage DosageTag</code>指定dosage tag。例如，如果VCF格式字段为<code>GT：EC</code>，单个基因型字段为<code>0/0：0.02</code>，则可以使用<code>--dosage EC</code>，并且rvtests将在回归模型中使用剂量0.02。</p>
<p>Rvtests支持BGEN输入格式 v1.0到v1.3。使用<code>--inBgen</code>指定一个BGEN文件,<code>--inBgenSample</code>指定SAMPLE文件。</p>
<p>Rvtests支持KGGSeq输入格式。此格式是对二进制PLINK格式的扩展。使用–inKgg来代替–inVcf。</p>
<h3 id="5-2-Phenotype-files"><a href="#5-2-Phenotype-files" class="headerlink" title="5.2 Phenotype files"></a>5.2 Phenotype files</h3><p>可以使用<code>--mpheno $phenotypeColumnNumber</code>或<code>--pheno-name</code>指定特定表型。</p>
<p>表型文件示例（<code>example.pheno</code>）具有以下格式：</p>
<pre><code>fid iid fatid matid sex y1 y2 y3 y4
P1 P1 0 0 0 1.7642934435605 -0.733862638327895 -0.980843608339726 2
P2 P2 0 0 0 0.457111744989746 0.623297281416372 -2.24266162284447 1
P3 P3 0 0 0 0.566689682543218 1.44136462889459 -1.6490100777089 1
P4 P4 0 0 0 0.350528353203767 -1.79533911725537 -1.11916876241804 1
P5 P5 0 0 1 2.72675074738545 -1.05487747371158 -0.33586430010589 2
</code></pre>
<p>表型文件由<code>--pheno example.pheno</code> 指定。默认表型列标题为<code>y1</code>。如果要使用替代列作为关联分析的表型（例如标题为y2的列），则可以使用列或名称来指定表型</p>
<pre><code>–mpheno 2
–pheno-name y2
</code></pre>
<p>注意：要使用<code>--pheno-name</code>，标题行必须以<code>fid iid</code>开头。</p>
<p>在表型文件中，缺失值可以用<code>NA</code>或<code>任何非数字值</code>表示。缺少表型的个体将自动从后续关联分析中删除。对于每个缺失的表型值，将生成一条警告并将其记录在日志文件中。</p>
<p>当表型值分别为0、1和2时，rvtests会自动将其视为binary traits。但是，如果要将其视为连续性状，需使用<code>--qtl</code>选项。</p>
<h3 id="5-3-Covariate-files"><a href="#5-3-Covariate-files" class="headerlink" title="5.3 Covariate files"></a>5.3 Covariate files</h3><p>可以使用<code>--covar</code>和<code>--covar-name</code>指定将用于单变量关联分析的协变量。这是一个可选参数。如果数据中没有协变量，则可以忽略此选项。</p>
<p>协变量文件（例如<code>example.covar</code>）与表型文件具有相似的格式：</p>
<pre><code>fid iid fatid matid sex y1 y2 y3 y4
P1 P1 0 0 0 1.911 -1.465 -0.817 1
P2 P2 0 0 0 2.146 -2.451 -0.178 2
P3 P3 0 0 0 1.086 -1.194 -0.899 1
P4 P4 0 0 0 0.704 -1.052 -0.237 1
P5 P5 0 0 1 2.512 -3.085 -2.579 1
</code></pre>
<p>协变量文件由<code>--covar</code>选项指定（例如<code>--covar example.covar</code>）。要指定将在关联分析中使用的协变量，可以使用该选项<code>--covar-name</code>。例如，当使用<code>age，bmi</code>和3个PCAs进行关联分析时，可以为rvtests程序指定以下选项，即 <code>--covar example.covar --covar-name age,bmi,pc1,pc2,pc3</code>。</p>
<p>注意：协变量文件中缺少的数据可以用任何非数字值（例如NA）标记。它们将自动推算为数据文件中的平均值。</p>
<h3 id="5-4-Trait-transformation"><a href="#5-4-Trait-transformation" class="headerlink" title="5.4 Trait transformation"></a>5.4 Trait transformation</h3><p>In this meta-analysis, we use inverse normal transformed residuals in the association analysis, which is achieved by using a combination of <code>--inverseNormal</code> and <code>--useResidualAsPhenotype</code>. Specifically, we first fit the null model by regressing phenotype on covariates. The residuals are then inverse normal transformed (see <strong>Appendix A</strong> more detailed formula for transformation). Transformed residuals will be used to obtain score statistics.</p>
<pre><code>./rvtest --inVcf $vcf --pheno $example.pheno --covar example.covar --covar-name age,bmi --inverseNormal --useResidualAsPhenotype  --meta score,cov --out $output_prefix
</code></pre>
<h3 id="6-Models"><a href="#6-Models" class="headerlink" title="6. Models"></a>6. Models</h3><h3 id="6-1-Utility-models"><a href="#6-1-Utility-models" class="headerlink" title="6.1 Utility models"></a>6.1 Utility models</h3><p>Rvtests has an convenient option –outputRaw. When specifying this, rvtests can output genotypes, phenotype, covariates (if any) and collapsed genotype to tabular files. These files can be imported into other software (e.g. R) for further analyses.</p>
<h3 id="7-Association-test-options"><a href="#7-Association-test-options" class="headerlink" title="7. Association test options"></a>7. Association test options</h3><h3 id="7-1-Sample-inclusion-x2F-exclusion"><a href="#7-1-Sample-inclusion-x2F-exclusion" class="headerlink" title="7.1 Sample inclusion&#x2F;exclusion"></a>7.1 Sample inclusion&#x2F;exclusion</h3><p>Rvtests可以灵活地指定要包括或排除的样本：</p>
<pre><code>--peopleIncludeID : give IDs of people that will be included in study
--peopleIncludeFile : from given file, set IDs of people that will be included in study
--peopleExcludeID : give IDs of people that will be included in study
--peopleExcludeFile : from given file, set IDs of people that will be included in study
</code></pre>
<h3 id="7-2-Variant-site-filters"><a href="#7-2-Variant-site-filters" class="headerlink" title="7.2 Variant site filters"></a>7.2 Variant site filters</h3><p>通常在稀有变异分析中使用不同的frequency cut-off。因此，rvtests指定frequency cut-offs。<br>Frequency Cut-off</p>
<pre><code>--freqUpper: Specify upper minor allele frequency bound to be included in analysis
--freqLower: Specify lower minor allele frequency bound to be included in analysis
</code></pre>
<p>如果指定<code>--freqLower 0.01 --freqUpper 0.05</code>，则仅分析次要等位基因频率在0.01到0.05（含边界）之间的变体。</p>
<p>与Sample inclusion&#x2F;exclusion类似，可以通过<code>--rangeList</code>选项来指定要包含的变异范围。例如·–rangeList 1:100-200·将1号染色体的位置包括100bp至200bp的区域。或者，使用单独的文件，<code>range.txt</code> <code>--rangeFile range.txt</code>指定关联分析范围。</p>
<pre><code>--rangeList: Specify some ranges to use, please use chr:begin-end format.
--rangeFile: Specify the file containing ranges, please use chr:begin-end format.
--siteFile: Specify the file containing sites to include, please use &quot;chr pos&quot; format.
</code></pre>
<p>It is supported to filter variant site by site depth, minor allele count or annotation (annotated VCF file is needed).</p>
<pre><code>--siteDepthMin : Specify minimum depth(inclusive) to be included in analysis
--siteDepthMax : Specify maximum depth(inclusive) to be included in analysis
--siteMACMin : Specify minimum Minor Allele Count(inclusive) to be included in analysis
--annoType : Specify annotation type that is followed by ANNO= in the VCF INFO field, regular expression is allowed
</code></pre>
<p>注意：<code>--annoType</code>可以基于正则表达式过滤变异。例如，<code>--annoType Nonsynonymous</code>将仅分析ANNO&#x3D;Nonsynonymous在INFO字段中存在的非同义变异。要提取一种以上的注释类型，请使用<code>--annoType &#39;Start_Gain|Stop_Loss|Start_Loss|Essential_Splice_Site|Stop_Gain|Normal_Splice_Site|Synonymous|Nonsynonymous</code>提取LOF（功能丧失）突变。要生成带注释的VCF文件，请阅读下一节<code>Annotation</code>。</p>
<h3 id="7-2-1-Annotation"><a href="#7-2-1-Annotation" class="headerlink" title="7.2.1 Annotation"></a>7.2.1 Annotation</h3><p>在此，我们将带有注释信息的VCF文件定义为带注释的VCF。可以使用ANNO（一种快速准确的注释软件）来完成注释步骤。最少有两个步骤来注释VCF文件：</p>
<ul>
<li>安装ANNO及其资源文件</li>
</ul>
<pre><code>git clone https://github.com/zhanxw/anno.git; cd anno; 使; cd资源；./download.sh
</code></pre>
<ul>
<li>运行以下脚本：</li>
</ul>
<pre><code>anno -i input.vcf -o output.vcf.gz -r anno / resources / hs37d5.fa -g anno / resources / refFlat_hg19.txt.gz -p anno / priority.txt -c anno / codon.txt –indexOutput
</code></pre>
<p>然后，您将获得带注释的VCF文件output.vcf.gz及其tabix索引output.vcf.gz.tbi。</p>
<h3 id="7-3-Genotype-filters"><a href="#7-3-Genotype-filters" class="headerlink" title="7.3 Genotype filters"></a>7.3 Genotype filters</h3><p>可以通过以下选项过滤掉深度较浅或质量较差的基因型：</p>
<pre><code>-indvDepthMin : Specify minimum depth(inclusive) of a sample to be included in analysis
--indvDepthMax : Specify maximum depth(inclusive) of a sample to be included in analysis
--indvQualMin : Specify minimum depth(inclusive) of a sample to be included in analysis
</code></pre>
<h3 id="7-4-处理缺失的基因型和表型"><a href="#7-4-处理缺失的基因型和表型" class="headerlink" title="7.4 处理缺失的基因型和表型"></a>7.4 处理缺失的基因型和表型</h3><p>When genotypes are missing (e.g. genotype &#x3D; “.&#x2F;.”) or genotypes are filtered out, there are three options to handle them: </p>
<ul>
<li><p>(1) impute to its mean(default option); </p>
</li>
<li><p>(2) impute by HWE equilibrium; </p>
</li>
<li><p>(3) remove from the model. </p>
<p>Use –impute [mean|hwe|drop] to specify which option to use.</p>
</li>
</ul>
<p>例如，当缺少定量表型时，某些样本具有基因型文件，但没有表型，rvtest可以将缺失表型推算为其平均值。</p>
<p>注意：请勿将<code>--imputePheno</code>用于binary trait。</p>
<p>总之，可以使用以下两个选项：</p>
<pre><code>--impute : Specify either of mean, hwe, and drop
--imputePheno : Impute phenotype to mean by those have genotypes but no phenotypes
</code></pre>
<h3 id="7-5-Specify-groups-e-g-burden-unit"><a href="#7-5-Specify-groups-e-g-burden-unit" class="headerlink" title="7.5 Specify groups (e.g burden unit)"></a>7.5 Specify groups (e.g burden unit)</h3><p>使用<code>--geneFile</code>选项指定基因文件名。例如，<code>--geneFile refFlat_hg19.txt.gz</code>将<code>refFlat_hg19.txt.gz</code>用作基因定义文件，然后对每个基因执行关联分析。使用<code>--gene</code>指定的基因的一个子集的测试。例如，<code>--gene CFH</code>将仅测试CFH基因。</p>
<p>Alternative grouping unit can be specified as set. These sets are treated similar to gene. You can thus use <code>--setFile </code>to define sets (similar to <code>--geneFile</code> option), and use <code>--set</code> to define a specific set (similar to<code> --gene option</code>). Additionally, use <code>--setList</code> can specify a set to test from command line.</p>
<p>The format of a set file is: (1) set names; (2) ranges (e.g. chrom:begin-end); For example, you have a set file, example.set, like this:</p>
<pre><code>set1 1:100-200,1:250-300
set2 2:500-600
</code></pre>
<p>您可以指定<code>--setFile example.set --set set2</code>将2号染色体内500至600bp的变体分组。如果要测试特定区域，例如2号染色体，位置500到550bp，但又不想制作另一个文件，则可以使用<code>--setList 2:500-600</code>。</p>
<p>概括来说，与分组单位有关的选项如下：</p>
<pre><code>--geneFile : specify a gene file (for burden tests)
--gene : specify which genes to test
--setList : specify a list to test (for burden tests)
--setFile : specify a list file (for burden tests, first two columns:setName chr:beg-end)
--set : specify which set to test (1st column)
</code></pre>
<h3 id="8-Sex-chromosome-analysis"><a href="#8-Sex-chromosome-analysis" class="headerlink" title="8.Sex chromosome analysis"></a>8.Sex chromosome analysis</h3><p>Rvtests support X chromosome analysis. In human X chromosome, there is PAR (pseudoautosomal region) and non-PAR region. For males, there are two X allele in PAR region and one allele in non-PAR region. While the PAR region is treated in the same way as autosomes, rvtests treat non-PAR region differently. Below we will describe the details about how rvtests handles non-PAR region.</p>
<p>Prepare data. According to VCF standard, male genotype needs to coded as 0 or 1. For compatibility, rvtests also support 0&#x2F;0 or 1&#x2F;1 coding. In VCF files, male genotypes can be written as <code>“0”, “1”, “0|0”, “0/0”, “1|1”, “1/1”</code>. All other genotypes will be treated as missing.</p>
<p>Genotype in regression models. For consistence, male genotypes are converted to 0 or 2. When male dosages are provided, we expect the values in the VCF file are between 0.0 and 1.0, and then will model them as twice the value.</p>
<p>MetaScore results. If –meta score is specified, the output file prefix.MetaScore.assoc.gz includes both PAR-region and non-PAR region analysis. However, in the non-PAR region, the difference is that Hardy-Weinberg P-value and homozygous-reference&#x2F;heterzygous&#x2F;homozygous-alternative sample sizes are calculated using female samples only.</p>
<p>Related individuals. Just append –xHemi to the vcf2kinship (more details in Kinship generation) and rvtest command lines. Rvtests can recognize non-PAR region kinship file and use it in the analysis.</p>
<p>PAR region. PAR region is defined as two regions <code>X:60001-2699520 </code>and <code>X:154931044-155260560</code>. Use <code>--xLabel</code> can specify which chromosome has PAR region (default: 23 or X) and use <code>--xParRegion</code> to specify PAR region (default: hg19, meaning ‘60001-2699520,154931044-155260560’ in the UCSC build hg19, specify “hg18” will use PAR region definition in the UCSC build hg18, or specify “hg38” will use UCSC build 38).</p>
<h3 id="9-Kinship-generation"><a href="#9-Kinship-generation" class="headerlink" title="9.Kinship generation"></a>9.Kinship generation</h3><p>分析相关个人通常需要估计亲属关系。可以使用<code>vcf2kinship</code>。</p>
<p><code>vcf2kinship</code> can calculate pedigree kinship using a pedigree input file (PED format, see Phenotype file, use option <code>--ped</code>). The output file name is specified by <code>--prefix</code> option. If you use <code>--prefix</code> output then the output files will include <code>output.kinship</code>.</p>
<p>It can also calculate empirical kinship using genotype input file (VCF format, see Genotype file (VCF), use option <code>--inVcf</code>). For empirical kinship, you also need to specify the kinship model, either Balding-Nicols model (use option <code>--bn</code>) or Identity-by-state model (use option <code>--ibs</code>).</p>
<p>In sex chromosome analysis, it is often required to generate kinship on X chromosome regions, then you need to specify <code>--xHemi</code>. If your input VCF file has different X chromosome label (e.g. chromosome name is ‘23’ instead of ‘X’, you can use <code>--xLabel 23</code>.</p>
<p>If principal component decomposition (PCA) results are needed, you can use option <code>--pca</code>. Then output files with suffix ‘.pca’ include PCA results.</p>
<p>When dealing with large input files, it is often preferred to use multiple CPU to speed up calculation using the option <code>--thread N </code>in which N is the number of CPU.</p>
<p>For example, to generate pedigree-based kinship (<code>--ped</code>) on both autosomal region and X chromosome (<code>--xHemi</code>) region, the command line is:</p>
<pre><code>vcf2kinship --ped input.ped --xHemi --out output
</code></pre>
<p>To generate empirical kinship (<code>--inVcf</code>) on both autosomal region and X chromosome (<code>--xHemi</code>) region using Balding-Nicols model, the command line is:</p>
<pre><code>vcf2kinship --inVcf input.vcf.gz --ped input.ped --bn --xHemi --out output
</code></pre>
<p>NOTE: you need to provide a pedigree file (PED) in the above case, as vcf2kinship needs the sex information of samples to construct kinship for sex chromosome.</p>
<p>For modern genetic datasets, genotype data is often stored by chromosomes, with each chromosome stored in a separate VCF file. In this case, kinship matrix can be first calcualted separately for each chromosome, and then combined using the python script <a target="_blank" rel="noopener" href="https://github.com/zhanxw/rvtests/tree/master/misc">combineKinship.py</a>. Specifically, if you generated two kinship matrices for chr1 chr2 (chr1.kinship, and chr2.kinship), you can run the python script to combine them, i.e.</p>
<pre><code>python combineKinship.py -o prefix chr1.kinship chr2.kinship
</code></pre>
<p>所得的亲属关系矩阵等效于使用合并的vcf文件计算的亲属关系矩阵。</p>
<h3 id="10-Resources"><a href="#10-Resources" class="headerlink" title="10.Resources"></a>10.Resources</h3><h3 id="10-1-UCSC-RefFlat-Genes"><a href="#10-1-UCSC-RefFlat-Genes" class="headerlink" title="10.1 UCSC RefFlat Genes"></a>10.1 UCSC RefFlat Genes</h3><pre><code>refFlat_hg19.txt.gz
</code></pre>
<p>UCSC基因组版本中refFlat格式的UCSC基因定义文件（<a target="_blank" rel="noopener" href="http://genome.ucsc.edu/goldenpath/gbdDescriptionsOld.html#RefFlat">详细信息</a>）。</p>
<p><strong>File link</strong>：http : &#x2F;&#x2F;qbrc.swmed.edu&#x2F;zhanxw&#x2F;seqminer&#x2F;data&#x2F;refFlat_hg19.txt.gz</p>
<h3 id="10-2-Gencode-Genes"><a href="#10-2-Gencode-Genes" class="headerlink" title="10.2 Gencode Genes"></a>10.2 Gencode Genes</h3><pre><code>refFlat.gencode.v19.gz
</code></pre>
<p>Gencode基因定义版本19，采用<code>refFlat</code>格式（<a target="_blank" rel="noopener" href="http://www.gencodegenes.org/releases/19.html">详细信息</a>）。</p>
<p><strong>File link</strong>：http : &#x2F;&#x2F;qbrc.swmed.edu&#x2F;zhanxw&#x2F;seqminer&#x2F;data&#x2F;refFlat.gencode.v19.gz</p>

            


        </div>
    </div>
    <div id="post-footer" class="post-footer main-content-wrap">
        
        
            <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-23-ancestry-zu-yuan-fen-xi-peddy/"
                    data-tooltip="Ancestry祖源分析|peddy"
                    aria-label="上一篇: Ancestry祖源分析|peddy"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-15-gwas-epacts-an-zhuang-ji-shi-yong-jiao-cheng/"
                    data-tooltip="GWAS|EPACTS安装及使用"
                    aria-label="下一篇: GWAS|EPACTS安装及使用"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


        
        
    </div>
</article>



                <footer id="footer" class="main-content-wrap">
    <span class="copyrights">
        Copyrights &copy; 2022 Dr.iuu. All Rights Reserved.
    </span>
</footer>

            </div>
            
                <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
                    <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-23-ancestry-zu-yuan-fen-xi-peddy/"
                    data-tooltip="Ancestry祖源分析|peddy"
                    aria-label="上一篇: Ancestry祖源分析|peddy"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-15-gwas-epacts-an-zhuang-ji-shi-yong-jiao-cheng/"
                    data-tooltip="GWAS|EPACTS安装及使用"
                    aria-label="下一篇: GWAS|EPACTS安装及使用"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


                </div>
                
    <div id="share-options-bar" class="share-options-bar" data-behavior="4">
        <i id="btn-close-shareoptions" class="fa fa-times"></i>
        <ul class="share-options">
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                        aria-label="分享到 Facebook"
                    >
                        <i class="fab fa-facebook" aria-hidden="true"></i><span>分享到 Facebook</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                        aria-label="分享到 Twitter"
                    >
                        <i class="fab fa-twitter" aria-hidden="true"></i><span>分享到 Twitter</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-20-gwas-rvtests-an-zhuang-ji-shi-yong-jiao-cheng/"
                        aria-label="分享到 Google+"
                    >
                        <i class="fab fa-google-plus" aria-hidden="true"></i><span>分享到 Google+</span>
                    </a>
                </li>
            
        </ul>
    </div>


            
        </div>
        


    
        
    

<div id="about">
    <div id="about-card">
        <div id="about-btn-close">
            <i class="fa fa-times"></i>
        </div>
        
            <img id="about-card-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
            <h4 id="about-card-name">Dr.iuu</h4>
        
            <div id="about-card-bio"><p>NCMC-SCMC-SWCH</p>
</div>
        
        
            <div id="about-card-job">
                <i class="fa fa-briefcase"></i>
                <br/>
                <p>Bioinfo-CHD</p>

            </div>
        
        
            <div id="about-card-location">
                <i class="fa fa-map-marker-alt"></i>
                <br/>
                Pudong, Shanghai
            </div>
        
    </div>
</div>

        
        
<div id="cover" style="background-image:url('/assets/images/cover2.jpg');"></div>
        <!--SCRIPTS-->

<script src="/assets/js/script-3nlvflbbvekvxx0g6h2clfqhhxs61cvvbofv7nmctigkdbbhu5htlvmrqg80.min.js"></script>

<!--SCRIPTS END-->





    </body>
</html>
