
<!DOCTYPE html>
<html lang="zh-cn">
    
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="iuu&#39;s blog">
    <title>代码分析 | Sing-cell Sequencing QC详解 - iuu&#39;s blog</title>
    <meta name="author" content="Dr.iuu">
    
    
        <link rel="icon" href="https://iuu19.github.io/assets/images/coveriuu.jpg">
    
    
        
            <link rel="alternate" type="application/atom+xml" title="RSS" href="/atom.xml">
        
    
    <script type="application/ld+json">{"@context":"http://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg"},"articleBody":"代码分析 | Sing-cell Sequencing QC详解\n\n\n\n\n\n\n\n最近找到了芬兰CSC-IT科学中心主讲的生物信息课程(https://www.csc.fi/web/training/-/scrnaseq)，官网上还提供了练习素材以及详细代码，今天就来练习一下单细胞数据QC的过程：\n第一部分是使用Seurat（https://satijalab.org/seurat/）来对QC进行可视化并进行细胞过滤；第二部分将详细探讨scater软件包（https://bioconductor.org/packages/release/bioc/html/scater.html）。\n\n\n加载环境\n\n\n如果需要加载conda环境（软件安装不上，可能是网速慢！Conda&#x2F;R&#x2F;pip&#x2F;brew等国内镜像大全拿走不谢~~），请在运行以下代码之前遵循以下说明：https://nbisweden.github.io/excelerate-scRNAseq/computing_environment_instructions.html。\n\n\n数据集\n\n在本教程中，我们将使用10x Genomics网站上的3个不同的PBMC数据集:https://support.10xgenomics.com/single-cell-gene-expression/datasets\n包括：\n\n1k PBMCs using 10x v2 chemistry\n1k PBMCs using 10x v3 chemistry\n1k PBMCs using 10x v3 chemistry in combination with cell surface proteins, but disregarding the protein data and only looking at gene expression.\n\n我们可以通过以下命令进行下载：\nmkdir data  #建立data文件夹\ncd data\ncurl -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v2/pbmc_1k_v2_filtered_feature_bc_matrix.h5\ncurl -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v3/pbmc_1k_v3_filtered_feature_bc_matrix.h5\ncurl -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_10k_v3/pbmc_10k_v3_filtered_feature_bc_matrix.h5\n\n\n\n加载R包\n\nsuppressMessages(require(Seurat))suppressMessages(require(scater))suppressMessages(require(Matrix))读入数据v3.1k &lt;- Read10X_h5(“&#x2F;scrna-seq2019&#x2F;day1&#x2F;1-qc&#x2F;pbmc_1k_v3_filtered_feature_bc_matrix.h5”, use.names &#x3D; T)v2.1k &lt;- Read10X_h5(“&#x2F;scrna-seq2019&#x2F;day1&#x2F;1-qc&#x2F;pbmc_1k_v2_filtered_feature_bc_matrix.h5”, use.names &#x3D; T)p3.1k &lt;- Read10X_h5(“~&#x2F;scrna-seq2019&#x2F;day1&#x2F;1-qc&#x2F;pbmc_1k_protein_v3_filtered_feature_bc_matrix.h5”, use.names &#x3D; T)\nCITE-seq data中只提取基因表达矩阵p3.1k &lt;- p3.1k$Gene ExpressionHemberg-lab单细胞转录组数据分析（七）- 导入10X和SmartSeq2数据Tabula Muris。Seurat构建Seurat对象\n首先，为每个数据集创建Seurat对象，然后合并为一个大的seurat对象。\nsdata.v2.1k &lt;- CreateSeuratObject(v2.1k, project &#x3D; “v2.1k”)sdata.v3.1k &lt;- CreateSeuratObject(v3.1k, project &#x3D; “v3.1k”)sdata.p3.1k &lt;- CreateSeuratObject(p3.1k, project &#x3D; “p3.1k”)\n使用merge函数合并为一个seurat object. 为了防止数据集之间的barcodes重叠，给细胞添加ID。alldata &lt;- merge(sdata.v2.1k, c(sdata.v3.1k,sdata.p3.1k), add.cell.ids&#x3D;c(“v2.1k”,”v3.1k”,”p3.1k”))\n在metadata中添加v2和v3标识。chemistry &lt;- rep(“v3”,ncol(alldata))chemistry[Idents(alldata) &#x3D;&#x3D; “v2.1k”] &lt;- “v2”alldata &lt;- AddMetaData(alldata, chemistry, col.name &#x3D; “Chemistry”)alldata\nAn object of class Seurat33538 features across 2931 samples within 1 assayActive assay: RNA (33538 features)检查每个样本的细胞数量，将其存储在metadata的orig.ident中，并自动设置为active ident。table(Idents(alldata))##\np3.1k v2.1k v3.1k713   996  1222计算线粒体比例\nSeurat可以自动计算一些QC统计信息，例如每个细胞的UMI数量和基因数分别存储在metadata的nCount_RNA和nFeature_RNA列（Hemberg-lab单细胞转录组数据分析（六）- 构建表达矩阵，UMI介绍）。\nhead(&#x61;&#x6c;&#108;&#x64;&#x61;&#x74;&#97;&#64;&#109;&#x65;&#x74;&#97;&#46;&#100;&#97;&#x74;&#97;)\norig.ident nCount_RNA nFeature_RNA Chemistryv2.1k_AAACCTGAGCGCTCCA-1      v2.1k       6631         2029        v2v2.1k_AAACCTGGTGATAAAC-1      v2.1k       2196          881        v2v2.1k_AAACGGGGTTTGTGTG-1      v2.1k       2700          791        v2v2.1k_AAAGATGAGTACTTGC-1      v2.1k       3551         1183        v2v2.1k_AAAGCAAGTCTCTTAT-1      v2.1k       3080         1333        v2v2.1k_AAAGCAATCCACGAAT-1      v2.1k       5769         1556        v2我们将手动计算线粒体reads的比例，并将其添加到metadata中。\nmt.genes &lt;- rownames(alldata)[grep(“^MT-“,rownames(alldata))]C&lt;-GetAssayData(object &#x3D; alldata, slot &#x3D; “counts”)\npercent.mito &lt;- colSums(C[mt.genes,])&#x2F;Matrix::colSums(C)*100alldata &lt;- AddMetaData(alldata, percent.mito, col.name &#x3D; “percent.mito”)计算核糖体比例\n我们用上面一样的方法来计算源自核糖体蛋白的基因表达比例。\nrb.genes &lt;- rownames(alldata)[grep(“^RP[SL]”,rownames(alldata))]percent.ribo &lt;- colSums(C[rb.genes,])&#x2F;Matrix::colSums(C)*100alldata &lt;- AddMetaData(alldata, percent.ribo, col.name &#x3D; “percent.ribo”)Plot QC\n绘制小提琴图展示一些QC指标（feature）（可视化之为什么要使用箱线图？）：\nVlnPlot(alldata, features &#x3D; “nFeature_RNA”, pt.size &#x3D; 0.1) + NoLegend()\nVlnPlot(alldata, features &#x3D; “nCount_RNA”, pt.size &#x3D; 0.1) + NoLegend()\nVlnPlot(alldata, features &#x3D; “percent.mito”, pt.size &#x3D; 0.1) + NoLegend()\nVlnPlot(alldata, features &#x3D; “percent.ribo”, pt.size &#x3D; 0.1) + NoLegend()\n如图所示，v2检测到的基因数较低，但检测出的核糖体蛋白比例却偏高。检测到较少的低表达基因是因为核糖体蛋白发生高表达，使得核糖体基因在转录景观中占较大比例。\nQC方式不同时的散点图：\nFeatureScatter(alldata, feature1 &#x3D; “nCount_RNA”, feature2 &#x3D; “nFeature_RNA”)\nFeatureScatter(alldata, feature1 &#x3D; “nFeature_RNA”, feature2 &#x3D; “percent.mito”)\nFeatureScatter(alldata, feature1&#x3D;”percent.ribo”, feature2&#x3D;”nFeature_RNA”)\n我们还可以展示一个数据集：\nFeatureScatter(alldata, feature1 &#x3D; “nCount_RNA”, feature2 &#x3D; “nFeature_RNA”, cells &#x3D; WhichCells(alldata, expression &#x3D; orig.ident &#x3D;&#x3D; “v3.1k”) )\n过滤\n线粒体过滤\n线粒体reads比例高的细胞有很多，如果过滤掉它们后还剩下足够的细胞，那删除这些细胞可能是较为明智的。另一个选择是从数据集中删除所有线粒体reads，并希望剩余的基因仍然具有足够的生物学信号。第三种选择是在标准化过程中去除percent.mito变量。\n在此次示例中，某些细胞中的线粒体reads比例高达99.7％，细胞信号很好的可能性不高，因此去除这些细胞。\n可以通过展示的图决定合理的cutoff值。在这次例子中，大部分细胞的线粒体reads低于25％，可以将其用作临界值。\n选择线粒体基因百分比&lt;25%的细胞selected &lt;- WhichCells(alldata, expression &#x3D; percent.mito &lt; 25)length(selected)\n[1] 2703对细胞取子集data.filt &lt;- subset(alldata, cells &#x3D; selected)\n画小提琴图VlnPlot(data.filt, features &#x3D; “percent.mito”)\n如图所示，mito百分比仍有很大差异，因此必须在数据分析步骤中进行处理。\n基因过滤\n检测到的基因数量高的离谱的话可能表明doublets，也有可能是因为样品中某一种细胞具有较多基因表达。\n在这次的数据集中，v2和v3之间在基因检测方面也存在明显差异，因此对所有数据应用相同的临界值可能并不合适。\n同时，在蛋白质测定数据p3.1k中，存在大量细胞几乎没有检测到具有双峰分布的基因。v3.1k和v2.12个数据集中也看不到这种分布类型。考虑到它们都是pbmc数据集，因此可以将这种分布视为低质量的库。\n通过设置cutoff值分别为4100（对于v3 chemistry）和2000（对于v2）过滤测到高基因的（可能的doublets）细胞。\n#从检测到许多基因的细胞开始。high.det.v3 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &gt; 4100)high.det.v2 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &gt; 2000 &amp; orig.ident &#x3D;&#x3D; “v2.1k”)\n细胞取子集data.filt &lt;- subset(data.filt, cells&#x3D;setdiff(WhichCells(data.filt),c(high.det.v2,high.det.v3)))\n查看细胞数ncol(data.filt)\n[1] 2631用低基因检测（低质量文库）过滤细胞，其中v3的基因数cutoff为1000，v2的基因数cutoff为500。\n#筛选基因low.det.v3 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &lt; 1000 &amp; orig.ident !&#x3D; “v2.1k”)low.det.v2 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &lt; 500 &amp; orig.ident &#x3D;&#x3D; “v2.1k”)\n去除细胞data.filt &lt;- subset(data.filt, cells&#x3D;setdiff(WhichCells(data.filt),c(low.det.v2,low.det.v3)))\n查看细胞数ncol(data.filt)\n[1] 2531再次绘制QC图\nVlnPlot(data.filt, features &#x3D; “nFeature_RNA”, pt.size &#x3D; 0.1) + NoLegend()\nVlnPlot(data.filt, features &#x3D; “nCount_RNA”, pt.size &#x3D; 0.1) + NoLegend()\nVlnPlot(data.filt, features &#x3D; “percent.mito”, pt.size &#x3D; 0.1) + NoLegend()\n在过滤前后检查每个样本的细胞数table(Idents(alldata))##\np3.1k v2.1k v3.1k713   996  1222table(Idents(data.filt))##\np3.1k v2.1k v3.1k526   933  1072计算细胞周期分数\nSeurat具有根据已知的S期和G2&#x2F;M期基因列表计算细胞周期分数的功能（Seurat亮点之细胞周期评分和回归）。\ndata.filt &lt;- CellCycleScoring(  object &#x3D; data.filt,  g2m.features &#x3D; cc.genes$g2m.genes,  s.features &#x3D; cc.genes$s.genes)\nVlnPlot(data.filt, features &#x3D; c(“S.Score”,”G2M.Score”))\n从上图可以看出不同数据集中处于S期和G2M期的细胞Score基本均为0，表明细胞周期对数据质量影响并不大。\nScater可以使用scater包完成相似的QC绘图和细胞过滤，但是由于我们使用Seurat筛选了细胞，因此现在仅使用scater探索数据中的技术偏差。\n可以直接从count矩阵创建SCE对象，网址为：https://rdrr.io/bioc/scater/f/vignettes/overview.Rmd\n在该例中，我们可以直接从Seurat对象转换为SCE对象。\nsce &lt;- as.SingleCellExperiment(data.filt)我们用访问函数来评估SingleCellExperiment对象的不同元素。\ncounts（object）：用于返回read计数矩阵。如上所示，如果未为该对象定义任何计数，则计数矩阵slot为NULL。exprs（object）：用于返回（对数计数）表达式值的矩阵，实际上为访问对象的对数计数slot（logcounts的同义词）。SCE对象还有的slot：\n细胞元数据（Cell metadata,），可以以DataFrame提供，其中行是细胞，列是细胞属性（例如细胞类型、培养条件、捕获的天数等）。特征元数据（Feature metadata），可以以DataFrame提供，其中行是特征（例如基因），列是特征属性，例如Ensembl ID，生物型，gc含量等。计算QC指标\n默认情况下，QC指标是根据count data计算得出的。但是可以通过exprs_values参数进行更改。我们还可以在函数调用中包加入有关哪些基因是线粒体的信息。\n计算QC指标sce &lt;- calculateQCMetrics(sce, feature_controls &#x3D; list(mito &#x3D; mt.genes))\ncheck what all entries are -colnames(colData(sce))\n[1] “orig.ident”[2] “nCount_RNA”[3] “nFeature_RNA”[4] “Chemistry”[5] “percent.mito”[6] “percent.ribo”[7] “S.Score”[8] “G2M.Score”[9] “Phase”[10] “ident”[11] “is_cell_control”[12] “total_features_by_counts”[13] “log10_total_features_by_counts”[14] “total_counts”[15] “log10_total_counts”[16] “pct_counts_in_top_50_features”[17] “pct_counts_in_top_100_features”[18] “pct_counts_in_top_200_features”[19] “pct_counts_in_top_500_features”[20] “total_features_by_counts_endogenous”[21] “log10_total_features_by_counts_endogenous”[22] “total_counts_endogenous”[23] “log10_total_counts_endogenous”[24] “pct_counts_endogenous”[25] “pct_counts_in_top_50_features_endogenous”[26] “pct_counts_in_top_100_features_endogenous”[27] “pct_counts_in_top_200_features_endogenous”[28] “pct_counts_in_top_500_features_endogenous”[29] “total_features_by_counts_feature_control”[30] “log10_total_features_by_counts_feature_control”[31] “total_counts_feature_control”[32] “log10_total_counts_feature_control”[33] “pct_counts_feature_control”[34] “pct_counts_in_top_50_features_feature_control”[35] “pct_counts_in_top_100_features_feature_control”[36] “pct_counts_in_top_200_features_feature_control”[37] “pct_counts_in_top_500_features_feature_control”[38] “total_features_by_counts_mito”[39] “log10_total_features_by_counts_mito”[40] “total_counts_mito”[41] “log10_total_counts_mito”[42] “pct_counts_mito”[43] “pct_counts_in_top_50_features_mito”[44] “pct_counts_in_top_100_features_mito”[45] “pct_counts_in_top_200_features_mito”[46] “pct_counts_in_top_500_features_mito”如您所见，scater为细胞计算了许多不同的QC指标。\nScater还可以根据基因计算一些统计信息：\ncolnames(rowData(sce))\n[1] “is_feature_control”      “is_feature_control_mito”[3] “mean_counts”             “log10_mean_counts”[5] “n_cells_by_counts”       “pct_dropout_by_counts”[7] “total_counts”            “log10_total_counts”绘制QC统计数据\n高变基因：\n让我们看看表达的前50位基因是什么。\nplotHighestExprs(sce, exprs_values &#x3D; “counts”)\n如图所示，MALAT1对应平均约4％的计数。在某些细胞中，一些count高达〜30％，可以考虑在进一步分析和聚类之前删除该基因。而且，线粒体基因在总计数中所占比例很高的话也应该被删除。\n累积表达：\n绘制文库大小的相对比例，该比例由每个细胞的最高表达基因（默认为500个基因）表示。这可以帮助我们寻找样本之间表达分布的差异。\n绘制每个样本plotScater(sce, block1 &#x3D; “ident”, nfeatures &#x3D; 1000)\n绘制基因统计：\n函数plotRowData可以绘制rowData中的任何统计信息，例如均值表达与检测到的细胞数。\nplotRowData(sce, x &#x3D; “n_cells_by_counts”, y &#x3D; “mean_counts”)\n绘制细胞统计：\n用函数plotColData可以以相同的方式绘制细胞的任何qc统计分析。\np1 &lt;- plotColData(sce, x &#x3D; “total_counts”,    y &#x3D; “total_features_by_counts”, colour_by &#x3D; “ident”)p2 &lt;- plotColData(sce, x &#x3D; “pct_counts_feature_control”,    y &#x3D; “total_features_by_counts”, colour_by &#x3D; “ident”)p3 &lt;- plotColData(sce, x &#x3D; “pct_counts_feature_control”,    y &#x3D; “pct_counts_in_top_50_features”, colour_by &#x3D; “ident”)multiplot(p1, p2, p3, cols &#x3D; 2)\n识别QC统计数据中的异常值\n识别低质量细胞的方法是在所有qc-stats上运行PCA，然后在PCA空间中识别异常值（Hemberg-lab单细胞转录组数据分析（十一）- Scater单细胞表达谱PCA可视化）。\nsce &lt;- runPCA(sce, use_coldata &#x3D; TRUE,    detect_outliers &#x3D; TRUE)\nsROC 0.1-2 loadedplotReducedDim(sce, use_dimred&#x3D;”PCA_coldata”, colour_by &#x3D; “ident”)\n＃检查是否有异常值table(colData(sce)$outlier)##\nFALSE2531在这种情况下，我们已经滤除了低质量的细胞，并且在QC PCA中未检测到任何异常值。\n降维建议先用runPCA、runTSNE等函数做降维（ 用了这么多年的PCA可视化竟然是错的！！！），以便将它们存储在SCE对象中，这样就不必在每次绘制时都重新运行这些函数。除了这两个函数可以做降维，还有其他函数，比如plotPCA和plotTSNE等等，或者使用函数plotReducedDim并指定use_dimred &#x3D; “pca”等类似的。\n在使用任意组分信息进行降维之前，请记住要设置随机种子，以便可以完全复现。\n使用前1000高变基因运行PCAsce &lt;- runPCA(sce, ntop &#x3D; 1000, exprs_values &#x3D; “logcounts”, ncomponents &#x3D; 20)\nPCA - with different coloring, first 4 componentsfirst by sampleplotPCA(sce,ncomponents&#x3D;4,colour_by&#x3D;”ident”)\nthen by CelltypeplotPCA(sce,ncomponents&#x3D;4,colour_by&#x3D;”percent.mito”)\nDiffusion map, 运行需要安装destiny包!set.seed(1)sce &lt;- runDiffusionMap(sce, ntop &#x3D; 1000, ncomponents &#x3D; 4)plotDiffusionMap(sce, colour_by&#x3D;”ident”,ncomponents&#x3D;4)\ntSNE -使用tSNE对降维数据进行可视化，top10PCsset.seed(1)sce &lt;- runTSNE(sce, ntop &#x3D; 1000, ncomponents &#x3D; 2, perplexity &#x3D; 30, n_dimred &#x3D; 10)plotTSNE(sce, colour_by&#x3D;”ident”)\nUMAP,需要安装umap包!set.seed(1)sce &lt;- runUMAP(sce)plotUMAP(object &#x3D; sce, colour_by&#x3D;”ident”)\n影响因子的可视化我们可以使用plotExplanatoryVariables函数调查不同因素的相对重要性。当拟合线性模型将每个基因的表达值与该因子进行回归时，我们计算colData（sce）中每个因子的可决系数R2来评估拟合优度，最好是在对数表达式值上执行此操作，可以减少平均值对方差的影响——因此，我们首先要进行归一化（取log值）。\n下面的可视化计算耗时较长。\nplotExplanatoryVariables(sce, variables &#x3D;  c(“ident”,”Chemistry”,”pct_counts_mito”, “total_features_by_counts”, “pct_counts_in_top_50_features”, “pct_counts_in_top_500_features”, “total_counts”, “S.Score”,”G2M.Score”))\n每条线对应一个因子在所有基因中R平方值的分布。\n我们要留意确定与QC或Metadata密切相关的PC。默认情况下是绘制前10个因子，但是在这里我们仅可视化一些特定因子。\nfor total_featuresplotExplanatoryPCs(sce, variables &#x3D; c(“ident”, “Chemistry”,”pct_counts_mito”, “total_features_by_counts”, “pct_counts_in_top_50_features”, “total_counts”,”S.Score”,”G2M.Score”), npcs_to_plot &#x3D; 20)\n问题：您认为可以在数据中看到一些明显的批次效应吗？您是否认为存在技术偏差？\nPC1显然与数据的分布相关，例如前50-500个高度表达的基因在计数的比例，以及占total_features&#x2F;total_counts的比例。这是scRNAseq数据中的常见问题，可能是技术偏差，或者是大小和转录方式非常不同的细胞类型的生物学特征造成的。\n同样很明显的是，许多topPC（尤其是PC 2、3、6、7、8）在很大程度上是由不同的样本（ident）或仅由v2 vs v3（Chemistry）来解释的。\nSession infosessionInfo()\nR version 3.5.1 (2018-07-02)Platform: x86_64-apple-darwin13.4.0 (64-bit)Running under: macOS  10.14.4Matrix products: defaultBLAS&#x2F;LAPACK: &#x2F;Users&#x2F;asbj&#x2F;Programs&#x2F;miniconda3_4.2.12&#x2F;envs&#x2F;elixir-course&#x2F;lib&#x2F;R&#x2F;lib&#x2F;libRblas.dyliblocale:[1] en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;C&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8attached base packages:[1] parallel  stats4    stats     graphics  grDevices utils     datasets[8] methods   baseother attached packages:[1] Matrix_1.2-17               scater_1.10.1[3] ggplot2_3.1.1               SingleCellExperiment_1.4.0[5] SummarizedExperiment_1.12.0 DelayedArray_0.8.0[7] BiocParallel_1.16.6         matrixStats_0.54.0[9] Biobase_2.42.0              GenomicRanges_1.34.0[11] GenomeInfoDb_1.18.1         IRanges_2.16.0[13] S4Vectors_0.20.1            BiocGenerics_0.28.0[15] Seurat_3.0.0loaded via a namespace (and not attached):[1] reticulate_1.12          R.utils_2.8.0[3] tidyselect_0.2.5         htmlwidgets_1.3[5] grid_3.5.1               trimcluster_0.1-2.1[7] ranger_0.11.1            Rtsne_0.15[9] munsell_0.5.0            destiny_2.12.0[11] codetools_0.2-16         ica_1.0-2[13] umap_0.2.0.0             future_1.12.0[15] sROC_0.1-2               withr_2.1.2[17] colorspace_1.4-1         knitr_1.20[19] ROCR_1.0-7               robustbase_0.93-2[21] vcd_1.4-4                VIM_4.8.0[23] TTR_0.23-4               gbRd_0.4-11[25] listenv_0.7.0            Rdpack_0.10-1[27] labeling_0.3             GenomeInfoDbData_1.2.1[29] cvTools_0.3.2            bit64_0.9-7[31] rhdf5_2.26.2             ggthemes_4.1.1[33] diptest_0.75-7           R6_2.4.0[35] ggbeeswarm_0.6.0         robCompositions_2.0.10[37] rsvd_1.0.0               RcppEigen_0.3.3.5.0[39] mvoutlier_2.0.9          hdf5r_1.2.0[41] flexmix_2.3-15           bitops_1.0-6[43] reshape_0.8.8            assertthat_0.2.1[45] SDMTools_1.1-221.1       scales_1.0.0[47] nnet_7.3-12              beeswarm_0.2.3[49] gtable_0.3.0             npsurv_0.4-0[51] globals_0.12.4           rlang_0.3.4[53] scatterplot3d_0.3-41     splines_3.5.1[55] lazyeval_0.2.2           yaml_2.2.0[57] reshape2_1.4.3           abind_1.4-5[59] tools_3.5.1              zCompositions_1.2.0[61] gplots_3.0.1.1           RColorBrewer_1.1-2[63] proxy_0.4-22             ggridges_0.5.1[65] Rcpp_1.0.1               plyr_1.8.4[67] zlibbioc_1.28.0          purrr_0.3.2[69] RCurl_1.95-4.12          pbapply_1.4-0[71] viridis_0.5.1            cowplot_0.9.4[73] zoo_1.8-5                haven_2.1.0[75] ggrepel_0.8.1            cluster_2.0.9[77] magrittr_1.5             RSpectra_0.14-0[79] data.table_1.12.2        openxlsx_4.1.0[81] lmtest_0.9-36            RANN_2.6[83] truncnorm_1.0-8          mvtnorm_1.0-10[85] fitdistrplus_1.0-14      hms_0.4.2[87] lsei_1.2-0               evaluate_0.13[89] smoother_1.1             rio_0.5.16[91] mclust_5.4.3             readxl_1.3.1[93] gridExtra_2.3            compiler_3.5.1[95] tibble_2.1.1             KernSmooth_2.23-15[97] crayon_1.3.4             R.oo_1.22.0[99] htmltools_0.3.6          pcaPP_1.9-73[101] tidyr_0.8.3              rrcov_1.4-4[103] MASS_7.3-51.4            fpc_2.1-11.1[105] boot_1.3-22              car_3.0-2[107] sgeostat_1.0-27          R.methodsS3_1.7.1[109] gdata_2.18.0             metap_1.1[111] igraph_1.2.4.1           forcats_0.4.0[113] pkgconfig_2.0.2          foreign_0.8-71[115] laeken_0.5.0             sp_1.3-1[117] plotly_4.8.0             vipor_0.4.5[119] XVector_0.22.0           bibtex_0.4.2[121] NADA_1.6-1               stringr_1.4.0[123] digest_0.6.18            sctransform_0.2.0[125] pls_2.7-0                tsne_0.1-3[127] rmarkdown_1.11           cellranger_1.1.0[129] DelayedMatrixStats_1.4.0 curl_3.3[131] kernlab_0.9-27           gtools_3.8.1[133] modeltools_0.2-22        nlme_3.1-139[135] jsonlite_1.6             Rhdf5lib_1.4.3[137] carData_3.0-2            viridisLite_0.3.0[139] pillar_1.3.1             lattice_0.20-38[141] GGally_1.4.0             httr_1.4.0[143] DEoptimR_1.0-8           survival_2.44-1.1[145] xts_0.11-1               glue_1.3.1[147] zip_2.0.1                png_0.1-7[149] prabclus_2.2-7           bit_1.1-14[151] class_7.3-15             stringi_1.4.3[153] HDF5Array_1.10.1         caTools_1.17.1.2[155] dplyr_0.8.0.1            irlba_2.3.3[157] e1071_1.7-0              future.apply_1.1.0[159] ape_5.3","dateCreated":"2020-04-20T23:49:31+08:00","dateModified":"2020-04-20T23:49:31+08:00","datePublished":"2020-04-20T23:49:31+08:00","description":"代码分析 | Sing-cell Sequencing QC详解","headline":"代码分析 | Sing-cell Sequencing QC详解","image":[],"mainEntityOfPage":{"@type":"WebPage","@id":"https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"},"publisher":{"@type":"Organization","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg","logo":{"@type":"ImageObject","url":"coveriuu.jpg"}},"url":"https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"}</script>
    <meta name="description" content="代码分析 | Sing-cell Sequencing QC详解">
<meta property="og:type" content="blog">
<meta property="og:title" content="代码分析 | Sing-cell Sequencing QC详解">
<meta property="og:url" content="https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/index.html">
<meta property="og:site_name" content="iuu&#39;s blog">
<meta property="og:description" content="代码分析 | Sing-cell Sequencing QC详解">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2020-04-20T15:49:31.000Z">
<meta property="article:modified_time" content="2020-04-20T15:49:31.000Z">
<meta property="article:author" content="Dr.iuu">
<meta name="twitter:card" content="summary">
    
    
        
    
    
        <meta property="og:image" content="https://iuu19.github.io/assets/images/coveriuu.jpg"/>
    
    
    
    
    <!--STYLES-->
    
<link rel="stylesheet" href="/assets/css/style-gf8wcm29a1hig4pgasvkpmpwcvjyqc9ufzpbo404wwnc2qcamao8rlll1csq.min.css">

    <!--STYLES END-->
    

    

    
<link rel="stylesheet" href="/css/prism-vsc-dark-plus.css" type="text/css">
<link rel="stylesheet" href="/css/prism-line-numbers.css" type="text/css"></head>

    <body>
        <div id="blog">
            <!-- Define author's picture -->


    
        
            
        
    

<header id="header" data-behavior="4">
    <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
    <div class="header-title">
        <a
            class="header-title-link"
            href="/"
            aria-label=""
        >
            iuu&#39;s blog
        </a>
    </div>
    
        
            <a
                class="header-right-picture "
                href="/all-about"
                aria-label="打开链接: /all-about"
            >
        
        
            <img class="header-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
        </a>
    
</header>

            <!-- Define author's picture -->



        
    

<nav id="sidebar" data-behavior="4">
    <div class="sidebar-container">
        
            <div class="sidebar-profile">
                <a
                    href="/#about"
                    aria-label="阅读有关作者的更多信息"
                >
                    <img class="sidebar-profile-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
                </a>
                <h4 class="sidebar-profile-name">Dr.iuu</h4>
                
                    <h5 class="sidebar-profile-bio"><p>NCMC-SCMC-SWCH</p>
</h5>
                
            </div>
        
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://iuu19.github.io"
                            
                            rel="noopener"
                            title="首页"
                        >
                        <i class="sidebar-button-icon fa fa-home" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">首页</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-categories"
                            
                            rel="noopener"
                            title="分类"
                        >
                        <i class="sidebar-button-icon fa fa-bookmark" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">分类</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-tags"
                            
                            rel="noopener"
                            title="标签"
                        >
                        <i class="sidebar-button-icon fa fa-tags" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">标签</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-archives"
                            
                            rel="noopener"
                            title="归档"
                        >
                        <i class="sidebar-button-icon fa fa-archive" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">归档</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-about"
                            
                            rel="noopener"
                            title="关于"
                        >
                        <i class="sidebar-button-icon fa fa-question" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">关于</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://github.com/iuu19/iuu19.github.io"
                            
                                target="_blank"
                            
                            rel="noopener"
                            title="GitHub"
                        >
                        <i class="sidebar-button-icon fab fa-github" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">GitHub</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/atom.xml"
                            
                            rel="noopener"
                            title="RSS"
                        >
                        <i class="sidebar-button-icon fa fa-rss" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">RSS</span>
                    </a>
            </li>
            
        </ul>
        
    </div>
</nav>

            
            <div id="main" data-behavior="4"
                 class="
                        hasCoverMetaIn
                        ">
                
<article class="post">
    
    
        <div class="post-header main-content-wrap text-center">
    
        <h1 class="post-title">
            代码分析 | Sing-cell Sequencing QC详解
        </h1>
    
    
        <div class="post-meta">
    <time datetime="2020-04-20T23:49:31+08:00">
	
		    4月 20, 2020
    	
    </time>
    
</div>

    
</div>

    
    <div class="post-content markdown">
        <div class="main-content-wrap">
            <p>代码分析 | Sing-cell Sequencing QC详解</p>
<span id="more"></span>


<h1 id="table-of-contents">目录</h1><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%8A%A0%E8%BD%BD%E7%8E%AF%E5%A2%83"><span class="toc-text">加载环境</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E9%9B%86"><span class="toc-text">数据集</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%8A%A0%E8%BD%BDR%E5%8C%85"><span class="toc-text">加载R包</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#CITE-seq-data%E4%B8%AD%E5%8F%AA%E6%8F%90%E5%8F%96%E5%9F%BA%E5%9B%A0%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5"><span class="toc-text">CITE-seq data中只提取基因表达矩阵</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E4%BD%BF%E7%94%A8merge%E5%87%BD%E6%95%B0%E5%90%88%E5%B9%B6%E4%B8%BA%E4%B8%80%E4%B8%AAseurat-object-%E4%B8%BA%E4%BA%86%E9%98%B2%E6%AD%A2%E6%95%B0%E6%8D%AE%E9%9B%86%E4%B9%8B%E9%97%B4%E7%9A%84barcodes%E9%87%8D%E5%8F%A0%EF%BC%8C%E7%BB%99%E7%BB%86%E8%83%9E%E6%B7%BB%E5%8A%A0ID%E3%80%82"><span class="toc-text">使用merge函数合并为一个seurat object. 为了防止数据集之间的barcodes重叠，给细胞添加ID。</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%9C%A8metadata%E4%B8%AD%E6%B7%BB%E5%8A%A0v2%E5%92%8Cv3%E6%A0%87%E8%AF%86%E3%80%82"><span class="toc-text">在metadata中添加v2和v3标识。</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#An-object-of-class-Seurat"><span class="toc-text">An object of class Seurat</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#33538-features-across-2931-samples-within-1-assay"><span class="toc-text">33538 features across 2931 samples within 1 assay</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Active-assay-RNA-33538-features"><span class="toc-text">Active assay: RNA (33538 features)</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%A3%80%E6%9F%A5%E6%AF%8F%E4%B8%AA%E6%A0%B7%E6%9C%AC%E7%9A%84%E7%BB%86%E8%83%9E%E6%95%B0%E9%87%8F%EF%BC%8C%E5%B0%86%E5%85%B6%E5%AD%98%E5%82%A8%E5%9C%A8metadata%E7%9A%84orig-ident%E4%B8%AD%EF%BC%8C%E5%B9%B6%E8%87%AA%E5%8A%A8%E8%AE%BE%E7%BD%AE%E4%B8%BAactive-ident%E3%80%82"><span class="toc-text">检查每个样本的细胞数量，将其存储在metadata的orig.ident中，并自动设置为active ident。</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#p3-1k-v2-1k-v3-1k"><span class="toc-text">p3.1k v2.1k v3.1k</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#713-996-1222"><span class="toc-text">713   996  1222</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#orig-ident-nCount-RNA-nFeature-RNA-Chemistry"><span class="toc-text">orig.ident nCount_RNA nFeature_RNA Chemistry</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#v2-1k-AAACCTGAGCGCTCCA-1-v2-1k-6631-2029-v2"><span class="toc-text">v2.1k_AAACCTGAGCGCTCCA-1      v2.1k       6631         2029        v2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#v2-1k-AAACCTGGTGATAAAC-1-v2-1k-2196-881-v2"><span class="toc-text">v2.1k_AAACCTGGTGATAAAC-1      v2.1k       2196          881        v2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#v2-1k-AAACGGGGTTTGTGTG-1-v2-1k-2700-791-v2"><span class="toc-text">v2.1k_AAACGGGGTTTGTGTG-1      v2.1k       2700          791        v2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#v2-1k-AAAGATGAGTACTTGC-1-v2-1k-3551-1183-v2"><span class="toc-text">v2.1k_AAAGATGAGTACTTGC-1      v2.1k       3551         1183        v2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#v2-1k-AAAGCAAGTCTCTTAT-1-v2-1k-3080-1333-v2"><span class="toc-text">v2.1k_AAAGCAAGTCTCTTAT-1      v2.1k       3080         1333        v2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#v2-1k-AAAGCAATCCACGAAT-1-v2-1k-5769-1556-v2"><span class="toc-text">v2.1k_AAAGCAATCCACGAAT-1      v2.1k       5769         1556        v2</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E9%80%89%E6%8B%A9%E7%BA%BF%E7%B2%92%E4%BD%93%E5%9F%BA%E5%9B%A0%E7%99%BE%E5%88%86%E6%AF%94-lt-25-%E7%9A%84%E7%BB%86%E8%83%9E"><span class="toc-text">选择线粒体基因百分比&lt;25%的细胞</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-2703"><span class="toc-text">[1] 2703</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%AF%B9%E7%BB%86%E8%83%9E%E5%8F%96%E5%AD%90%E9%9B%86"><span class="toc-text">对细胞取子集</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%94%BB%E5%B0%8F%E6%8F%90%E7%90%B4%E5%9B%BE"><span class="toc-text">画小提琴图</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%BB%86%E8%83%9E%E5%8F%96%E5%AD%90%E9%9B%86"><span class="toc-text">细胞取子集</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%9F%A5%E7%9C%8B%E7%BB%86%E8%83%9E%E6%95%B0"><span class="toc-text">查看细胞数</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-2631"><span class="toc-text">[1] 2631</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%8E%BB%E9%99%A4%E7%BB%86%E8%83%9E"><span class="toc-text">去除细胞</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%9F%A5%E7%9C%8B%E7%BB%86%E8%83%9E%E6%95%B0-1"><span class="toc-text">查看细胞数</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-2531"><span class="toc-text">[1] 2531</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%9C%A8%E8%BF%87%E6%BB%A4%E5%89%8D%E5%90%8E%E6%A3%80%E6%9F%A5%E6%AF%8F%E4%B8%AA%E6%A0%B7%E6%9C%AC%E7%9A%84%E7%BB%86%E8%83%9E%E6%95%B0"><span class="toc-text">在过滤前后检查每个样本的细胞数</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#p3-1k-v2-1k-v3-1k-1"><span class="toc-text">p3.1k v2.1k v3.1k</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#713-996-1222-1"><span class="toc-text">713   996  1222</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#p3-1k-v2-1k-v3-1k-2"><span class="toc-text">p3.1k v2.1k v3.1k</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#526-933-1072"><span class="toc-text">526   933  1072</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E8%AE%A1%E7%AE%97QC%E6%8C%87%E6%A0%87"><span class="toc-text">计算QC指标</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#check-what-all-entries-are"><span class="toc-text">check what all entries are -</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-%E2%80%9Corig-ident%E2%80%9D"><span class="toc-text">[1] “orig.ident”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-%E2%80%9CnCount-RNA%E2%80%9D"><span class="toc-text">[2] “nCount_RNA”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-%E2%80%9CnFeature-RNA%E2%80%9D"><span class="toc-text">[3] “nFeature_RNA”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-%E2%80%9CChemistry%E2%80%9D"><span class="toc-text">[4] “Chemistry”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-%E2%80%9Cpercent-mito%E2%80%9D"><span class="toc-text">[5] “percent.mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#6-%E2%80%9Cpercent-ribo%E2%80%9D"><span class="toc-text">[6] “percent.ribo”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#7-%E2%80%9CS-Score%E2%80%9D"><span class="toc-text">[7] “S.Score”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#8-%E2%80%9CG2M-Score%E2%80%9D"><span class="toc-text">[8] “G2M.Score”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#9-%E2%80%9CPhase%E2%80%9D"><span class="toc-text">[9] “Phase”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#10-%E2%80%9Cident%E2%80%9D"><span class="toc-text">[10] “ident”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#11-%E2%80%9Cis-cell-control%E2%80%9D"><span class="toc-text">[11] “is_cell_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#12-%E2%80%9Ctotal-features-by-counts%E2%80%9D"><span class="toc-text">[12] “total_features_by_counts”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#13-%E2%80%9Clog10-total-features-by-counts%E2%80%9D"><span class="toc-text">[13] “log10_total_features_by_counts”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#14-%E2%80%9Ctotal-counts%E2%80%9D"><span class="toc-text">[14] “total_counts”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#15-%E2%80%9Clog10-total-counts%E2%80%9D"><span class="toc-text">[15] “log10_total_counts”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#16-%E2%80%9Cpct-counts-in-top-50-features%E2%80%9D"><span class="toc-text">[16] “pct_counts_in_top_50_features”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#17-%E2%80%9Cpct-counts-in-top-100-features%E2%80%9D"><span class="toc-text">[17] “pct_counts_in_top_100_features”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#18-%E2%80%9Cpct-counts-in-top-200-features%E2%80%9D"><span class="toc-text">[18] “pct_counts_in_top_200_features”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#19-%E2%80%9Cpct-counts-in-top-500-features%E2%80%9D"><span class="toc-text">[19] “pct_counts_in_top_500_features”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#20-%E2%80%9Ctotal-features-by-counts-endogenous%E2%80%9D"><span class="toc-text">[20] “total_features_by_counts_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#21-%E2%80%9Clog10-total-features-by-counts-endogenous%E2%80%9D"><span class="toc-text">[21] “log10_total_features_by_counts_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#22-%E2%80%9Ctotal-counts-endogenous%E2%80%9D"><span class="toc-text">[22] “total_counts_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#23-%E2%80%9Clog10-total-counts-endogenous%E2%80%9D"><span class="toc-text">[23] “log10_total_counts_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#24-%E2%80%9Cpct-counts-endogenous%E2%80%9D"><span class="toc-text">[24] “pct_counts_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#25-%E2%80%9Cpct-counts-in-top-50-features-endogenous%E2%80%9D"><span class="toc-text">[25] “pct_counts_in_top_50_features_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#26-%E2%80%9Cpct-counts-in-top-100-features-endogenous%E2%80%9D"><span class="toc-text">[26] “pct_counts_in_top_100_features_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#27-%E2%80%9Cpct-counts-in-top-200-features-endogenous%E2%80%9D"><span class="toc-text">[27] “pct_counts_in_top_200_features_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#28-%E2%80%9Cpct-counts-in-top-500-features-endogenous%E2%80%9D"><span class="toc-text">[28] “pct_counts_in_top_500_features_endogenous”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#29-%E2%80%9Ctotal-features-by-counts-feature-control%E2%80%9D"><span class="toc-text">[29] “total_features_by_counts_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#30-%E2%80%9Clog10-total-features-by-counts-feature-control%E2%80%9D"><span class="toc-text">[30] “log10_total_features_by_counts_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#31-%E2%80%9Ctotal-counts-feature-control%E2%80%9D"><span class="toc-text">[31] “total_counts_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#32-%E2%80%9Clog10-total-counts-feature-control%E2%80%9D"><span class="toc-text">[32] “log10_total_counts_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#33-%E2%80%9Cpct-counts-feature-control%E2%80%9D"><span class="toc-text">[33] “pct_counts_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#34-%E2%80%9Cpct-counts-in-top-50-features-feature-control%E2%80%9D"><span class="toc-text">[34] “pct_counts_in_top_50_features_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#35-%E2%80%9Cpct-counts-in-top-100-features-feature-control%E2%80%9D"><span class="toc-text">[35] “pct_counts_in_top_100_features_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#36-%E2%80%9Cpct-counts-in-top-200-features-feature-control%E2%80%9D"><span class="toc-text">[36] “pct_counts_in_top_200_features_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#37-%E2%80%9Cpct-counts-in-top-500-features-feature-control%E2%80%9D"><span class="toc-text">[37] “pct_counts_in_top_500_features_feature_control”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#38-%E2%80%9Ctotal-features-by-counts-mito%E2%80%9D"><span class="toc-text">[38] “total_features_by_counts_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#39-%E2%80%9Clog10-total-features-by-counts-mito%E2%80%9D"><span class="toc-text">[39] “log10_total_features_by_counts_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#40-%E2%80%9Ctotal-counts-mito%E2%80%9D"><span class="toc-text">[40] “total_counts_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#41-%E2%80%9Clog10-total-counts-mito%E2%80%9D"><span class="toc-text">[41] “log10_total_counts_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#42-%E2%80%9Cpct-counts-mito%E2%80%9D"><span class="toc-text">[42] “pct_counts_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#43-%E2%80%9Cpct-counts-in-top-50-features-mito%E2%80%9D"><span class="toc-text">[43] “pct_counts_in_top_50_features_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#44-%E2%80%9Cpct-counts-in-top-100-features-mito%E2%80%9D"><span class="toc-text">[44] “pct_counts_in_top_100_features_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#45-%E2%80%9Cpct-counts-in-top-200-features-mito%E2%80%9D"><span class="toc-text">[45] “pct_counts_in_top_200_features_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#46-%E2%80%9Cpct-counts-in-top-500-features-mito%E2%80%9D"><span class="toc-text">[46] “pct_counts_in_top_500_features_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-%E2%80%9Cis-feature-control%E2%80%9D-%E2%80%9Cis-feature-control-mito%E2%80%9D"><span class="toc-text">[1] “is_feature_control”      “is_feature_control_mito”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-%E2%80%9Cmean-counts%E2%80%9D-%E2%80%9Clog10-mean-counts%E2%80%9D"><span class="toc-text">[3] “mean_counts”             “log10_mean_counts”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-%E2%80%9Cn-cells-by-counts%E2%80%9D-%E2%80%9Cpct-dropout-by-counts%E2%80%9D"><span class="toc-text">[5] “n_cells_by_counts”       “pct_dropout_by_counts”</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#7-%E2%80%9Ctotal-counts%E2%80%9D-%E2%80%9Clog10-total-counts%E2%80%9D"><span class="toc-text">[7] “total_counts”            “log10_total_counts”</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%BB%98%E5%88%B6%E6%AF%8F%E4%B8%AA%E6%A0%B7%E6%9C%AC"><span class="toc-text">绘制每个样本</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#sROC-0-1-2-loaded"><span class="toc-text">sROC 0.1-2 loaded</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#FALSE"><span class="toc-text">FALSE</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2531"><span class="toc-text">2531</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E4%BD%BF%E7%94%A8%E5%89%8D1000%E9%AB%98%E5%8F%98%E5%9F%BA%E5%9B%A0%E8%BF%90%E8%A1%8CPCA"><span class="toc-text">使用前1000高变基因运行PCA</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#PCA-with-different-coloring-first-4-components"><span class="toc-text">PCA - with different coloring, first 4 components</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#first-by-sample"><span class="toc-text">first by sample</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#then-by-Celltype"><span class="toc-text">then by Celltype</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Diffusion-map-%E8%BF%90%E8%A1%8C%E9%9C%80%E8%A6%81%E5%AE%89%E8%A3%85destiny%E5%8C%85"><span class="toc-text">Diffusion map, 运行需要安装destiny包!</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#tSNE-%E4%BD%BF%E7%94%A8tSNE%E5%AF%B9%E9%99%8D%E7%BB%B4%E6%95%B0%E6%8D%AE%E8%BF%9B%E8%A1%8C%E5%8F%AF%E8%A7%86%E5%8C%96%EF%BC%8Ctop10PCs"><span class="toc-text">tSNE -使用tSNE对降维数据进行可视化，top10PCs</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#UMAP-%E9%9C%80%E8%A6%81%E5%AE%89%E8%A3%85umap%E5%8C%85"><span class="toc-text">UMAP,需要安装umap包!</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#for-total-features"><span class="toc-text">for total_features</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#R-version-3-5-1-2018-07-02"><span class="toc-text">R version 3.5.1 (2018-07-02)</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Platform-x86-64-apple-darwin13-4-0-64-bit"><span class="toc-text">Platform: x86_64-apple-darwin13.4.0 (64-bit)</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Running-under-macOS-10-14-4"><span class="toc-text">Running under: macOS  10.14.4</span></a></li><li class="toc-item toc-level-2"><a class="toc-link"><span class="toc-text"></span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Matrix-products-default"><span class="toc-text">Matrix products: default</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#BLAS-x2F-LAPACK-x2F-Users-x2F-asbj-x2F-Programs-x2F-miniconda3-4-2-12-x2F-envs-x2F-elixir-course-x2F-lib-x2F-R-x2F-lib-x2F-libRblas-dylib"><span class="toc-text">BLAS&#x2F;LAPACK: &#x2F;Users&#x2F;asbj&#x2F;Programs&#x2F;miniconda3_4.2.12&#x2F;envs&#x2F;elixir-course&#x2F;lib&#x2F;R&#x2F;lib&#x2F;libRblas.dylib</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#-1"><span class="toc-text"></span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#locale"><span class="toc-text">locale:</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-en-US-UTF-8-x2F-en-US-UTF-8-x2F-en-US-UTF-8-x2F-C-x2F-en-US-UTF-8-x2F-en-US-UTF-8"><span class="toc-text">[1] en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;C&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#-2"><span class="toc-text"></span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#attached-base-packages"><span class="toc-text">attached base packages:</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-parallel-stats4-stats-graphics-grDevices-utils-datasets"><span class="toc-text">[1] parallel  stats4    stats     graphics  grDevices utils     datasets</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#8-methods-base"><span class="toc-text">[8] methods   base</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#-3"><span class="toc-text"></span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#other-attached-packages"><span class="toc-text">other attached packages:</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-Matrix-1-2-17-scater-1-10-1"><span class="toc-text">[1] Matrix_1.2-17               scater_1.10.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-ggplot2-3-1-1-SingleCellExperiment-1-4-0"><span class="toc-text">[3] ggplot2_3.1.1               SingleCellExperiment_1.4.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-SummarizedExperiment-1-12-0-DelayedArray-0-8-0"><span class="toc-text">[5] SummarizedExperiment_1.12.0 DelayedArray_0.8.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#7-BiocParallel-1-16-6-matrixStats-0-54-0"><span class="toc-text">[7] BiocParallel_1.16.6         matrixStats_0.54.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#9-Biobase-2-42-0-GenomicRanges-1-34-0"><span class="toc-text">[9] Biobase_2.42.0              GenomicRanges_1.34.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#11-GenomeInfoDb-1-18-1-IRanges-2-16-0"><span class="toc-text">[11] GenomeInfoDb_1.18.1         IRanges_2.16.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#13-S4Vectors-0-20-1-BiocGenerics-0-28-0"><span class="toc-text">[13] S4Vectors_0.20.1            BiocGenerics_0.28.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#15-Seurat-3-0-0"><span class="toc-text">[15] Seurat_3.0.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#-4"><span class="toc-text"></span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#loaded-via-a-namespace-and-not-attached"><span class="toc-text">loaded via a namespace (and not attached):</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-reticulate-1-12-R-utils-2-8-0"><span class="toc-text">[1] reticulate_1.12          R.utils_2.8.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-tidyselect-0-2-5-htmlwidgets-1-3"><span class="toc-text">[3] tidyselect_0.2.5         htmlwidgets_1.3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-grid-3-5-1-trimcluster-0-1-2-1"><span class="toc-text">[5] grid_3.5.1               trimcluster_0.1-2.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#7-ranger-0-11-1-Rtsne-0-15"><span class="toc-text">[7] ranger_0.11.1            Rtsne_0.15</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#9-munsell-0-5-0-destiny-2-12-0"><span class="toc-text">[9] munsell_0.5.0            destiny_2.12.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#11-codetools-0-2-16-ica-1-0-2"><span class="toc-text">[11] codetools_0.2-16         ica_1.0-2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#13-umap-0-2-0-0-future-1-12-0"><span class="toc-text">[13] umap_0.2.0.0             future_1.12.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#15-sROC-0-1-2-withr-2-1-2"><span class="toc-text">[15] sROC_0.1-2               withr_2.1.2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#17-colorspace-1-4-1-knitr-1-20"><span class="toc-text">[17] colorspace_1.4-1         knitr_1.20</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#19-ROCR-1-0-7-robustbase-0-93-2"><span class="toc-text">[19] ROCR_1.0-7               robustbase_0.93-2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#21-vcd-1-4-4-VIM-4-8-0"><span class="toc-text">[21] vcd_1.4-4                VIM_4.8.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#23-TTR-0-23-4-gbRd-0-4-11"><span class="toc-text">[23] TTR_0.23-4               gbRd_0.4-11</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#25-listenv-0-7-0-Rdpack-0-10-1"><span class="toc-text">[25] listenv_0.7.0            Rdpack_0.10-1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#27-labeling-0-3-GenomeInfoDbData-1-2-1"><span class="toc-text">[27] labeling_0.3             GenomeInfoDbData_1.2.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#29-cvTools-0-3-2-bit64-0-9-7"><span class="toc-text">[29] cvTools_0.3.2            bit64_0.9-7</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#31-rhdf5-2-26-2-ggthemes-4-1-1"><span class="toc-text">[31] rhdf5_2.26.2             ggthemes_4.1.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#33-diptest-0-75-7-R6-2-4-0"><span class="toc-text">[33] diptest_0.75-7           R6_2.4.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#35-ggbeeswarm-0-6-0-robCompositions-2-0-10"><span class="toc-text">[35] ggbeeswarm_0.6.0         robCompositions_2.0.10</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#37-rsvd-1-0-0-RcppEigen-0-3-3-5-0"><span class="toc-text">[37] rsvd_1.0.0               RcppEigen_0.3.3.5.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#39-mvoutlier-2-0-9-hdf5r-1-2-0"><span class="toc-text">[39] mvoutlier_2.0.9          hdf5r_1.2.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#41-flexmix-2-3-15-bitops-1-0-6"><span class="toc-text">[41] flexmix_2.3-15           bitops_1.0-6</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#43-reshape-0-8-8-assertthat-0-2-1"><span class="toc-text">[43] reshape_0.8.8            assertthat_0.2.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#45-SDMTools-1-1-221-1-scales-1-0-0"><span class="toc-text">[45] SDMTools_1.1-221.1       scales_1.0.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#47-nnet-7-3-12-beeswarm-0-2-3"><span class="toc-text">[47] nnet_7.3-12              beeswarm_0.2.3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#49-gtable-0-3-0-npsurv-0-4-0"><span class="toc-text">[49] gtable_0.3.0             npsurv_0.4-0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#51-globals-0-12-4-rlang-0-3-4"><span class="toc-text">[51] globals_0.12.4           rlang_0.3.4</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#53-scatterplot3d-0-3-41-splines-3-5-1"><span class="toc-text">[53] scatterplot3d_0.3-41     splines_3.5.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#55-lazyeval-0-2-2-yaml-2-2-0"><span class="toc-text">[55] lazyeval_0.2.2           yaml_2.2.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#57-reshape2-1-4-3-abind-1-4-5"><span class="toc-text">[57] reshape2_1.4.3           abind_1.4-5</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#59-tools-3-5-1-zCompositions-1-2-0"><span class="toc-text">[59] tools_3.5.1              zCompositions_1.2.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#61-gplots-3-0-1-1-RColorBrewer-1-1-2"><span class="toc-text">[61] gplots_3.0.1.1           RColorBrewer_1.1-2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#63-proxy-0-4-22-ggridges-0-5-1"><span class="toc-text">[63] proxy_0.4-22             ggridges_0.5.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#65-Rcpp-1-0-1-plyr-1-8-4"><span class="toc-text">[65] Rcpp_1.0.1               plyr_1.8.4</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#67-zlibbioc-1-28-0-purrr-0-3-2"><span class="toc-text">[67] zlibbioc_1.28.0          purrr_0.3.2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#69-RCurl-1-95-4-12-pbapply-1-4-0"><span class="toc-text">[69] RCurl_1.95-4.12          pbapply_1.4-0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#71-viridis-0-5-1-cowplot-0-9-4"><span class="toc-text">[71] viridis_0.5.1            cowplot_0.9.4</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#73-zoo-1-8-5-haven-2-1-0"><span class="toc-text">[73] zoo_1.8-5                haven_2.1.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#75-ggrepel-0-8-1-cluster-2-0-9"><span class="toc-text">[75] ggrepel_0.8.1            cluster_2.0.9</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#77-magrittr-1-5-RSpectra-0-14-0"><span class="toc-text">[77] magrittr_1.5             RSpectra_0.14-0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#79-data-table-1-12-2-openxlsx-4-1-0"><span class="toc-text">[79] data.table_1.12.2        openxlsx_4.1.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#81-lmtest-0-9-36-RANN-2-6"><span class="toc-text">[81] lmtest_0.9-36            RANN_2.6</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#83-truncnorm-1-0-8-mvtnorm-1-0-10"><span class="toc-text">[83] truncnorm_1.0-8          mvtnorm_1.0-10</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#85-fitdistrplus-1-0-14-hms-0-4-2"><span class="toc-text">[85] fitdistrplus_1.0-14      hms_0.4.2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#87-lsei-1-2-0-evaluate-0-13"><span class="toc-text">[87] lsei_1.2-0               evaluate_0.13</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#89-smoother-1-1-rio-0-5-16"><span class="toc-text">[89] smoother_1.1             rio_0.5.16</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#91-mclust-5-4-3-readxl-1-3-1"><span class="toc-text">[91] mclust_5.4.3             readxl_1.3.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#93-gridExtra-2-3-compiler-3-5-1"><span class="toc-text">[93] gridExtra_2.3            compiler_3.5.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#95-tibble-2-1-1-KernSmooth-2-23-15"><span class="toc-text">[95] tibble_2.1.1             KernSmooth_2.23-15</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#97-crayon-1-3-4-R-oo-1-22-0"><span class="toc-text">[97] crayon_1.3.4             R.oo_1.22.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#99-htmltools-0-3-6-pcaPP-1-9-73"><span class="toc-text">[99] htmltools_0.3.6          pcaPP_1.9-73</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#101-tidyr-0-8-3-rrcov-1-4-4"><span class="toc-text">[101] tidyr_0.8.3              rrcov_1.4-4</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#103-MASS-7-3-51-4-fpc-2-1-11-1"><span class="toc-text">[103] MASS_7.3-51.4            fpc_2.1-11.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#105-boot-1-3-22-car-3-0-2"><span class="toc-text">[105] boot_1.3-22              car_3.0-2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#107-sgeostat-1-0-27-R-methodsS3-1-7-1"><span class="toc-text">[107] sgeostat_1.0-27          R.methodsS3_1.7.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#109-gdata-2-18-0-metap-1-1"><span class="toc-text">[109] gdata_2.18.0             metap_1.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#111-igraph-1-2-4-1-forcats-0-4-0"><span class="toc-text">[111] igraph_1.2.4.1           forcats_0.4.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#113-pkgconfig-2-0-2-foreign-0-8-71"><span class="toc-text">[113] pkgconfig_2.0.2          foreign_0.8-71</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#115-laeken-0-5-0-sp-1-3-1"><span class="toc-text">[115] laeken_0.5.0             sp_1.3-1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#117-plotly-4-8-0-vipor-0-4-5"><span class="toc-text">[117] plotly_4.8.0             vipor_0.4.5</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#119-XVector-0-22-0-bibtex-0-4-2"><span class="toc-text">[119] XVector_0.22.0           bibtex_0.4.2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#121-NADA-1-6-1-stringr-1-4-0"><span class="toc-text">[121] NADA_1.6-1               stringr_1.4.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#123-digest-0-6-18-sctransform-0-2-0"><span class="toc-text">[123] digest_0.6.18            sctransform_0.2.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#125-pls-2-7-0-tsne-0-1-3"><span class="toc-text">[125] pls_2.7-0                tsne_0.1-3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#127-rmarkdown-1-11-cellranger-1-1-0"><span class="toc-text">[127] rmarkdown_1.11           cellranger_1.1.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#129-DelayedMatrixStats-1-4-0-curl-3-3"><span class="toc-text">[129] DelayedMatrixStats_1.4.0 curl_3.3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#131-kernlab-0-9-27-gtools-3-8-1"><span class="toc-text">[131] kernlab_0.9-27           gtools_3.8.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#133-modeltools-0-2-22-nlme-3-1-139"><span class="toc-text">[133] modeltools_0.2-22        nlme_3.1-139</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#135-jsonlite-1-6-Rhdf5lib-1-4-3"><span class="toc-text">[135] jsonlite_1.6             Rhdf5lib_1.4.3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#137-carData-3-0-2-viridisLite-0-3-0"><span class="toc-text">[137] carData_3.0-2            viridisLite_0.3.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#139-pillar-1-3-1-lattice-0-20-38"><span class="toc-text">[139] pillar_1.3.1             lattice_0.20-38</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#141-GGally-1-4-0-httr-1-4-0"><span class="toc-text">[141] GGally_1.4.0             httr_1.4.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#143-DEoptimR-1-0-8-survival-2-44-1-1"><span class="toc-text">[143] DEoptimR_1.0-8           survival_2.44-1.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#145-xts-0-11-1-glue-1-3-1"><span class="toc-text">[145] xts_0.11-1               glue_1.3.1</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#147-zip-2-0-1-png-0-1-7"><span class="toc-text">[147] zip_2.0.1                png_0.1-7</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#149-prabclus-2-2-7-bit-1-1-14"><span class="toc-text">[149] prabclus_2.2-7           bit_1.1-14</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#151-class-7-3-15-stringi-1-4-3"><span class="toc-text">[151] class_7.3-15             stringi_1.4.3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#153-HDF5Array-1-10-1-caTools-1-17-1-2"><span class="toc-text">[153] HDF5Array_1.10.1         caTools_1.17.1.2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#155-dplyr-0-8-0-1-irlba-2-3-3"><span class="toc-text">[155] dplyr_0.8.0.1            irlba_2.3.3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#157-e1071-1-7-0-future-apply-1-1-0"><span class="toc-text">[157] e1071_1.7-0              future.apply_1.1.0</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#159-ape-5-3"><span class="toc-text">[159] ape_5.3</span></a></li></ol></li></ol>
<hr>
<font face="黑体" color=black size=4>

<p>最近找到了芬兰CSC-IT科学中心主讲的生物信息课程(<a target="_blank" rel="noopener" href="https://www.csc.fi/web/training/-/scrnaseq)%EF%BC%8C">https://www.csc.fi/web/training/-/scrnaseq)，</a><br>官网上还提供了练习素材以及详细代码，今天就来练习一下单细胞数据QC的过程：</p>
<p>第一部分是使用Seurat（<a target="_blank" rel="noopener" href="https://satijalab.org/seurat/%EF%BC%89%E6%9D%A5%E5%AF%B9QC%E8%BF%9B%E8%A1%8C%E5%8F%AF%E8%A7%86%E5%8C%96%E5%B9%B6%E8%BF%9B%E8%A1%8C%E7%BB%86%E8%83%9E%E8%BF%87%E6%BB%A4%EF%BC%9B">https://satijalab.org/seurat/）来对QC进行可视化并进行细胞过滤；</a><br>第二部分将详细探讨scater软件包（<a target="_blank" rel="noopener" href="https://bioconductor.org/packages/release/bioc/html/scater.html%EF%BC%89%E3%80%82">https://bioconductor.org/packages/release/bioc/html/scater.html）。</a></p>
<font face="黑体" color=darkgreen size=3>

<h1 id="加载环境"><a href="#加载环境" class="headerlink" title="加载环境"></a>加载环境</h1></font>


<p>如果需要加载conda环境（软件安装不上，可能是网速慢！Conda&#x2F;R&#x2F;pip&#x2F;brew等国内镜像大全拿走不谢~~），请在运行以下代码之前遵循以下说明：<br><a target="_blank" rel="noopener" href="https://nbisweden.github.io/excelerate-scRNAseq/computing_environment_instructions.html%E3%80%82">https://nbisweden.github.io/excelerate-scRNAseq/computing_environment_instructions.html。</a></p>
<font face="黑体" color=darkgreen size=3>

<h1 id="数据集"><a href="#数据集" class="headerlink" title="数据集"></a>数据集</h1></font>

<p>在本教程中，我们将使用10x Genomics网站上的3个不同的PBMC数据集:<br><a target="_blank" rel="noopener" href="https://support.10xgenomics.com/single-cell-gene-expression/datasets">https://support.10xgenomics.com/single-cell-gene-expression/datasets</a></p>
<p>包括：</p>
<ul>
<li>1k PBMCs using 10x v2 chemistry</li>
<li>1k PBMCs using 10x v3 chemistry</li>
<li>1k PBMCs using 10x v3 chemistry in combination with cell surface proteins, but disregarding the protein data and only looking at gene expression.</li>
</ul>
<p>我们可以通过以下命令进行下载：</p>
<pre><code>mkdir data  #建立data文件夹
cd data
curl -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v2/pbmc_1k_v2_filtered_feature_bc_matrix.h5
curl -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v3/pbmc_1k_v3_filtered_feature_bc_matrix.h5
curl -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_10k_v3/pbmc_10k_v3_filtered_feature_bc_matrix.h5
</code></pre>
<font face="黑体" color=darkgreen size=3>

<h1 id="加载R包"><a href="#加载R包" class="headerlink" title="加载R包"></a>加载R包</h1></font>

<p>suppressMessages(require(Seurat))<br>suppressMessages(require(scater))<br>suppressMessages(require(Matrix))<br>读入数据<br>v3.1k &lt;- Read10X_h5(“<del>&#x2F;scrna-seq2019&#x2F;day1&#x2F;1-qc&#x2F;pbmc_1k_v3_filtered_feature_bc_matrix.h5”, use.names &#x3D; T)<br>v2.1k &lt;- Read10X_h5(“</del>&#x2F;scrna-seq2019&#x2F;day1&#x2F;1-qc&#x2F;pbmc_1k_v2_filtered_feature_bc_matrix.h5”, use.names &#x3D; T)<br>p3.1k &lt;- Read10X_h5(“~&#x2F;scrna-seq2019&#x2F;day1&#x2F;1-qc&#x2F;pbmc_1k_protein_v3_filtered_feature_bc_matrix.h5”, use.names &#x3D; T)</p>
<h1 id="CITE-seq-data中只提取基因表达矩阵"><a href="#CITE-seq-data中只提取基因表达矩阵" class="headerlink" title="CITE-seq data中只提取基因表达矩阵"></a>CITE-seq data中只提取基因表达矩阵</h1><p>p3.1k &lt;- p3.1k$<code>Gene Expression</code><br>Hemberg-lab单细胞转录组数据分析（七）- 导入10X和SmartSeq2数据Tabula Muris。<br>Seurat<br>构建Seurat对象</p>
<p>首先，为每个数据集创建Seurat对象，然后合并为一个大的seurat对象。</p>
<p>sdata.v2.1k &lt;- CreateSeuratObject(v2.1k, project &#x3D; “v2.1k”)<br>sdata.v3.1k &lt;- CreateSeuratObject(v3.1k, project &#x3D; “v3.1k”)<br>sdata.p3.1k &lt;- CreateSeuratObject(p3.1k, project &#x3D; “p3.1k”)</p>
<h1 id="使用merge函数合并为一个seurat-object-为了防止数据集之间的barcodes重叠，给细胞添加ID。"><a href="#使用merge函数合并为一个seurat-object-为了防止数据集之间的barcodes重叠，给细胞添加ID。" class="headerlink" title="使用merge函数合并为一个seurat object. 为了防止数据集之间的barcodes重叠，给细胞添加ID。"></a>使用merge函数合并为一个seurat object. 为了防止数据集之间的barcodes重叠，给细胞添加ID。</h1><p>alldata &lt;- merge(sdata.v2.1k, c(sdata.v3.1k,sdata.p3.1k), add.cell.ids&#x3D;c(“v2.1k”,”v3.1k”,”p3.1k”))</p>
<h1 id="在metadata中添加v2和v3标识。"><a href="#在metadata中添加v2和v3标识。" class="headerlink" title="在metadata中添加v2和v3标识。"></a>在metadata中添加v2和v3标识。</h1><p>chemistry &lt;- rep(“v3”,ncol(alldata))<br>chemistry[Idents(alldata) &#x3D;&#x3D; “v2.1k”] &lt;- “v2”<br>alldata &lt;- AddMetaData(alldata, chemistry, col.name &#x3D; “Chemistry”)<br>alldata</p>
<h2 id="An-object-of-class-Seurat"><a href="#An-object-of-class-Seurat" class="headerlink" title="An object of class Seurat"></a>An object of class Seurat</h2><h2 id="33538-features-across-2931-samples-within-1-assay"><a href="#33538-features-across-2931-samples-within-1-assay" class="headerlink" title="33538 features across 2931 samples within 1 assay"></a>33538 features across 2931 samples within 1 assay</h2><h2 id="Active-assay-RNA-33538-features"><a href="#Active-assay-RNA-33538-features" class="headerlink" title="Active assay: RNA (33538 features)"></a>Active assay: RNA (33538 features)</h2><h1 id="检查每个样本的细胞数量，将其存储在metadata的orig-ident中，并自动设置为active-ident。"><a href="#检查每个样本的细胞数量，将其存储在metadata的orig-ident中，并自动设置为active-ident。" class="headerlink" title="检查每个样本的细胞数量，将其存储在metadata的orig.ident中，并自动设置为active ident。"></a>检查每个样本的细胞数量，将其存储在metadata的orig.ident中，并自动设置为active ident。</h1><p>table(Idents(alldata))<br>##</p>
<h2 id="p3-1k-v2-1k-v3-1k"><a href="#p3-1k-v2-1k-v3-1k" class="headerlink" title="p3.1k v2.1k v3.1k"></a>p3.1k v2.1k v3.1k</h2><h2 id="713-996-1222"><a href="#713-996-1222" class="headerlink" title="713   996  1222"></a>713   996  1222</h2><p>计算线粒体比例</p>
<p>Seurat可以自动计算一些QC统计信息，例如每个细胞的UMI数量和基因数分别存储在metadata的nCount_RNA和nFeature_RNA列（Hemberg-lab单细胞转录组数据分析（六）- 构建表达矩阵，UMI介绍）。</p>
<p>head(<a href="mailto:&#x61;&#x6c;&#108;&#x64;&#x61;&#x74;&#97;&#64;&#109;&#x65;&#x74;&#97;&#46;&#100;&#97;&#x74;&#97;">&#x61;&#x6c;&#108;&#x64;&#x61;&#x74;&#97;&#64;&#109;&#x65;&#x74;&#97;&#46;&#100;&#97;&#x74;&#97;</a>)</p>
<h2 id="orig-ident-nCount-RNA-nFeature-RNA-Chemistry"><a href="#orig-ident-nCount-RNA-nFeature-RNA-Chemistry" class="headerlink" title="orig.ident nCount_RNA nFeature_RNA Chemistry"></a>orig.ident nCount_RNA nFeature_RNA Chemistry</h2><h2 id="v2-1k-AAACCTGAGCGCTCCA-1-v2-1k-6631-2029-v2"><a href="#v2-1k-AAACCTGAGCGCTCCA-1-v2-1k-6631-2029-v2" class="headerlink" title="v2.1k_AAACCTGAGCGCTCCA-1      v2.1k       6631         2029        v2"></a>v2.1k_AAACCTGAGCGCTCCA-1      v2.1k       6631         2029        v2</h2><h2 id="v2-1k-AAACCTGGTGATAAAC-1-v2-1k-2196-881-v2"><a href="#v2-1k-AAACCTGGTGATAAAC-1-v2-1k-2196-881-v2" class="headerlink" title="v2.1k_AAACCTGGTGATAAAC-1      v2.1k       2196          881        v2"></a>v2.1k_AAACCTGGTGATAAAC-1      v2.1k       2196          881        v2</h2><h2 id="v2-1k-AAACGGGGTTTGTGTG-1-v2-1k-2700-791-v2"><a href="#v2-1k-AAACGGGGTTTGTGTG-1-v2-1k-2700-791-v2" class="headerlink" title="v2.1k_AAACGGGGTTTGTGTG-1      v2.1k       2700          791        v2"></a>v2.1k_AAACGGGGTTTGTGTG-1      v2.1k       2700          791        v2</h2><h2 id="v2-1k-AAAGATGAGTACTTGC-1-v2-1k-3551-1183-v2"><a href="#v2-1k-AAAGATGAGTACTTGC-1-v2-1k-3551-1183-v2" class="headerlink" title="v2.1k_AAAGATGAGTACTTGC-1      v2.1k       3551         1183        v2"></a>v2.1k_AAAGATGAGTACTTGC-1      v2.1k       3551         1183        v2</h2><h2 id="v2-1k-AAAGCAAGTCTCTTAT-1-v2-1k-3080-1333-v2"><a href="#v2-1k-AAAGCAAGTCTCTTAT-1-v2-1k-3080-1333-v2" class="headerlink" title="v2.1k_AAAGCAAGTCTCTTAT-1      v2.1k       3080         1333        v2"></a>v2.1k_AAAGCAAGTCTCTTAT-1      v2.1k       3080         1333        v2</h2><h2 id="v2-1k-AAAGCAATCCACGAAT-1-v2-1k-5769-1556-v2"><a href="#v2-1k-AAAGCAATCCACGAAT-1-v2-1k-5769-1556-v2" class="headerlink" title="v2.1k_AAAGCAATCCACGAAT-1      v2.1k       5769         1556        v2"></a>v2.1k_AAAGCAATCCACGAAT-1      v2.1k       5769         1556        v2</h2><p>我们将手动计算线粒体reads的比例，并将其添加到metadata中。</p>
<p>mt.genes &lt;- rownames(alldata)[grep(“^MT-“,rownames(alldata))]<br>C&lt;-GetAssayData(object &#x3D; alldata, slot &#x3D; “counts”)</p>
<p>percent.mito &lt;- colSums(C[mt.genes,])&#x2F;Matrix::colSums(C)*100<br>alldata &lt;- AddMetaData(alldata, percent.mito, col.name &#x3D; “percent.mito”)<br>计算核糖体比例</p>
<p>我们用上面一样的方法来计算源自核糖体蛋白的基因表达比例。</p>
<p>rb.genes &lt;- rownames(alldata)[grep(“^RP[SL]”,rownames(alldata))]<br>percent.ribo &lt;- colSums(C[rb.genes,])&#x2F;Matrix::colSums(C)*100<br>alldata &lt;- AddMetaData(alldata, percent.ribo, col.name &#x3D; “percent.ribo”)<br>Plot QC</p>
<p>绘制小提琴图展示一些QC指标（feature）（可视化之为什么要使用箱线图？）：</p>
<p>VlnPlot(alldata, features &#x3D; “nFeature_RNA”, pt.size &#x3D; 0.1) + NoLegend()</p>
<p>VlnPlot(alldata, features &#x3D; “nCount_RNA”, pt.size &#x3D; 0.1) + NoLegend()</p>
<p>VlnPlot(alldata, features &#x3D; “percent.mito”, pt.size &#x3D; 0.1) + NoLegend()</p>
<p>VlnPlot(alldata, features &#x3D; “percent.ribo”, pt.size &#x3D; 0.1) + NoLegend()</p>
<p>如图所示，v2检测到的基因数较低，但检测出的核糖体蛋白比例却偏高。检测到较少的低表达基因是因为核糖体蛋白发生高表达，使得核糖体基因在转录景观中占较大比例。</p>
<p>QC方式不同时的散点图：</p>
<p>FeatureScatter(alldata, feature1 &#x3D; “nCount_RNA”, feature2 &#x3D; “nFeature_RNA”)</p>
<p>FeatureScatter(alldata, feature1 &#x3D; “nFeature_RNA”, feature2 &#x3D; “percent.mito”)</p>
<p>FeatureScatter(alldata, feature1&#x3D;”percent.ribo”, feature2&#x3D;”nFeature_RNA”)</p>
<p>我们还可以展示一个数据集：</p>
<p>FeatureScatter(alldata, feature1 &#x3D; “nCount_RNA”, feature2 &#x3D; “nFeature_RNA”, cells &#x3D; WhichCells(alldata, expression &#x3D; orig.ident &#x3D;&#x3D; “v3.1k”) )</p>
<p>过滤</p>
<p>线粒体过滤</p>
<p>线粒体reads比例高的细胞有很多，如果过滤掉它们后还剩下足够的细胞，那删除这些细胞可能是较为明智的。另一个选择是从数据集中删除所有线粒体reads，并希望剩余的基因仍然具有足够的生物学信号。第三种选择是在标准化过程中去除percent.mito变量。</p>
<p>在此次示例中，某些细胞中的线粒体reads比例高达99.7％，细胞信号很好的可能性不高，因此去除这些细胞。</p>
<p>可以通过展示的图决定合理的cutoff值。在这次例子中，大部分细胞的线粒体reads低于25％，可以将其用作临界值。</p>
<h1 id="选择线粒体基因百分比-lt-25-的细胞"><a href="#选择线粒体基因百分比-lt-25-的细胞" class="headerlink" title="选择线粒体基因百分比&lt;25%的细胞"></a>选择线粒体基因百分比&lt;25%的细胞</h1><p>selected &lt;- WhichCells(alldata, expression &#x3D; percent.mito &lt; 25)<br>length(selected)</p>
<h2 id="1-2703"><a href="#1-2703" class="headerlink" title="[1] 2703"></a>[1] 2703</h2><h1 id="对细胞取子集"><a href="#对细胞取子集" class="headerlink" title="对细胞取子集"></a>对细胞取子集</h1><p>data.filt &lt;- subset(alldata, cells &#x3D; selected)</p>
<h1 id="画小提琴图"><a href="#画小提琴图" class="headerlink" title="画小提琴图"></a>画小提琴图</h1><p>VlnPlot(data.filt, features &#x3D; “percent.mito”)</p>
<p>如图所示，mito百分比仍有很大差异，因此必须在数据分析步骤中进行处理。</p>
<p>基因过滤</p>
<p>检测到的基因数量高的离谱的话可能表明doublets，也有可能是因为样品中某一种细胞具有较多基因表达。</p>
<p>在这次的数据集中，v2和v3之间在基因检测方面也存在明显差异，因此对所有数据应用相同的临界值可能并不合适。</p>
<p>同时，在蛋白质测定数据p3.1k中，存在大量细胞几乎没有检测到具有双峰分布的基因。v3.1k和v2.12个数据集中也看不到这种分布类型。考虑到它们都是pbmc数据集，因此可以将这种分布视为低质量的库。</p>
<p>通过设置cutoff值分别为4100（对于v3 chemistry）和2000（对于v2）过滤测到高基因的（可能的doublets）细胞。</p>
<p>#从检测到许多基因的细胞开始。<br>high.det.v3 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &gt; 4100)<br>high.det.v2 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &gt; 2000 &amp; orig.ident &#x3D;&#x3D; “v2.1k”)</p>
<h1 id="细胞取子集"><a href="#细胞取子集" class="headerlink" title="细胞取子集"></a>细胞取子集</h1><p>data.filt &lt;- subset(data.filt, cells&#x3D;setdiff(WhichCells(data.filt),c(high.det.v2,high.det.v3)))</p>
<h1 id="查看细胞数"><a href="#查看细胞数" class="headerlink" title="查看细胞数"></a>查看细胞数</h1><p>ncol(data.filt)</p>
<h2 id="1-2631"><a href="#1-2631" class="headerlink" title="[1] 2631"></a>[1] 2631</h2><p>用低基因检测（低质量文库）过滤细胞，其中v3的基因数cutoff为1000，v2的基因数cutoff为500。</p>
<p>#筛选基因<br>low.det.v3 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &lt; 1000 &amp; orig.ident !&#x3D; “v2.1k”)<br>low.det.v2 &lt;- WhichCells(data.filt, expression &#x3D; nFeature_RNA &lt; 500 &amp; orig.ident &#x3D;&#x3D; “v2.1k”)</p>
<h1 id="去除细胞"><a href="#去除细胞" class="headerlink" title="去除细胞"></a>去除细胞</h1><p>data.filt &lt;- subset(data.filt, cells&#x3D;setdiff(WhichCells(data.filt),c(low.det.v2,low.det.v3)))</p>
<h1 id="查看细胞数-1"><a href="#查看细胞数-1" class="headerlink" title="查看细胞数"></a>查看细胞数</h1><p>ncol(data.filt)</p>
<h2 id="1-2531"><a href="#1-2531" class="headerlink" title="[1] 2531"></a>[1] 2531</h2><p>再次绘制QC图</p>
<p>VlnPlot(data.filt, features &#x3D; “nFeature_RNA”, pt.size &#x3D; 0.1) + NoLegend()</p>
<p>VlnPlot(data.filt, features &#x3D; “nCount_RNA”, pt.size &#x3D; 0.1) + NoLegend()</p>
<p>VlnPlot(data.filt, features &#x3D; “percent.mito”, pt.size &#x3D; 0.1) + NoLegend()</p>
<h1 id="在过滤前后检查每个样本的细胞数"><a href="#在过滤前后检查每个样本的细胞数" class="headerlink" title="在过滤前后检查每个样本的细胞数"></a>在过滤前后检查每个样本的细胞数</h1><p>table(Idents(alldata))<br>##</p>
<h2 id="p3-1k-v2-1k-v3-1k-1"><a href="#p3-1k-v2-1k-v3-1k-1" class="headerlink" title="p3.1k v2.1k v3.1k"></a>p3.1k v2.1k v3.1k</h2><h2 id="713-996-1222-1"><a href="#713-996-1222-1" class="headerlink" title="713   996  1222"></a>713   996  1222</h2><p>table(Idents(data.filt))<br>##</p>
<h2 id="p3-1k-v2-1k-v3-1k-2"><a href="#p3-1k-v2-1k-v3-1k-2" class="headerlink" title="p3.1k v2.1k v3.1k"></a>p3.1k v2.1k v3.1k</h2><h2 id="526-933-1072"><a href="#526-933-1072" class="headerlink" title="526   933  1072"></a>526   933  1072</h2><p>计算细胞周期分数</p>
<p>Seurat具有根据已知的S期和G2&#x2F;M期基因列表计算细胞周期分数的功能（Seurat亮点之细胞周期评分和回归）。</p>
<p>data.filt &lt;- CellCycleScoring(<br>  object &#x3D; data.filt,<br>  g2m.features &#x3D; cc.genes$g2m.genes,<br>  s.features &#x3D; cc.genes$s.genes<br>)</p>
<p>VlnPlot(data.filt, features &#x3D; c(“S.Score”,”G2M.Score”))</p>
<p>从上图可以看出不同数据集中处于S期和G2M期的细胞Score基本均为0，表明细胞周期对数据质量影响并不大。</p>
<p>Scater<br>可以使用scater包完成相似的QC绘图和细胞过滤，但是由于我们使用Seurat筛选了细胞，因此现在仅使用scater探索数据中的技术偏差。</p>
<p>可以直接从count矩阵创建SCE对象，网址为：<a target="_blank" rel="noopener" href="https://rdrr.io/bioc/scater/f/vignettes/overview.Rmd">https://rdrr.io/bioc/scater/f/vignettes/overview.Rmd</a></p>
<p>在该例中，我们可以直接从Seurat对象转换为SCE对象。</p>
<p>sce &lt;- as.SingleCellExperiment(data.filt)<br>我们用访问函数来评估SingleCellExperiment对象的不同元素。</p>
<p>counts（object）：用于返回read计数矩阵。如上所示，如果未为该对象定义任何计数，则计数矩阵slot为NULL。<br>exprs（object）：用于返回（对数计数）表达式值的矩阵，实际上为访问对象的对数计数slot（logcounts的同义词）。<br>SCE对象还有的slot：</p>
<p>细胞元数据（Cell metadata,），可以以DataFrame提供，其中行是细胞，列是细胞属性（例如细胞类型、培养条件、捕获的天数等）。<br>特征元数据（Feature metadata），可以以DataFrame提供，其中行是特征（例如基因），列是特征属性，例如Ensembl ID，生物型，gc含量等。<br>计算QC指标</p>
<p>默认情况下，QC指标是根据count data计算得出的。但是可以通过exprs_values参数进行更改。我们还可以在函数调用中包加入有关哪些基因是线粒体的信息。</p>
<h1 id="计算QC指标"><a href="#计算QC指标" class="headerlink" title="计算QC指标"></a>计算QC指标</h1><p>sce &lt;- calculateQCMetrics(sce, feature_controls &#x3D; list(mito &#x3D; mt.genes))</p>
<h1 id="check-what-all-entries-are"><a href="#check-what-all-entries-are" class="headerlink" title="check what all entries are -"></a>check what all entries are -</h1><p>colnames(colData(sce))</p>
<h2 id="1-“orig-ident”"><a href="#1-“orig-ident”" class="headerlink" title="[1] “orig.ident”"></a>[1] “orig.ident”</h2><h2 id="2-“nCount-RNA”"><a href="#2-“nCount-RNA”" class="headerlink" title="[2] “nCount_RNA”"></a>[2] “nCount_RNA”</h2><h2 id="3-“nFeature-RNA”"><a href="#3-“nFeature-RNA”" class="headerlink" title="[3] “nFeature_RNA”"></a>[3] “nFeature_RNA”</h2><h2 id="4-“Chemistry”"><a href="#4-“Chemistry”" class="headerlink" title="[4] “Chemistry”"></a>[4] “Chemistry”</h2><h2 id="5-“percent-mito”"><a href="#5-“percent-mito”" class="headerlink" title="[5] “percent.mito”"></a>[5] “percent.mito”</h2><h2 id="6-“percent-ribo”"><a href="#6-“percent-ribo”" class="headerlink" title="[6] “percent.ribo”"></a>[6] “percent.ribo”</h2><h2 id="7-“S-Score”"><a href="#7-“S-Score”" class="headerlink" title="[7] “S.Score”"></a>[7] “S.Score”</h2><h2 id="8-“G2M-Score”"><a href="#8-“G2M-Score”" class="headerlink" title="[8] “G2M.Score”"></a>[8] “G2M.Score”</h2><h2 id="9-“Phase”"><a href="#9-“Phase”" class="headerlink" title="[9] “Phase”"></a>[9] “Phase”</h2><h2 id="10-“ident”"><a href="#10-“ident”" class="headerlink" title="[10] “ident”"></a>[10] “ident”</h2><h2 id="11-“is-cell-control”"><a href="#11-“is-cell-control”" class="headerlink" title="[11] “is_cell_control”"></a>[11] “is_cell_control”</h2><h2 id="12-“total-features-by-counts”"><a href="#12-“total-features-by-counts”" class="headerlink" title="[12] “total_features_by_counts”"></a>[12] “total_features_by_counts”</h2><h2 id="13-“log10-total-features-by-counts”"><a href="#13-“log10-total-features-by-counts”" class="headerlink" title="[13] “log10_total_features_by_counts”"></a>[13] “log10_total_features_by_counts”</h2><h2 id="14-“total-counts”"><a href="#14-“total-counts”" class="headerlink" title="[14] “total_counts”"></a>[14] “total_counts”</h2><h2 id="15-“log10-total-counts”"><a href="#15-“log10-total-counts”" class="headerlink" title="[15] “log10_total_counts”"></a>[15] “log10_total_counts”</h2><h2 id="16-“pct-counts-in-top-50-features”"><a href="#16-“pct-counts-in-top-50-features”" class="headerlink" title="[16] “pct_counts_in_top_50_features”"></a>[16] “pct_counts_in_top_50_features”</h2><h2 id="17-“pct-counts-in-top-100-features”"><a href="#17-“pct-counts-in-top-100-features”" class="headerlink" title="[17] “pct_counts_in_top_100_features”"></a>[17] “pct_counts_in_top_100_features”</h2><h2 id="18-“pct-counts-in-top-200-features”"><a href="#18-“pct-counts-in-top-200-features”" class="headerlink" title="[18] “pct_counts_in_top_200_features”"></a>[18] “pct_counts_in_top_200_features”</h2><h2 id="19-“pct-counts-in-top-500-features”"><a href="#19-“pct-counts-in-top-500-features”" class="headerlink" title="[19] “pct_counts_in_top_500_features”"></a>[19] “pct_counts_in_top_500_features”</h2><h2 id="20-“total-features-by-counts-endogenous”"><a href="#20-“total-features-by-counts-endogenous”" class="headerlink" title="[20] “total_features_by_counts_endogenous”"></a>[20] “total_features_by_counts_endogenous”</h2><h2 id="21-“log10-total-features-by-counts-endogenous”"><a href="#21-“log10-total-features-by-counts-endogenous”" class="headerlink" title="[21] “log10_total_features_by_counts_endogenous”"></a>[21] “log10_total_features_by_counts_endogenous”</h2><h2 id="22-“total-counts-endogenous”"><a href="#22-“total-counts-endogenous”" class="headerlink" title="[22] “total_counts_endogenous”"></a>[22] “total_counts_endogenous”</h2><h2 id="23-“log10-total-counts-endogenous”"><a href="#23-“log10-total-counts-endogenous”" class="headerlink" title="[23] “log10_total_counts_endogenous”"></a>[23] “log10_total_counts_endogenous”</h2><h2 id="24-“pct-counts-endogenous”"><a href="#24-“pct-counts-endogenous”" class="headerlink" title="[24] “pct_counts_endogenous”"></a>[24] “pct_counts_endogenous”</h2><h2 id="25-“pct-counts-in-top-50-features-endogenous”"><a href="#25-“pct-counts-in-top-50-features-endogenous”" class="headerlink" title="[25] “pct_counts_in_top_50_features_endogenous”"></a>[25] “pct_counts_in_top_50_features_endogenous”</h2><h2 id="26-“pct-counts-in-top-100-features-endogenous”"><a href="#26-“pct-counts-in-top-100-features-endogenous”" class="headerlink" title="[26] “pct_counts_in_top_100_features_endogenous”"></a>[26] “pct_counts_in_top_100_features_endogenous”</h2><h2 id="27-“pct-counts-in-top-200-features-endogenous”"><a href="#27-“pct-counts-in-top-200-features-endogenous”" class="headerlink" title="[27] “pct_counts_in_top_200_features_endogenous”"></a>[27] “pct_counts_in_top_200_features_endogenous”</h2><h2 id="28-“pct-counts-in-top-500-features-endogenous”"><a href="#28-“pct-counts-in-top-500-features-endogenous”" class="headerlink" title="[28] “pct_counts_in_top_500_features_endogenous”"></a>[28] “pct_counts_in_top_500_features_endogenous”</h2><h2 id="29-“total-features-by-counts-feature-control”"><a href="#29-“total-features-by-counts-feature-control”" class="headerlink" title="[29] “total_features_by_counts_feature_control”"></a>[29] “total_features_by_counts_feature_control”</h2><h2 id="30-“log10-total-features-by-counts-feature-control”"><a href="#30-“log10-total-features-by-counts-feature-control”" class="headerlink" title="[30] “log10_total_features_by_counts_feature_control”"></a>[30] “log10_total_features_by_counts_feature_control”</h2><h2 id="31-“total-counts-feature-control”"><a href="#31-“total-counts-feature-control”" class="headerlink" title="[31] “total_counts_feature_control”"></a>[31] “total_counts_feature_control”</h2><h2 id="32-“log10-total-counts-feature-control”"><a href="#32-“log10-total-counts-feature-control”" class="headerlink" title="[32] “log10_total_counts_feature_control”"></a>[32] “log10_total_counts_feature_control”</h2><h2 id="33-“pct-counts-feature-control”"><a href="#33-“pct-counts-feature-control”" class="headerlink" title="[33] “pct_counts_feature_control”"></a>[33] “pct_counts_feature_control”</h2><h2 id="34-“pct-counts-in-top-50-features-feature-control”"><a href="#34-“pct-counts-in-top-50-features-feature-control”" class="headerlink" title="[34] “pct_counts_in_top_50_features_feature_control”"></a>[34] “pct_counts_in_top_50_features_feature_control”</h2><h2 id="35-“pct-counts-in-top-100-features-feature-control”"><a href="#35-“pct-counts-in-top-100-features-feature-control”" class="headerlink" title="[35] “pct_counts_in_top_100_features_feature_control”"></a>[35] “pct_counts_in_top_100_features_feature_control”</h2><h2 id="36-“pct-counts-in-top-200-features-feature-control”"><a href="#36-“pct-counts-in-top-200-features-feature-control”" class="headerlink" title="[36] “pct_counts_in_top_200_features_feature_control”"></a>[36] “pct_counts_in_top_200_features_feature_control”</h2><h2 id="37-“pct-counts-in-top-500-features-feature-control”"><a href="#37-“pct-counts-in-top-500-features-feature-control”" class="headerlink" title="[37] “pct_counts_in_top_500_features_feature_control”"></a>[37] “pct_counts_in_top_500_features_feature_control”</h2><h2 id="38-“total-features-by-counts-mito”"><a href="#38-“total-features-by-counts-mito”" class="headerlink" title="[38] “total_features_by_counts_mito”"></a>[38] “total_features_by_counts_mito”</h2><h2 id="39-“log10-total-features-by-counts-mito”"><a href="#39-“log10-total-features-by-counts-mito”" class="headerlink" title="[39] “log10_total_features_by_counts_mito”"></a>[39] “log10_total_features_by_counts_mito”</h2><h2 id="40-“total-counts-mito”"><a href="#40-“total-counts-mito”" class="headerlink" title="[40] “total_counts_mito”"></a>[40] “total_counts_mito”</h2><h2 id="41-“log10-total-counts-mito”"><a href="#41-“log10-total-counts-mito”" class="headerlink" title="[41] “log10_total_counts_mito”"></a>[41] “log10_total_counts_mito”</h2><h2 id="42-“pct-counts-mito”"><a href="#42-“pct-counts-mito”" class="headerlink" title="[42] “pct_counts_mito”"></a>[42] “pct_counts_mito”</h2><h2 id="43-“pct-counts-in-top-50-features-mito”"><a href="#43-“pct-counts-in-top-50-features-mito”" class="headerlink" title="[43] “pct_counts_in_top_50_features_mito”"></a>[43] “pct_counts_in_top_50_features_mito”</h2><h2 id="44-“pct-counts-in-top-100-features-mito”"><a href="#44-“pct-counts-in-top-100-features-mito”" class="headerlink" title="[44] “pct_counts_in_top_100_features_mito”"></a>[44] “pct_counts_in_top_100_features_mito”</h2><h2 id="45-“pct-counts-in-top-200-features-mito”"><a href="#45-“pct-counts-in-top-200-features-mito”" class="headerlink" title="[45] “pct_counts_in_top_200_features_mito”"></a>[45] “pct_counts_in_top_200_features_mito”</h2><h2 id="46-“pct-counts-in-top-500-features-mito”"><a href="#46-“pct-counts-in-top-500-features-mito”" class="headerlink" title="[46] “pct_counts_in_top_500_features_mito”"></a>[46] “pct_counts_in_top_500_features_mito”</h2><p>如您所见，scater为细胞计算了许多不同的QC指标。</p>
<p>Scater还可以根据基因计算一些统计信息：</p>
<p>colnames(rowData(sce))</p>
<h2 id="1-“is-feature-control”-“is-feature-control-mito”"><a href="#1-“is-feature-control”-“is-feature-control-mito”" class="headerlink" title="[1] “is_feature_control”      “is_feature_control_mito”"></a>[1] “is_feature_control”      “is_feature_control_mito”</h2><h2 id="3-“mean-counts”-“log10-mean-counts”"><a href="#3-“mean-counts”-“log10-mean-counts”" class="headerlink" title="[3] “mean_counts”             “log10_mean_counts”"></a>[3] “mean_counts”             “log10_mean_counts”</h2><h2 id="5-“n-cells-by-counts”-“pct-dropout-by-counts”"><a href="#5-“n-cells-by-counts”-“pct-dropout-by-counts”" class="headerlink" title="[5] “n_cells_by_counts”       “pct_dropout_by_counts”"></a>[5] “n_cells_by_counts”       “pct_dropout_by_counts”</h2><h2 id="7-“total-counts”-“log10-total-counts”"><a href="#7-“total-counts”-“log10-total-counts”" class="headerlink" title="[7] “total_counts”            “log10_total_counts”"></a>[7] “total_counts”            “log10_total_counts”</h2><p>绘制QC统计数据</p>
<p>高变基因：</p>
<p>让我们看看表达的前50位基因是什么。</p>
<p>plotHighestExprs(sce, exprs_values &#x3D; “counts”)</p>
<p>如图所示，MALAT1对应平均约4％的计数。在某些细胞中，一些count高达〜30％，可以考虑在进一步分析和聚类之前删除该基因。而且，线粒体基因在总计数中所占比例很高的话也应该被删除。</p>
<p>累积表达：</p>
<p>绘制文库大小的相对比例，该比例由每个细胞的最高表达基因（默认为500个基因）表示。这可以帮助我们寻找样本之间表达分布的差异。</p>
<h1 id="绘制每个样本"><a href="#绘制每个样本" class="headerlink" title="绘制每个样本"></a>绘制每个样本</h1><p>plotScater(sce, block1 &#x3D; “ident”, nfeatures &#x3D; 1000)</p>
<p>绘制基因统计：</p>
<p>函数plotRowData可以绘制rowData中的任何统计信息，例如均值表达与检测到的细胞数。</p>
<p>plotRowData(sce, x &#x3D; “n_cells_by_counts”, y &#x3D; “mean_counts”)</p>
<p>绘制细胞统计：</p>
<p>用函数plotColData可以以相同的方式绘制细胞的任何qc统计分析。</p>
<p>p1 &lt;- plotColData(sce, x &#x3D; “total_counts”,<br>    y &#x3D; “total_features_by_counts”, colour_by &#x3D; “ident”)<br>p2 &lt;- plotColData(sce, x &#x3D; “pct_counts_feature_control”,<br>    y &#x3D; “total_features_by_counts”, colour_by &#x3D; “ident”)<br>p3 &lt;- plotColData(sce, x &#x3D; “pct_counts_feature_control”,<br>    y &#x3D; “pct_counts_in_top_50_features”, colour_by &#x3D; “ident”)<br>multiplot(p1, p2, p3, cols &#x3D; 2)</p>
<p>识别QC统计数据中的异常值</p>
<p>识别低质量细胞的方法是在所有qc-stats上运行PCA，然后在PCA空间中识别异常值（Hemberg-lab单细胞转录组数据分析（十一）- Scater单细胞表达谱PCA可视化）。</p>
<p>sce &lt;- runPCA(sce, use_coldata &#x3D; TRUE,<br>    detect_outliers &#x3D; TRUE)</p>
<h2 id="sROC-0-1-2-loaded"><a href="#sROC-0-1-2-loaded" class="headerlink" title="sROC 0.1-2 loaded"></a>sROC 0.1-2 loaded</h2><p>plotReducedDim(sce, use_dimred&#x3D;”PCA_coldata”, colour_by &#x3D; “ident”)</p>
<p>＃检查是否有异常值<br>table(colData(sce)$outlier)<br>##</p>
<h2 id="FALSE"><a href="#FALSE" class="headerlink" title="FALSE"></a>FALSE</h2><h2 id="2531"><a href="#2531" class="headerlink" title="2531"></a>2531</h2><p>在这种情况下，我们已经滤除了低质量的细胞，并且在QC PCA中未检测到任何异常值。</p>
<p>降维<br>建议先用runPCA、runTSNE等函数做降维（ 用了这么多年的PCA可视化竟然是错的！！！），以便将它们存储在SCE对象中，这样就不必在每次绘制时都重新运行这些函数。除了这两个函数可以做降维，还有其他函数，比如plotPCA和plotTSNE等等，或者使用函数plotReducedDim并指定use_dimred &#x3D; “pca”等类似的。</p>
<p>在使用任意组分信息进行降维之前，请记住要设置随机种子，以便可以完全复现。</p>
<h1 id="使用前1000高变基因运行PCA"><a href="#使用前1000高变基因运行PCA" class="headerlink" title="使用前1000高变基因运行PCA"></a>使用前1000高变基因运行PCA</h1><p>sce &lt;- runPCA(sce, ntop &#x3D; 1000, exprs_values &#x3D; “logcounts”, ncomponents &#x3D; 20)</p>
<h1 id="PCA-with-different-coloring-first-4-components"><a href="#PCA-with-different-coloring-first-4-components" class="headerlink" title="PCA - with different coloring, first 4 components"></a>PCA - with different coloring, first 4 components</h1><h1 id="first-by-sample"><a href="#first-by-sample" class="headerlink" title="first by sample"></a>first by sample</h1><p>plotPCA(sce,ncomponents&#x3D;4,colour_by&#x3D;”ident”)</p>
<h1 id="then-by-Celltype"><a href="#then-by-Celltype" class="headerlink" title="then by Celltype"></a>then by Celltype</h1><p>plotPCA(sce,ncomponents&#x3D;4,colour_by&#x3D;”percent.mito”)</p>
<h1 id="Diffusion-map-运行需要安装destiny包"><a href="#Diffusion-map-运行需要安装destiny包" class="headerlink" title="Diffusion map, 运行需要安装destiny包!"></a>Diffusion map, 运行需要安装destiny包!</h1><p>set.seed(1)<br>sce &lt;- runDiffusionMap(sce, ntop &#x3D; 1000, ncomponents &#x3D; 4)<br>plotDiffusionMap(sce, colour_by&#x3D;”ident”,ncomponents&#x3D;4)</p>
<h1 id="tSNE-使用tSNE对降维数据进行可视化，top10PCs"><a href="#tSNE-使用tSNE对降维数据进行可视化，top10PCs" class="headerlink" title="tSNE -使用tSNE对降维数据进行可视化，top10PCs"></a>tSNE -使用tSNE对降维数据进行可视化，top10PCs</h1><p>set.seed(1)<br>sce &lt;- runTSNE(sce, ntop &#x3D; 1000, ncomponents &#x3D; 2, perplexity &#x3D; 30, n_dimred &#x3D; 10)<br>plotTSNE(sce, colour_by&#x3D;”ident”)</p>
<h1 id="UMAP-需要安装umap包"><a href="#UMAP-需要安装umap包" class="headerlink" title="UMAP,需要安装umap包!"></a>UMAP,需要安装umap包!</h1><p>set.seed(1)<br>sce &lt;- runUMAP(sce)<br>plotUMAP(object &#x3D; sce, colour_by&#x3D;”ident”)</p>
<p>影响因子的可视化<br>我们可以使用plotExplanatoryVariables函数调查不同因素的相对重要性。当拟合线性模型将每个基因的表达值与该因子进行回归时，我们计算colData（sce）中每个因子的可决系数R2来评估拟合优度，最好是在对数表达式值上执行此操作，可以减少平均值对方差的影响——因此，我们首先要进行归一化（取log值）。</p>
<p>下面的可视化计算耗时较长。</p>
<p>plotExplanatoryVariables(sce, variables &#x3D;  c(“ident”,”Chemistry”,”pct_counts_mito”, “total_features_by_counts”, “pct_counts_in_top_50_features”, “pct_counts_in_top_500_features”, “total_counts”, “S.Score”,”G2M.Score”))</p>
<p>每条线对应一个因子在所有基因中R平方值的分布。</p>
<p>我们要留意确定与QC或Metadata密切相关的PC。默认情况下是绘制前10个因子，但是在这里我们仅可视化一些特定因子。</p>
<h1 id="for-total-features"><a href="#for-total-features" class="headerlink" title="for total_features"></a>for total_features</h1><p>plotExplanatoryPCs(sce, variables &#x3D; c(“ident”, “Chemistry”,”pct_counts_mito”, “total_features_by_counts”, “pct_counts_in_top_50_features”, “total_counts”,”S.Score”,”G2M.Score”), npcs_to_plot &#x3D; 20)</p>
<p>问题：您认为可以在数据中看到一些明显的批次效应吗？您是否认为存在技术偏差？</p>
<p>PC1显然与数据的分布相关，例如前50-500个高度表达的基因在计数的比例，以及占total_features&#x2F;total_counts的比例。这是scRNAseq数据中的常见问题，可能是技术偏差，或者是大小和转录方式非常不同的细胞类型的生物学特征造成的。</p>
<p>同样很明显的是，许多topPC（尤其是PC 2、3、6、7、8）在很大程度上是由不同的样本（ident）或仅由v2 vs v3（Chemistry）来解释的。</p>
<p>Session info<br>sessionInfo()</p>
<h2 id="R-version-3-5-1-2018-07-02"><a href="#R-version-3-5-1-2018-07-02" class="headerlink" title="R version 3.5.1 (2018-07-02)"></a>R version 3.5.1 (2018-07-02)</h2><h2 id="Platform-x86-64-apple-darwin13-4-0-64-bit"><a href="#Platform-x86-64-apple-darwin13-4-0-64-bit" class="headerlink" title="Platform: x86_64-apple-darwin13.4.0 (64-bit)"></a>Platform: x86_64-apple-darwin13.4.0 (64-bit)</h2><h2 id="Running-under-macOS-10-14-4"><a href="#Running-under-macOS-10-14-4" class="headerlink" title="Running under: macOS  10.14.4"></a>Running under: macOS  10.14.4</h2><h2 id=""><a href="#" class="headerlink" title=""></a></h2><h2 id="Matrix-products-default"><a href="#Matrix-products-default" class="headerlink" title="Matrix products: default"></a>Matrix products: default</h2><h2 id="BLAS-x2F-LAPACK-x2F-Users-x2F-asbj-x2F-Programs-x2F-miniconda3-4-2-12-x2F-envs-x2F-elixir-course-x2F-lib-x2F-R-x2F-lib-x2F-libRblas-dylib"><a href="#BLAS-x2F-LAPACK-x2F-Users-x2F-asbj-x2F-Programs-x2F-miniconda3-4-2-12-x2F-envs-x2F-elixir-course-x2F-lib-x2F-R-x2F-lib-x2F-libRblas-dylib" class="headerlink" title="BLAS&#x2F;LAPACK: &#x2F;Users&#x2F;asbj&#x2F;Programs&#x2F;miniconda3_4.2.12&#x2F;envs&#x2F;elixir-course&#x2F;lib&#x2F;R&#x2F;lib&#x2F;libRblas.dylib"></a>BLAS&#x2F;LAPACK: &#x2F;Users&#x2F;asbj&#x2F;Programs&#x2F;miniconda3_4.2.12&#x2F;envs&#x2F;elixir-course&#x2F;lib&#x2F;R&#x2F;lib&#x2F;libRblas.dylib</h2><h2 id="-1"><a href="#-1" class="headerlink" title=""></a></h2><h2 id="locale"><a href="#locale" class="headerlink" title="locale:"></a>locale:</h2><h2 id="1-en-US-UTF-8-x2F-en-US-UTF-8-x2F-en-US-UTF-8-x2F-C-x2F-en-US-UTF-8-x2F-en-US-UTF-8"><a href="#1-en-US-UTF-8-x2F-en-US-UTF-8-x2F-en-US-UTF-8-x2F-C-x2F-en-US-UTF-8-x2F-en-US-UTF-8" class="headerlink" title="[1] en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;C&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8"></a>[1] en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8&#x2F;C&#x2F;en_US.UTF-8&#x2F;en_US.UTF-8</h2><h2 id="-2"><a href="#-2" class="headerlink" title=""></a></h2><h2 id="attached-base-packages"><a href="#attached-base-packages" class="headerlink" title="attached base packages:"></a>attached base packages:</h2><h2 id="1-parallel-stats4-stats-graphics-grDevices-utils-datasets"><a href="#1-parallel-stats4-stats-graphics-grDevices-utils-datasets" class="headerlink" title="[1] parallel  stats4    stats     graphics  grDevices utils     datasets"></a>[1] parallel  stats4    stats     graphics  grDevices utils     datasets</h2><h2 id="8-methods-base"><a href="#8-methods-base" class="headerlink" title="[8] methods   base"></a>[8] methods   base</h2><h2 id="-3"><a href="#-3" class="headerlink" title=""></a></h2><h2 id="other-attached-packages"><a href="#other-attached-packages" class="headerlink" title="other attached packages:"></a>other attached packages:</h2><h2 id="1-Matrix-1-2-17-scater-1-10-1"><a href="#1-Matrix-1-2-17-scater-1-10-1" class="headerlink" title="[1] Matrix_1.2-17               scater_1.10.1"></a>[1] Matrix_1.2-17               scater_1.10.1</h2><h2 id="3-ggplot2-3-1-1-SingleCellExperiment-1-4-0"><a href="#3-ggplot2-3-1-1-SingleCellExperiment-1-4-0" class="headerlink" title="[3] ggplot2_3.1.1               SingleCellExperiment_1.4.0"></a>[3] ggplot2_3.1.1               SingleCellExperiment_1.4.0</h2><h2 id="5-SummarizedExperiment-1-12-0-DelayedArray-0-8-0"><a href="#5-SummarizedExperiment-1-12-0-DelayedArray-0-8-0" class="headerlink" title="[5] SummarizedExperiment_1.12.0 DelayedArray_0.8.0"></a>[5] SummarizedExperiment_1.12.0 DelayedArray_0.8.0</h2><h2 id="7-BiocParallel-1-16-6-matrixStats-0-54-0"><a href="#7-BiocParallel-1-16-6-matrixStats-0-54-0" class="headerlink" title="[7] BiocParallel_1.16.6         matrixStats_0.54.0"></a>[7] BiocParallel_1.16.6         matrixStats_0.54.0</h2><h2 id="9-Biobase-2-42-0-GenomicRanges-1-34-0"><a href="#9-Biobase-2-42-0-GenomicRanges-1-34-0" class="headerlink" title="[9] Biobase_2.42.0              GenomicRanges_1.34.0"></a>[9] Biobase_2.42.0              GenomicRanges_1.34.0</h2><h2 id="11-GenomeInfoDb-1-18-1-IRanges-2-16-0"><a href="#11-GenomeInfoDb-1-18-1-IRanges-2-16-0" class="headerlink" title="[11] GenomeInfoDb_1.18.1         IRanges_2.16.0"></a>[11] GenomeInfoDb_1.18.1         IRanges_2.16.0</h2><h2 id="13-S4Vectors-0-20-1-BiocGenerics-0-28-0"><a href="#13-S4Vectors-0-20-1-BiocGenerics-0-28-0" class="headerlink" title="[13] S4Vectors_0.20.1            BiocGenerics_0.28.0"></a>[13] S4Vectors_0.20.1            BiocGenerics_0.28.0</h2><h2 id="15-Seurat-3-0-0"><a href="#15-Seurat-3-0-0" class="headerlink" title="[15] Seurat_3.0.0"></a>[15] Seurat_3.0.0</h2><h2 id="-4"><a href="#-4" class="headerlink" title=""></a></h2><h2 id="loaded-via-a-namespace-and-not-attached"><a href="#loaded-via-a-namespace-and-not-attached" class="headerlink" title="loaded via a namespace (and not attached):"></a>loaded via a namespace (and not attached):</h2><h2 id="1-reticulate-1-12-R-utils-2-8-0"><a href="#1-reticulate-1-12-R-utils-2-8-0" class="headerlink" title="[1] reticulate_1.12          R.utils_2.8.0"></a>[1] reticulate_1.12          R.utils_2.8.0</h2><h2 id="3-tidyselect-0-2-5-htmlwidgets-1-3"><a href="#3-tidyselect-0-2-5-htmlwidgets-1-3" class="headerlink" title="[3] tidyselect_0.2.5         htmlwidgets_1.3"></a>[3] tidyselect_0.2.5         htmlwidgets_1.3</h2><h2 id="5-grid-3-5-1-trimcluster-0-1-2-1"><a href="#5-grid-3-5-1-trimcluster-0-1-2-1" class="headerlink" title="[5] grid_3.5.1               trimcluster_0.1-2.1"></a>[5] grid_3.5.1               trimcluster_0.1-2.1</h2><h2 id="7-ranger-0-11-1-Rtsne-0-15"><a href="#7-ranger-0-11-1-Rtsne-0-15" class="headerlink" title="[7] ranger_0.11.1            Rtsne_0.15"></a>[7] ranger_0.11.1            Rtsne_0.15</h2><h2 id="9-munsell-0-5-0-destiny-2-12-0"><a href="#9-munsell-0-5-0-destiny-2-12-0" class="headerlink" title="[9] munsell_0.5.0            destiny_2.12.0"></a>[9] munsell_0.5.0            destiny_2.12.0</h2><h2 id="11-codetools-0-2-16-ica-1-0-2"><a href="#11-codetools-0-2-16-ica-1-0-2" class="headerlink" title="[11] codetools_0.2-16         ica_1.0-2"></a>[11] codetools_0.2-16         ica_1.0-2</h2><h2 id="13-umap-0-2-0-0-future-1-12-0"><a href="#13-umap-0-2-0-0-future-1-12-0" class="headerlink" title="[13] umap_0.2.0.0             future_1.12.0"></a>[13] umap_0.2.0.0             future_1.12.0</h2><h2 id="15-sROC-0-1-2-withr-2-1-2"><a href="#15-sROC-0-1-2-withr-2-1-2" class="headerlink" title="[15] sROC_0.1-2               withr_2.1.2"></a>[15] sROC_0.1-2               withr_2.1.2</h2><h2 id="17-colorspace-1-4-1-knitr-1-20"><a href="#17-colorspace-1-4-1-knitr-1-20" class="headerlink" title="[17] colorspace_1.4-1         knitr_1.20"></a>[17] colorspace_1.4-1         knitr_1.20</h2><h2 id="19-ROCR-1-0-7-robustbase-0-93-2"><a href="#19-ROCR-1-0-7-robustbase-0-93-2" class="headerlink" title="[19] ROCR_1.0-7               robustbase_0.93-2"></a>[19] ROCR_1.0-7               robustbase_0.93-2</h2><h2 id="21-vcd-1-4-4-VIM-4-8-0"><a href="#21-vcd-1-4-4-VIM-4-8-0" class="headerlink" title="[21] vcd_1.4-4                VIM_4.8.0"></a>[21] vcd_1.4-4                VIM_4.8.0</h2><h2 id="23-TTR-0-23-4-gbRd-0-4-11"><a href="#23-TTR-0-23-4-gbRd-0-4-11" class="headerlink" title="[23] TTR_0.23-4               gbRd_0.4-11"></a>[23] TTR_0.23-4               gbRd_0.4-11</h2><h2 id="25-listenv-0-7-0-Rdpack-0-10-1"><a href="#25-listenv-0-7-0-Rdpack-0-10-1" class="headerlink" title="[25] listenv_0.7.0            Rdpack_0.10-1"></a>[25] listenv_0.7.0            Rdpack_0.10-1</h2><h2 id="27-labeling-0-3-GenomeInfoDbData-1-2-1"><a href="#27-labeling-0-3-GenomeInfoDbData-1-2-1" class="headerlink" title="[27] labeling_0.3             GenomeInfoDbData_1.2.1"></a>[27] labeling_0.3             GenomeInfoDbData_1.2.1</h2><h2 id="29-cvTools-0-3-2-bit64-0-9-7"><a href="#29-cvTools-0-3-2-bit64-0-9-7" class="headerlink" title="[29] cvTools_0.3.2            bit64_0.9-7"></a>[29] cvTools_0.3.2            bit64_0.9-7</h2><h2 id="31-rhdf5-2-26-2-ggthemes-4-1-1"><a href="#31-rhdf5-2-26-2-ggthemes-4-1-1" class="headerlink" title="[31] rhdf5_2.26.2             ggthemes_4.1.1"></a>[31] rhdf5_2.26.2             ggthemes_4.1.1</h2><h2 id="33-diptest-0-75-7-R6-2-4-0"><a href="#33-diptest-0-75-7-R6-2-4-0" class="headerlink" title="[33] diptest_0.75-7           R6_2.4.0"></a>[33] diptest_0.75-7           R6_2.4.0</h2><h2 id="35-ggbeeswarm-0-6-0-robCompositions-2-0-10"><a href="#35-ggbeeswarm-0-6-0-robCompositions-2-0-10" class="headerlink" title="[35] ggbeeswarm_0.6.0         robCompositions_2.0.10"></a>[35] ggbeeswarm_0.6.0         robCompositions_2.0.10</h2><h2 id="37-rsvd-1-0-0-RcppEigen-0-3-3-5-0"><a href="#37-rsvd-1-0-0-RcppEigen-0-3-3-5-0" class="headerlink" title="[37] rsvd_1.0.0               RcppEigen_0.3.3.5.0"></a>[37] rsvd_1.0.0               RcppEigen_0.3.3.5.0</h2><h2 id="39-mvoutlier-2-0-9-hdf5r-1-2-0"><a href="#39-mvoutlier-2-0-9-hdf5r-1-2-0" class="headerlink" title="[39] mvoutlier_2.0.9          hdf5r_1.2.0"></a>[39] mvoutlier_2.0.9          hdf5r_1.2.0</h2><h2 id="41-flexmix-2-3-15-bitops-1-0-6"><a href="#41-flexmix-2-3-15-bitops-1-0-6" class="headerlink" title="[41] flexmix_2.3-15           bitops_1.0-6"></a>[41] flexmix_2.3-15           bitops_1.0-6</h2><h2 id="43-reshape-0-8-8-assertthat-0-2-1"><a href="#43-reshape-0-8-8-assertthat-0-2-1" class="headerlink" title="[43] reshape_0.8.8            assertthat_0.2.1"></a>[43] reshape_0.8.8            assertthat_0.2.1</h2><h2 id="45-SDMTools-1-1-221-1-scales-1-0-0"><a href="#45-SDMTools-1-1-221-1-scales-1-0-0" class="headerlink" title="[45] SDMTools_1.1-221.1       scales_1.0.0"></a>[45] SDMTools_1.1-221.1       scales_1.0.0</h2><h2 id="47-nnet-7-3-12-beeswarm-0-2-3"><a href="#47-nnet-7-3-12-beeswarm-0-2-3" class="headerlink" title="[47] nnet_7.3-12              beeswarm_0.2.3"></a>[47] nnet_7.3-12              beeswarm_0.2.3</h2><h2 id="49-gtable-0-3-0-npsurv-0-4-0"><a href="#49-gtable-0-3-0-npsurv-0-4-0" class="headerlink" title="[49] gtable_0.3.0             npsurv_0.4-0"></a>[49] gtable_0.3.0             npsurv_0.4-0</h2><h2 id="51-globals-0-12-4-rlang-0-3-4"><a href="#51-globals-0-12-4-rlang-0-3-4" class="headerlink" title="[51] globals_0.12.4           rlang_0.3.4"></a>[51] globals_0.12.4           rlang_0.3.4</h2><h2 id="53-scatterplot3d-0-3-41-splines-3-5-1"><a href="#53-scatterplot3d-0-3-41-splines-3-5-1" class="headerlink" title="[53] scatterplot3d_0.3-41     splines_3.5.1"></a>[53] scatterplot3d_0.3-41     splines_3.5.1</h2><h2 id="55-lazyeval-0-2-2-yaml-2-2-0"><a href="#55-lazyeval-0-2-2-yaml-2-2-0" class="headerlink" title="[55] lazyeval_0.2.2           yaml_2.2.0"></a>[55] lazyeval_0.2.2           yaml_2.2.0</h2><h2 id="57-reshape2-1-4-3-abind-1-4-5"><a href="#57-reshape2-1-4-3-abind-1-4-5" class="headerlink" title="[57] reshape2_1.4.3           abind_1.4-5"></a>[57] reshape2_1.4.3           abind_1.4-5</h2><h2 id="59-tools-3-5-1-zCompositions-1-2-0"><a href="#59-tools-3-5-1-zCompositions-1-2-0" class="headerlink" title="[59] tools_3.5.1              zCompositions_1.2.0"></a>[59] tools_3.5.1              zCompositions_1.2.0</h2><h2 id="61-gplots-3-0-1-1-RColorBrewer-1-1-2"><a href="#61-gplots-3-0-1-1-RColorBrewer-1-1-2" class="headerlink" title="[61] gplots_3.0.1.1           RColorBrewer_1.1-2"></a>[61] gplots_3.0.1.1           RColorBrewer_1.1-2</h2><h2 id="63-proxy-0-4-22-ggridges-0-5-1"><a href="#63-proxy-0-4-22-ggridges-0-5-1" class="headerlink" title="[63] proxy_0.4-22             ggridges_0.5.1"></a>[63] proxy_0.4-22             ggridges_0.5.1</h2><h2 id="65-Rcpp-1-0-1-plyr-1-8-4"><a href="#65-Rcpp-1-0-1-plyr-1-8-4" class="headerlink" title="[65] Rcpp_1.0.1               plyr_1.8.4"></a>[65] Rcpp_1.0.1               plyr_1.8.4</h2><h2 id="67-zlibbioc-1-28-0-purrr-0-3-2"><a href="#67-zlibbioc-1-28-0-purrr-0-3-2" class="headerlink" title="[67] zlibbioc_1.28.0          purrr_0.3.2"></a>[67] zlibbioc_1.28.0          purrr_0.3.2</h2><h2 id="69-RCurl-1-95-4-12-pbapply-1-4-0"><a href="#69-RCurl-1-95-4-12-pbapply-1-4-0" class="headerlink" title="[69] RCurl_1.95-4.12          pbapply_1.4-0"></a>[69] RCurl_1.95-4.12          pbapply_1.4-0</h2><h2 id="71-viridis-0-5-1-cowplot-0-9-4"><a href="#71-viridis-0-5-1-cowplot-0-9-4" class="headerlink" title="[71] viridis_0.5.1            cowplot_0.9.4"></a>[71] viridis_0.5.1            cowplot_0.9.4</h2><h2 id="73-zoo-1-8-5-haven-2-1-0"><a href="#73-zoo-1-8-5-haven-2-1-0" class="headerlink" title="[73] zoo_1.8-5                haven_2.1.0"></a>[73] zoo_1.8-5                haven_2.1.0</h2><h2 id="75-ggrepel-0-8-1-cluster-2-0-9"><a href="#75-ggrepel-0-8-1-cluster-2-0-9" class="headerlink" title="[75] ggrepel_0.8.1            cluster_2.0.9"></a>[75] ggrepel_0.8.1            cluster_2.0.9</h2><h2 id="77-magrittr-1-5-RSpectra-0-14-0"><a href="#77-magrittr-1-5-RSpectra-0-14-0" class="headerlink" title="[77] magrittr_1.5             RSpectra_0.14-0"></a>[77] magrittr_1.5             RSpectra_0.14-0</h2><h2 id="79-data-table-1-12-2-openxlsx-4-1-0"><a href="#79-data-table-1-12-2-openxlsx-4-1-0" class="headerlink" title="[79] data.table_1.12.2        openxlsx_4.1.0"></a>[79] data.table_1.12.2        openxlsx_4.1.0</h2><h2 id="81-lmtest-0-9-36-RANN-2-6"><a href="#81-lmtest-0-9-36-RANN-2-6" class="headerlink" title="[81] lmtest_0.9-36            RANN_2.6"></a>[81] lmtest_0.9-36            RANN_2.6</h2><h2 id="83-truncnorm-1-0-8-mvtnorm-1-0-10"><a href="#83-truncnorm-1-0-8-mvtnorm-1-0-10" class="headerlink" title="[83] truncnorm_1.0-8          mvtnorm_1.0-10"></a>[83] truncnorm_1.0-8          mvtnorm_1.0-10</h2><h2 id="85-fitdistrplus-1-0-14-hms-0-4-2"><a href="#85-fitdistrplus-1-0-14-hms-0-4-2" class="headerlink" title="[85] fitdistrplus_1.0-14      hms_0.4.2"></a>[85] fitdistrplus_1.0-14      hms_0.4.2</h2><h2 id="87-lsei-1-2-0-evaluate-0-13"><a href="#87-lsei-1-2-0-evaluate-0-13" class="headerlink" title="[87] lsei_1.2-0               evaluate_0.13"></a>[87] lsei_1.2-0               evaluate_0.13</h2><h2 id="89-smoother-1-1-rio-0-5-16"><a href="#89-smoother-1-1-rio-0-5-16" class="headerlink" title="[89] smoother_1.1             rio_0.5.16"></a>[89] smoother_1.1             rio_0.5.16</h2><h2 id="91-mclust-5-4-3-readxl-1-3-1"><a href="#91-mclust-5-4-3-readxl-1-3-1" class="headerlink" title="[91] mclust_5.4.3             readxl_1.3.1"></a>[91] mclust_5.4.3             readxl_1.3.1</h2><h2 id="93-gridExtra-2-3-compiler-3-5-1"><a href="#93-gridExtra-2-3-compiler-3-5-1" class="headerlink" title="[93] gridExtra_2.3            compiler_3.5.1"></a>[93] gridExtra_2.3            compiler_3.5.1</h2><h2 id="95-tibble-2-1-1-KernSmooth-2-23-15"><a href="#95-tibble-2-1-1-KernSmooth-2-23-15" class="headerlink" title="[95] tibble_2.1.1             KernSmooth_2.23-15"></a>[95] tibble_2.1.1             KernSmooth_2.23-15</h2><h2 id="97-crayon-1-3-4-R-oo-1-22-0"><a href="#97-crayon-1-3-4-R-oo-1-22-0" class="headerlink" title="[97] crayon_1.3.4             R.oo_1.22.0"></a>[97] crayon_1.3.4             R.oo_1.22.0</h2><h2 id="99-htmltools-0-3-6-pcaPP-1-9-73"><a href="#99-htmltools-0-3-6-pcaPP-1-9-73" class="headerlink" title="[99] htmltools_0.3.6          pcaPP_1.9-73"></a>[99] htmltools_0.3.6          pcaPP_1.9-73</h2><h2 id="101-tidyr-0-8-3-rrcov-1-4-4"><a href="#101-tidyr-0-8-3-rrcov-1-4-4" class="headerlink" title="[101] tidyr_0.8.3              rrcov_1.4-4"></a>[101] tidyr_0.8.3              rrcov_1.4-4</h2><h2 id="103-MASS-7-3-51-4-fpc-2-1-11-1"><a href="#103-MASS-7-3-51-4-fpc-2-1-11-1" class="headerlink" title="[103] MASS_7.3-51.4            fpc_2.1-11.1"></a>[103] MASS_7.3-51.4            fpc_2.1-11.1</h2><h2 id="105-boot-1-3-22-car-3-0-2"><a href="#105-boot-1-3-22-car-3-0-2" class="headerlink" title="[105] boot_1.3-22              car_3.0-2"></a>[105] boot_1.3-22              car_3.0-2</h2><h2 id="107-sgeostat-1-0-27-R-methodsS3-1-7-1"><a href="#107-sgeostat-1-0-27-R-methodsS3-1-7-1" class="headerlink" title="[107] sgeostat_1.0-27          R.methodsS3_1.7.1"></a>[107] sgeostat_1.0-27          R.methodsS3_1.7.1</h2><h2 id="109-gdata-2-18-0-metap-1-1"><a href="#109-gdata-2-18-0-metap-1-1" class="headerlink" title="[109] gdata_2.18.0             metap_1.1"></a>[109] gdata_2.18.0             metap_1.1</h2><h2 id="111-igraph-1-2-4-1-forcats-0-4-0"><a href="#111-igraph-1-2-4-1-forcats-0-4-0" class="headerlink" title="[111] igraph_1.2.4.1           forcats_0.4.0"></a>[111] igraph_1.2.4.1           forcats_0.4.0</h2><h2 id="113-pkgconfig-2-0-2-foreign-0-8-71"><a href="#113-pkgconfig-2-0-2-foreign-0-8-71" class="headerlink" title="[113] pkgconfig_2.0.2          foreign_0.8-71"></a>[113] pkgconfig_2.0.2          foreign_0.8-71</h2><h2 id="115-laeken-0-5-0-sp-1-3-1"><a href="#115-laeken-0-5-0-sp-1-3-1" class="headerlink" title="[115] laeken_0.5.0             sp_1.3-1"></a>[115] laeken_0.5.0             sp_1.3-1</h2><h2 id="117-plotly-4-8-0-vipor-0-4-5"><a href="#117-plotly-4-8-0-vipor-0-4-5" class="headerlink" title="[117] plotly_4.8.0             vipor_0.4.5"></a>[117] plotly_4.8.0             vipor_0.4.5</h2><h2 id="119-XVector-0-22-0-bibtex-0-4-2"><a href="#119-XVector-0-22-0-bibtex-0-4-2" class="headerlink" title="[119] XVector_0.22.0           bibtex_0.4.2"></a>[119] XVector_0.22.0           bibtex_0.4.2</h2><h2 id="121-NADA-1-6-1-stringr-1-4-0"><a href="#121-NADA-1-6-1-stringr-1-4-0" class="headerlink" title="[121] NADA_1.6-1               stringr_1.4.0"></a>[121] NADA_1.6-1               stringr_1.4.0</h2><h2 id="123-digest-0-6-18-sctransform-0-2-0"><a href="#123-digest-0-6-18-sctransform-0-2-0" class="headerlink" title="[123] digest_0.6.18            sctransform_0.2.0"></a>[123] digest_0.6.18            sctransform_0.2.0</h2><h2 id="125-pls-2-7-0-tsne-0-1-3"><a href="#125-pls-2-7-0-tsne-0-1-3" class="headerlink" title="[125] pls_2.7-0                tsne_0.1-3"></a>[125] pls_2.7-0                tsne_0.1-3</h2><h2 id="127-rmarkdown-1-11-cellranger-1-1-0"><a href="#127-rmarkdown-1-11-cellranger-1-1-0" class="headerlink" title="[127] rmarkdown_1.11           cellranger_1.1.0"></a>[127] rmarkdown_1.11           cellranger_1.1.0</h2><h2 id="129-DelayedMatrixStats-1-4-0-curl-3-3"><a href="#129-DelayedMatrixStats-1-4-0-curl-3-3" class="headerlink" title="[129] DelayedMatrixStats_1.4.0 curl_3.3"></a>[129] DelayedMatrixStats_1.4.0 curl_3.3</h2><h2 id="131-kernlab-0-9-27-gtools-3-8-1"><a href="#131-kernlab-0-9-27-gtools-3-8-1" class="headerlink" title="[131] kernlab_0.9-27           gtools_3.8.1"></a>[131] kernlab_0.9-27           gtools_3.8.1</h2><h2 id="133-modeltools-0-2-22-nlme-3-1-139"><a href="#133-modeltools-0-2-22-nlme-3-1-139" class="headerlink" title="[133] modeltools_0.2-22        nlme_3.1-139"></a>[133] modeltools_0.2-22        nlme_3.1-139</h2><h2 id="135-jsonlite-1-6-Rhdf5lib-1-4-3"><a href="#135-jsonlite-1-6-Rhdf5lib-1-4-3" class="headerlink" title="[135] jsonlite_1.6             Rhdf5lib_1.4.3"></a>[135] jsonlite_1.6             Rhdf5lib_1.4.3</h2><h2 id="137-carData-3-0-2-viridisLite-0-3-0"><a href="#137-carData-3-0-2-viridisLite-0-3-0" class="headerlink" title="[137] carData_3.0-2            viridisLite_0.3.0"></a>[137] carData_3.0-2            viridisLite_0.3.0</h2><h2 id="139-pillar-1-3-1-lattice-0-20-38"><a href="#139-pillar-1-3-1-lattice-0-20-38" class="headerlink" title="[139] pillar_1.3.1             lattice_0.20-38"></a>[139] pillar_1.3.1             lattice_0.20-38</h2><h2 id="141-GGally-1-4-0-httr-1-4-0"><a href="#141-GGally-1-4-0-httr-1-4-0" class="headerlink" title="[141] GGally_1.4.0             httr_1.4.0"></a>[141] GGally_1.4.0             httr_1.4.0</h2><h2 id="143-DEoptimR-1-0-8-survival-2-44-1-1"><a href="#143-DEoptimR-1-0-8-survival-2-44-1-1" class="headerlink" title="[143] DEoptimR_1.0-8           survival_2.44-1.1"></a>[143] DEoptimR_1.0-8           survival_2.44-1.1</h2><h2 id="145-xts-0-11-1-glue-1-3-1"><a href="#145-xts-0-11-1-glue-1-3-1" class="headerlink" title="[145] xts_0.11-1               glue_1.3.1"></a>[145] xts_0.11-1               glue_1.3.1</h2><h2 id="147-zip-2-0-1-png-0-1-7"><a href="#147-zip-2-0-1-png-0-1-7" class="headerlink" title="[147] zip_2.0.1                png_0.1-7"></a>[147] zip_2.0.1                png_0.1-7</h2><h2 id="149-prabclus-2-2-7-bit-1-1-14"><a href="#149-prabclus-2-2-7-bit-1-1-14" class="headerlink" title="[149] prabclus_2.2-7           bit_1.1-14"></a>[149] prabclus_2.2-7           bit_1.1-14</h2><h2 id="151-class-7-3-15-stringi-1-4-3"><a href="#151-class-7-3-15-stringi-1-4-3" class="headerlink" title="[151] class_7.3-15             stringi_1.4.3"></a>[151] class_7.3-15             stringi_1.4.3</h2><h2 id="153-HDF5Array-1-10-1-caTools-1-17-1-2"><a href="#153-HDF5Array-1-10-1-caTools-1-17-1-2" class="headerlink" title="[153] HDF5Array_1.10.1         caTools_1.17.1.2"></a>[153] HDF5Array_1.10.1         caTools_1.17.1.2</h2><h2 id="155-dplyr-0-8-0-1-irlba-2-3-3"><a href="#155-dplyr-0-8-0-1-irlba-2-3-3" class="headerlink" title="[155] dplyr_0.8.0.1            irlba_2.3.3"></a>[155] dplyr_0.8.0.1            irlba_2.3.3</h2><h2 id="157-e1071-1-7-0-future-apply-1-1-0"><a href="#157-e1071-1-7-0-future-apply-1-1-0" class="headerlink" title="[157] e1071_1.7-0              future.apply_1.1.0"></a>[157] e1071_1.7-0              future.apply_1.1.0</h2><h2 id="159-ape-5-3"><a href="#159-ape-5-3" class="headerlink" title="[159] ape_5.3"></a>[159] ape_5.3</h2>
            


        </div>
    </div>
    <div id="post-footer" class="post-footer main-content-wrap">
        
        
            <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-04-21-%E4%BD%BF%E7%94%A8Aspera%E9%AB%98%E9%80%9F%E4%B8%8B%E8%BD%BDSRA%E6%95%B0%E6%8D%AE/"
                    data-tooltip="使用Aspera高速下载SRA数据"
                    aria-label="上一篇: 使用Aspera高速下载SRA数据"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-04-15-%E4%BB%8E%E5%9F%BA%E5%9B%A0%E5%90%8D%E5%88%B0GO%E6%B3%A8%E9%87%8A%E4%B8%80%E6%AD%A5%E5%88%B0%E4%BD%8D/"
                    data-tooltip="从基因名到GO注释一步到位"
                    aria-label="下一篇: 从基因名到GO注释一步到位"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


        
        
    </div>
</article>



                <footer id="footer" class="main-content-wrap">
    <span class="copyrights">
        Copyrights &copy; 2022 Dr.iuu. All Rights Reserved.
    </span>
</footer>

            </div>
            
                <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
                    <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-04-21-%E4%BD%BF%E7%94%A8Aspera%E9%AB%98%E9%80%9F%E4%B8%8B%E8%BD%BDSRA%E6%95%B0%E6%8D%AE/"
                    data-tooltip="使用Aspera高速下载SRA数据"
                    aria-label="上一篇: 使用Aspera高速下载SRA数据"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-04-15-%E4%BB%8E%E5%9F%BA%E5%9B%A0%E5%90%8D%E5%88%B0GO%E6%B3%A8%E9%87%8A%E4%B8%80%E6%AD%A5%E5%88%B0%E4%BD%8D/"
                    data-tooltip="从基因名到GO注释一步到位"
                    aria-label="下一篇: 从基因名到GO注释一步到位"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


                </div>
                
    <div id="share-options-bar" class="share-options-bar" data-behavior="4">
        <i id="btn-close-shareoptions" class="fa fa-times"></i>
        <ul class="share-options">
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                        aria-label="分享到 Facebook"
                    >
                        <i class="fab fa-facebook" aria-hidden="true"></i><span>分享到 Facebook</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                        aria-label="分享到 Twitter"
                    >
                        <i class="fab fa-twitter" aria-hidden="true"></i><span>分享到 Twitter</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://plus.google.com/share?url=https://iuu19.github.io/2020-04-20-%E4%BB%A3%E7%A0%81%E5%88%86%E6%9E%90-Sing-cell-Sequencing-QC%E8%AF%A6%E8%A7%A3/"
                        aria-label="分享到 Google+"
                    >
                        <i class="fab fa-google-plus" aria-hidden="true"></i><span>分享到 Google+</span>
                    </a>
                </li>
            
        </ul>
    </div>


            
        </div>
        


    
        
    

<div id="about">
    <div id="about-card">
        <div id="about-btn-close">
            <i class="fa fa-times"></i>
        </div>
        
            <img id="about-card-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
            <h4 id="about-card-name">Dr.iuu</h4>
        
            <div id="about-card-bio"><p>NCMC-SCMC-SWCH</p>
</div>
        
        
            <div id="about-card-job">
                <i class="fa fa-briefcase"></i>
                <br/>
                <p>Bioinfo-CHD</p>

            </div>
        
        
            <div id="about-card-location">
                <i class="fa fa-map-marker-alt"></i>
                <br/>
                Pudong, Shanghai
            </div>
        
    </div>
</div>

        
        
<div id="cover" style="background-image:url('/assets/images/cover2.jpg');"></div>
        <!--SCRIPTS-->

<script src="/assets/js/script-3nlvflbbvekvxx0g6h2clfqhhxs61cvvbofv7nmctigkdbbhu5htlvmrqg80.min.js"></script>

<!--SCRIPTS END-->





    </body>
</html>
