
<!DOCTYPE html>
<html lang="zh-cn">
    
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="iuu&#39;s blog">
    <title>GWAS|GMMAT安装及使用指南 - iuu&#39;s blog</title>
    <meta name="author" content="Dr.iuu">
    
    
        <link rel="icon" href="https://iuu19.github.io/assets/images/coveriuu.jpg">
    
    
        
            <link rel="alternate" type="application/atom+xml" title="RSS" href="/atom.xml">
        
    
    <script type="application/ld+json">{"@context":"http://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg"},"articleBody":"\n\n\n\nGWAS|GMMAT安装及使用指南\n\n\n1\tIntroductionGMMAT is an R package for performing association tests using generalized linear mixed models (GLMMs)1 in genome-wide association studies (GWAS) and sequencing associa- tion studies. GLMMs provide a broad range of models for correlated data analysis. In  the GWAS and sequencing association study context, examples of correlated data include those from family studies, samples with cryptic relatedness and&#x2F;or shared environmental effects, as well as samples generated from complex sampling designs.\nGMMAT first fits a GLMM with covariate adjustment and random effects to account for population structure and family or cryptic relatedness. \n\nFor GWAS, GMMAT performs score tests for each genetic variant. \nFor candidate gene studies, GMMAT can also perform Wald tests to get the effect size estimate for each genetic variant. \nFor rare variant analysis from sequencing association studies, GMMAT performs the burden test, the sequence kernel association test (SKAT), SKAT-O and the efficient hybrid test of the burden test and SKAT, in the variant Set Mixed Model Association Tests (SMMAT) framework, based on user-defined variant sets.\n\n2\tThe modelIn the context of single variant test, GMMAT works with the following GLMM\nηi &#x3D; g(µi) &#x3D; Xiα + Giβ + bi\n3\tGetting started3.1\tInstalling GMMATinstall.packages(\"GMMAT\")\n\ninstall.packages(c(\"devtools\",  \"RcppArmadillo\",  \"CompQuadForm\",  \"doMC\", \"foreach\", \"Matrix\",  \"BiocManager\",  \"testthat\"))\n\nif (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\nBiocManager::install(c(\"SeqArray\",\"SeqVarTools\"))\n\n4\tInputGMMAT requires the phenotype and covariates in an R data frame, known positive semi- definite matrices Vk as an R matrix (in the case of a single matrix) or an R list (in the case of multiple matrices), and genotypes saved in a plain text file (or in a compressed plain text file .gz or .bz2), a PLINK binary PED file, or in a GDS format file. We describe how to prepare these data below.\n4.1\tPhenotype and covariatesFor example, here we show the header and first 6 rows of the example text file pheno.txt:\nid\tdisease\ttrait\tage\tsex\n1\t1\t5.45\t61\t0\n2\t1\t5.61\t50\t1\n3\t0\t3.1\t54\t0\n4\t1\t6.22\t48\t1\n5\t1\t5.42\t49\t0\n6\t0\t6.22\t50\t1\n...\t\t\t\t\n\nIn this example, there are an ID column (id), one binary phenotype (disease), one quantitative phenotype (trait) and two covariates (age and sex). There can be additional columns for unused variables, and the order of columns does not matter. To read it into  R as a data frame, you can use\npheno.file  ","dateCreated":"2020-06-13T13:52:45+08:00","dateModified":"2020-06-13T13:52:45+08:00","datePublished":"2020-06-13T13:52:45+08:00","description":"","headline":"GWAS|GMMAT安装及使用指南","image":["1.png"],"mainEntityOfPage":{"@type":"WebPage","@id":"https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"},"publisher":{"@type":"Organization","name":"Dr.iuu","sameAs":["https://github.com/iuu19/iuu19.github.io"],"image":"coveriuu.jpg","logo":{"@type":"ImageObject","url":"coveriuu.jpg"}},"url":"https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/","keywords":"GWAS, GMMAT, SMMAT","thumbnailUrl":"1.png"}</script>
    <meta property="og:type" content="blog">
<meta property="og:title" content="GWAS|GMMAT安装及使用指南">
<meta property="og:url" content="https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/index.html">
<meta property="og:site_name" content="iuu&#39;s blog">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2020-06-13T05:52:45.000Z">
<meta property="article:modified_time" content="2020-06-13T05:52:45.000Z">
<meta property="article:author" content="Dr.iuu">
<meta property="article:tag" content="GWAS">
<meta property="article:tag" content="GMMAT">
<meta property="article:tag" content="SMMAT">
<meta name="twitter:card" content="summary">
    
    
        
    
    
        <meta property="og:image" content="https://iuu19.github.io/assets/images/coveriuu.jpg"/>
    
    
        <meta property="og:image" content="https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/1.png"/>
        <meta class="swiftype" name="image" data-type="enum" content="https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/1.png"/>
    
    
    
    <!--STYLES-->
    
<link rel="stylesheet" href="/assets/css/style-gf8wcm29a1hig4pgasvkpmpwcvjyqc9ufzpbo404wwnc2qcamao8rlll1csq.min.css">

    <!--STYLES END-->
    

    

    
<link rel="stylesheet" href="/css/prism-vsc-dark-plus.css" type="text/css">
<link rel="stylesheet" href="/css/prism-line-numbers.css" type="text/css"></head>

    <body>
        <div id="blog">
            <!-- Define author's picture -->


    
        
            
        
    

<header id="header" data-behavior="4">
    <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
    <div class="header-title">
        <a
            class="header-title-link"
            href="/"
            aria-label=""
        >
            iuu&#39;s blog
        </a>
    </div>
    
        
            <a
                class="header-right-picture "
                href="/all-about"
                aria-label="打开链接: /all-about"
            >
        
        
            <img class="header-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
        </a>
    
</header>

            <!-- Define author's picture -->



        
    

<nav id="sidebar" data-behavior="4">
    <div class="sidebar-container">
        
            <div class="sidebar-profile">
                <a
                    href="/#about"
                    aria-label="阅读有关作者的更多信息"
                >
                    <img class="sidebar-profile-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
                </a>
                <h4 class="sidebar-profile-name">Dr.iuu</h4>
                
                    <h5 class="sidebar-profile-bio"><p>NCMC-SCMC-SWCH</p>
</h5>
                
            </div>
        
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://iuu19.github.io"
                            
                            rel="noopener"
                            title="首页"
                        >
                        <i class="sidebar-button-icon fa fa-home" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">首页</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-categories"
                            
                            rel="noopener"
                            title="分类"
                        >
                        <i class="sidebar-button-icon fa fa-bookmark" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">分类</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-tags"
                            
                            rel="noopener"
                            title="标签"
                        >
                        <i class="sidebar-button-icon fa fa-tags" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">标签</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-archives"
                            
                            rel="noopener"
                            title="归档"
                        >
                        <i class="sidebar-button-icon fa fa-archive" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">归档</span>
                    </a>
            </li>
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/all-about"
                            
                            rel="noopener"
                            title="关于"
                        >
                        <i class="sidebar-button-icon fa fa-question" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">关于</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="https://github.com/iuu19/iuu19.github.io"
                            
                                target="_blank"
                            
                            rel="noopener"
                            title="GitHub"
                        >
                        <i class="sidebar-button-icon fab fa-github" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">GitHub</span>
                    </a>
            </li>
            
        </ul>
        
            <ul class="sidebar-buttons">
            
                <li class="sidebar-button">
                    
                        <a
                            class="sidebar-button-link "
                            href="/atom.xml"
                            
                            rel="noopener"
                            title="RSS"
                        >
                        <i class="sidebar-button-icon fa fa-rss" aria-hidden="true"></i>
                        <span class="sidebar-button-desc">RSS</span>
                    </a>
            </li>
            
        </ul>
        
    </div>
</nav>

            
            <div id="main" data-behavior="4"
                 class="
                        hasCoverMetaIn
                        ">
                
<article class="post">
    
    
        <div class="post-header main-content-wrap text-center">
    
        <h1 class="post-title">
            GWAS|GMMAT安装及使用指南
        </h1>
    
    
        <div class="post-meta">
    <time datetime="2020-06-13T13:52:45+08:00">
	
		    6月 13, 2020
    	
    </time>
    
        <span>发布在 </span>
        
    <a class="category-link" href="/all-categories/GWAS/">GWAS</a>


    
</div>

    
</div>

    
    <div class="post-content markdown">
        <div class="main-content-wrap">
            <font face="times new roman" size=4>

<span id="more"></span>

<p>GWAS|GMMAT安装及使用指南</p>
<h1 id="table-of-contents">目录</h1><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#1Introduction"><span class="toc-text">1	Introduction</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#2The-model"><span class="toc-text">2	The model</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#3Getting-started"><span class="toc-text">3	Getting started</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#3-1Installing-GMMAT"><span class="toc-text">3.1	Installing GMMAT</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#4Input"><span class="toc-text">4	Input</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#4-1Phenotype-and-covariates"><span class="toc-text">4.1	Phenotype and covariates</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-2Matrices-of-covariance-structure"><span class="toc-text">4.2	Matrices of covariance structure</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-3Genotypes"><span class="toc-text">4.3	Genotypes</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5Running-GMMAT"><span class="toc-text">5	Running GMMAT</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#5-1Fitting-GLMM"><span class="toc-text">5.1	Fitting GLMM</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#5-2Single-variant-tests"><span class="toc-text">5.2	Single variant tests</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#5-2-1Score-tests"><span class="toc-text">5.2.1	Score tests</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#5-2-2Wald-tests"><span class="toc-text">5.2.2	Wald tests</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#5-2-3Meta-analysis"><span class="toc-text">5.2.3	Meta-analysis</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#5-3-2Meta-analysis"><span class="toc-text">5.3.2	Meta-analysis</span></a></li></ol></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#6Output"><span class="toc-text">6	Output</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#7Advanced-options"><span class="toc-text">7	Advanced options</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#7-1Alternative-model-fitting-algorithms"><span class="toc-text">7.1	Alternative model fitting algorithms</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-2Changing-model-fitting-parameters"><span class="toc-text">7.2	Changing model fitting parameters</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-3Missing-genotypes"><span class="toc-text">7.3	Missing genotypes</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-4Reordered-genotypes"><span class="toc-text">7.4	Reordered genotypes</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-5Parallel-computing"><span class="toc-text">7.5	Parallel computing</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-6Variant-filters"><span class="toc-text">7.6	Variant filters</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-7Internal-minor-allele-frequency-weights"><span class="toc-text">7.7	Internal minor allele frequency weights</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-8Allele-flipping"><span class="toc-text">7.8	Allele flipping</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-9P-values-of-weighted-sum-of-chi-squares"><span class="toc-text">7.9	P values of weighted sum of chi-squares</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-10Heterogeneous-genetic-effects-in-variant-set-meta-analysis"><span class="toc-text">7.10	Heterogeneous genetic effects in variant set meta-analysis</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-11Other-options"><span class="toc-text">7.11	Other options</span></a></li></ol></li></ol></li></ol></li></ol>
<hr>
<h1 id="1Introduction"><a href="#1Introduction" class="headerlink" title="1	Introduction"></a>1	Introduction</h1><p>GMMAT is an R package for performing association tests using generalized linear mixed models (GLMMs)1 in genome-wide association studies (GWAS) and sequencing associa- tion studies. GLMMs provide a broad range of models for correlated data analysis. In  the GWAS and sequencing association study context, examples of correlated data include those from family studies, samples with cryptic relatedness and&#x2F;or shared environmental effects, as well as samples generated from complex sampling designs.</p>
<p>GMMAT first fits a GLMM with covariate adjustment and random effects to account for population structure and family or cryptic relatedness. </p>
<ul>
<li><strong>For GWAS</strong>, GMMAT performs score tests for each genetic variant. </li>
<li><strong>For candidate gene studies</strong>, GMMAT can also perform Wald tests to get the effect size estimate for each genetic variant. </li>
<li><strong>For rare variant analysis</strong> from sequencing association studies, GMMAT performs the <code>burden test</code>, <code>the sequence kernel association test (SKAT)</code>, <code>SKAT-O</code> and <code>the efficient hybrid test of the burden test and SKAT</code>, in the variant <strong>Set Mixed Model Association Tests (SMMAT)</strong> framework, based on user-defined variant sets.</li>
</ul>
<h1 id="2The-model"><a href="#2The-model" class="headerlink" title="2	The model"></a>2	The model</h1><p>In the context of single variant test, GMMAT works with the following GLMM</p>
<p><em>ηi &#x3D; g(µi) &#x3D; Xiα + Giβ + bi</em></p>
<h1 id="3Getting-started"><a href="#3Getting-started" class="headerlink" title="3	Getting started"></a>3	Getting started</h1><h2 id="3-1Installing-GMMAT"><a href="#3-1Installing-GMMAT" class="headerlink" title="3.1	Installing GMMAT"></a>3.1	Installing GMMAT</h2><pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">install.packages("GMMAT")
<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">install.packages(c("devtools",  "RcppArmadillo",  "CompQuadForm",  "doMC", "foreach", "Matrix",  "BiocManager",  "testthat"))
<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(c("SeqArray","SeqVarTools"))
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span></span></code></pre>
<h1 id="4Input"><a href="#4Input" class="headerlink" title="4	Input"></a>4	Input</h1><p>GMMAT requires the phenotype and covariates in an R data frame, known positive semi- definite matrices Vk as an R matrix (in the case of a single matrix) or an R list (in the case of multiple matrices), and genotypes saved in a plain text file (or in a compressed plain text file .gz or .bz2), a PLINK binary PED file, or in a GDS format file. We describe how to prepare these data below.</p>
<h2 id="4-1Phenotype-and-covariates"><a href="#4-1Phenotype-and-covariates" class="headerlink" title="4.1	Phenotype and covariates"></a>4.1	Phenotype and covariates</h2><p>For example, here we show the header and first 6 rows of the example text file pheno.txt:</p>
<pre><code>id	disease	trait	age	sex
1	1	5.45	61	0
2	1	5.61	50	1
3	0	3.1	54	0
4	1	6.22	48	1
5	1	5.42	49	0
6	0	6.22	50	1
...				
</code></pre>
<p>In this example, there are an ID column (id), one binary phenotype (disease), one quantitative phenotype (trait) and two covariates (age and sex). There can be additional columns for unused variables, and the order of columns does not matter. To read it into  R as a data frame, you can use</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">pheno.file  <-  system.file("extdata",  "pheno.txt",  package  =  "GMMAT")
pheno  <-  read.table(pheno.file,  header  =  TRUE)
# Missing values in the data frame should be recognizable by  R as NA. For  example, if  you use . (period) to denote missing values in the text file, you can use
pheno  <-  read.table(pheno.file,  header  =  TRUE,  na.strings  =  ".")
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span></span></code></pre>
<h2 id="4-2Matrices-of-covariance-structure"><a href="#4-2Matrices-of-covariance-structure" class="headerlink" title="4.2	Matrices of covariance structure"></a>4.2	Matrices of covariance structure</h2><p>GMMAT requires at least one positive semi-definite matrix Vk to model the covariance structure of the random effects (for cross-sectional data). In the simplest case, this is usually a GRM estimated from the genotype data. <strong>Currently GMMAT does not provide  a function to calculate the GRM</strong>, but there are many software packages that can do this job. For example, <strong>GEMMA</strong> can be used to estimate either the centered GRM or the standardized GRM. GRM saved in an external file must be read into R as a matrix. For example</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">GRM.file  <-  system.file("extdata",  "GRM.txt.bz2",  package  =  "GMMAT")
GRM  <-  as.matrix(read.table(GRM.file,  check.names  =  FALSE))
# This matrix can include more than n individuals in practice, but the rownames and colnames must include all individuals’ id in the phenotype and covariates data frame. Multiple matrices can be used to allow multiple components of random effects. In such cases, the matrices should be constructed as a list of matrices, and each matrix should comply with the rownames and colnames requirements described above (although they don’t have to be in the same order). For example, if you have 3 R matrices Mat1, Mat2 and Mat3, you can construct the R list
Mats  <-  list(Mat1,  Mat2,  Mat3)
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span></span></code></pre>
<p>All matrices must be positive semi-definite. Sparse matrices from the Matrix package are also allowed.</p>
<h2 id="4-3Genotypes"><a href="#4-3Genotypes" class="headerlink" title="4.3	Genotypes"></a>4.3	Genotypes</h2><p>GMMAT can take genotype files either in plain text format (or the compressed version<br>.gz or .bz2), PLINK binary PED format, or in the GDS format. Non-integer imputed genotypes (dosages) should be saved  in plain text files (or the compressed version .gz   or .bz2). The plain text file can be space-, tab-, comma-, or even special character- delimited, and there can be additional rows (e.g., comments) and&#x2F;or columns before the genotype data matrix. Here is an example of part of a tab-delimited plain text genotype file geno.txt:</p>
<pre><code>#  This  is  an  example  genotype  file  for  demonstrating  GMMAT
# Each row represents one SNP for all individuals in the study
#  First  column  is  SNP  name,  second  and  third  columns  are  alleles  (Allele1 and  Allele2):  it  is  recommended  to  use  Allele1  for  the  reference  allele and  Allele2  for  the  effect  allele,  but  reversed  coding  is  also  allowed and does not affect association test results (users should be cautious with  allele  coding  when  interpreting  results)
#  Starting  from  fourth  column,  each  column  represents  one  individual #  In  this  example,  there  are  400  individuals  and  100  SNPs
SNP1	A	T	0	0	NA	NA	...
SNP2	A	C	1	0	1	0	...
SNP3	A	C	0	0	0	1	...
SNP4	A	G	1	0	1	1	...
SNP5	A	G	1	0	2	1	...
```	

Genotypes in Variant Call Format (VCF) and PLINK binary PED format can be converted to the GDS format using seqVCF2GDS and seqBED2GDS functions from the SeqArray package:
```&#123;r&#125;
SeqArray::seqVCF2GDS(&quot;VCF_file_name&quot;,  &quot;GDS_file_name&quot;)
SeqArray::seqBED2GDS(&quot;BED_file_name&quot;,  &quot;FAM_file_name&quot;,  &quot;BIM_file_name&quot;, &quot;GDS_file_name&quot;)
</code></pre>
<h2 id="5Running-GMMAT"><a href="#5Running-GMMAT" class="headerlink" title="5	Running GMMAT"></a>5	Running GMMAT</h2><pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">#If GMMAT has been successfully installed, you can load it in an R session using

library(GMMAT)
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span></span></code></pre>
<p>We provide 6 functions in GMMAT : </p>
<ul>
<li><p>glmmkin for fitting the GLMM with known Vk</p>
</li>
<li><p>glmm.score for running single variant score tests</p>
</li>
<li><p>glmm.wald for single variant Wald tests</p>
</li>
<li><p>glmm.score.meta for performing meta-analysis on score test results</p>
</li>
<li><p>SM-MAT for running variant set tests SMMAT (also known as gene-based tests or aggregate variant tests)</p>
</li>
<li><p>SMMAT.meta for performing variant set tests meta-analysis using intermediate files (single variant scores and their covariance matrices in each variant set, from the SMMAT function). </p>
<p>Details about how to use these functions, their arguments and returned values can be found in the R help document of GMMAT.</p>
</li>
</ul>
<h4 id="5-1Fitting-GLMM"><a href="#5-1Fitting-GLMM" class="headerlink" title="5.1	Fitting GLMM"></a>5.1	Fitting GLMM</h4><p>Here we provide a simple example of fitting GLMM using glmmkin. We have the binary phenotype disease, the quantitative phenotype trait, and two covariates age and sex,  saved in a plain text file pheno.txt. We also have computed the GRM externally and  saved it in a compressed file GRM.txt.bz2. In this example we fit a GLMM assuming Bernoulli distribution of the disease and logit link function (also known as a logistic mixed model). We adjust for age and sex, and use one <em>n</em> x <em>n</em> matrix as Vk (the GRM) to model the covariance structure of the random effects.</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">model0  <-  glmmkin(disease  ~  age  +  sex,  data  =  pheno,  kins  =  GRM, id  =  "id",  family  =  binomial(link  =  "logit"))
model0$theta
model0$coefficients
model0$cov
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span></span></code></pre>
<p>Note that pheno and GRM must be read into R as a data frame and a matrix, respectively. When using the function glmmkin, the data frame of phenotype and covariates (in our example, pheno) should be passed to the argument ”data”, and the matrix or the list of matrices for random effects (in our example, the matrix GRM) should be passed  to the argument ”kins”. The first argument of the function glmmkin is ”fixed”, which requires a formula for fixed effects. The syntax of the formula is the same as the formula used in a linear model lm and a generalized linear model glm. The example model above is equivalent to</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">model0  <-  glmmkin(fixed  =  disease  ~  age  +  sex,  data  =  pheno,  kins  =  GRM, id  =  "id",  family  =  binomial(link  =  "logit"))
<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>
<p>The argument ”id” specifies the ID column name in the data frame pheno. The argument ”family” takes the same syntax as used in a generalized linear model glm. For example, if you would like to fit a LMM for a quantitative trait, you can use</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">model1  <-  glmmkin(fixed  =  trait  ~  age  +  sex,  data  =  pheno,  kins  =  GRM,	id  =  "id",  family  =  gaussian(link  =  "identity"))
<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>
<p><strong>Please avoid using LMMs for binary traits.</strong></p>
<p>To fit a heteroscedastic LMM for a quantitative trait (allowing heterogeneous residual variances among different groups), you can use</p>
<pre><code>model2  &lt;-  glmmkin(fixed  =  trait  ~  age  +  sex,  data  =  pheno,  kins  =  GRM,id  =  &quot;id&quot;,  groups  =  &quot;disease&quot;, family  =  gaussian(link  =  &quot;identity&quot;))
model2$theta
</code></pre>
<p>In this example, groups are defined by disease status. Therefore, disease cases and controls are assumed to have different residual variances on the quantitative trait, after adjusting for age and sex.</p>
<p>In the following example, we have the same binary phenotype disease, covariates age and sex, and the same GRM as in the previous example. In addition to the GRM, we  have another n n matrix to model the covariance structure of the random effects.</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">M10  <-  matrix(0,  400,  400)
for(i  in  1:40)  M10[(i-1)*10+(1:10),  (i-1)*10+(1:10)]  <-  1
rownames(M10)  <-  colnames(M10)  <-  1:400
Mats  <-  list(GRM,  M10)
model3  <-  glmmkin(fixed  =  disease  ~  age  +  sex,  data  =  pheno,  id  =  "id", kins  =  Mats,  family  =  binomial(link  =  "logit"))
model3$theta
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>
<p>For longitudinal data, the following example illustrates a random intercept only model, which is appropriate for analyzing repeated measures with no time trends. We have 5 exchangeable observations of the continuous phenotype y.repeated for each individual, adjusting for sex as a fixed effects covariate.</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">pheno2.file  <-  system.file("extdata",  "pheno2.txt",  package  =  "GMMAT")
pheno2  <-  read.table(pheno2.file,  header  =  TRUE)
model4  <-  glmmkin(y.repeated  ~  sex,  data  =  pheno2,  kins  =  GRM,  id  =  "id",	family  =  gaussian(link  =  "identity"))
model4$theta
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span></span></code></pre>
<p>The last example of this section illustrates a random intercept and random slope model, which is appropriate for analyzing longitudinal data with individual-specific time trends. We have observations of the continuous phenotype y.trend at 5 time points for each individual, adjusting for sex and time as fixed effects covariates. </p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">model5  <-  glmmkin(y.trend  ~  sex  +  time,  data  =  pheno2,  kins  =  GRM,  id  =  "id", random.slope  =  "time",  family  =  gaussian(link  =  "identity"))
model5$theta
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span></span></code></pre>
<h4 id="5-2Single-variant-tests"><a href="#5-2Single-variant-tests" class="headerlink" title="5.2	Single variant tests"></a>5.2	Single variant tests</h4><h5 id="5-2-1Score-tests"><a href="#5-2-1Score-tests" class="headerlink" title="5.2.1	Score tests"></a>5.2.1	Score tests</h5><p>When performing score tests in GWAS, we need a fitted GLMM under the null hypothesis H0 : β &#x3D; 0 and a genotype file. We can construct score tests using the glmmkin class object returned from the function glmmkin for the null GLMM. Note that score tests require only vector&#x2F;matrix multiplications and are much faster than Wald tests, which require fitting a new GLMM for each SNP. Score tests give the direction of effects but not effect size estimates. However, we can simply add score statistics and their variances from different studies to perform a meta-analysis. Here we provide a simple example of score tests using the plain text genotype file ”geno.txt”:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">geno.file  <-  system.file("extdata",  "geno.txt",  package  =  "GMMAT")
glmm.score(model0,  infile  =  geno.file,  outfile  =	"glmm.score.text.testoutfile.txt",  infile.nrow.skip  =  5,	infile.ncol.skip  =  3,  infile.ncol.print  =  1:3,	infile.header.print  =  c("SNP",  "Allele1",  "Allele2"))
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span></span></code></pre>
<p>The first argument in glmm.score is the returned glmmkin class object from the null GLMM. The argument ”infile” is the name (and path if not in the current working di- rectory) of the plain text genotype file (or compressed files .gz and .bz2), and the ar- gument ”outfile” is the name of the output file.  In this example genotype file, we  have    5 comment lines to skip using ”infile.nrow.skip”. The first 3 columns contain informa- tion on SNP name and alleles, which we skip from the analysis using ”infile.ncol.skip” but subsequently keep in the output file using ”infile.ncol.print” to select the 1st,  2nd   and 3rd columns. Corresponding column names in the output file can be assigned using ”infile.header.print”.<br>If your genotype information is saved in a PLINK binary PED file ”geno.bed”, you can use:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">geno.file  <-  strsplit(system.file("extdata",  "geno.bed",	package  =  "GMMAT"),  ".bed",  fixed  =  TRUE)[[1]]
glmm.score(model0,  infile  =  geno.file,  outfile  =	"glmm.score.bed.testoutfile.txt")
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span></span></code></pre>
<p>Here ”infile” is the prefix (and path if not in the current working directory) of the PLINK files (.bed, .bim and .fam). SNP information in the .bim file (in our example, ”geno.bim”) is carried over to the output file.</p>
<p>Alternatively, if your genotype information is saved in a GDS file ”geno.gds”, you can use:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">geno.file  <-  system.file("extdata",  "geno.gds",  package  =  "GMMAT")
glmm.score(model0,  infile  =  geno.file,  outfile  =	"glmm.score.gds.testoutfile.txt")
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span></span></code></pre>
<p>The function glmm.score returns the actual computation time in seconds from its function call for plain text genotype files (or compressed files .gz and .bz2) and PLINK binary PED files. For GDS genotype files, no value is returned.</p>
<h5 id="5-2-2Wald-tests"><a href="#5-2-2Wald-tests" class="headerlink" title="5.2.2	Wald tests"></a>5.2.2	Wald tests</h5><p>When performing Wald tests for candidate SNPs to get effect size estimates, we need the phenotype (and covariates) data frame,  the matrices modeling the covariance  structure  of the random effects, and the genotype file. To perform Wald tests, we do not need fitting the null GLMM required in score tests using glmmkin. In the example below, we perform Wald tests for 4 candidate SNPs of interest and get their effect estimates:</p>
<pre><code>geno.file  &lt;-  system.file(&quot;extdata&quot;,  &quot;geno.txt&quot;,  package  =  &quot;GMMAT&quot;)
snps  &lt;-  c(&quot;SNP10&quot;,  &quot;SNP25&quot;,  &quot;SNP1&quot;,  &quot;SNP0&quot;)
glmm.wald(fixed  =  disease  ~  age  +  sex,  data  =  pheno,  kins  =  GRM,  id  =  &quot;id&quot;,	family  =  binomial(link  =  &quot;logit&quot;),  infile  =  geno.file,  snps  =  snps,	infile.nrow.skip  =  5,  infile.ncol.skip  =  3,  infile.ncol.print  =  1:3,	infile.header.print  =  c(&quot;SNP&quot;,  &quot;Allele1&quot;,  &quot;Allele2&quot;))
</code></pre>
<p>The syntax is a hybrid of glmmkin and glmm.score. Note that the argument ”fixed” is a formula including the covariates  but NOT the test SNPs.  The argument ”snps” is      a character vector of the names of the test SNPs. If ”infile” is a plain text genotype file  (or compressed files .gz and .bz2), the function glmm.wald returns a data frame with first columns copied from the genotype file using ”infile.ncol.print” and names specified using ”infile.header.print”, followed by the sample size N, the allele frequency (AF) of the effect allele (Allele2 in this example, but you can also define Allele1 as the effect allele in your coded genotype file), effect size estimate BETA of the effect allele, standard error SE, Wald test P value PVAL, and an indicator for whether the  GLMM  is converged. Note that in the example above, SNP0 is not actually included in the genotype file, so all results are missing.</p>
<p>If your genotype information is saved in a PLINK binary PED file ”geno.bed”, you can use:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">geno.file  <-  strsplit(system.file("extdata",  "geno.bed",	package  =  "GMMAT"),  ".bed",  fixed  =  TRUE)[[1]]
glmm.wald(fixed  =  disease  ~  age  +  sex,  data  =  pheno,  kins  =  GRM,  id  =  "id",	family  =  binomial(link  =  "logit"),  infile  =  geno.file,  snps  =  snps)
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span></span></code></pre>
<p>It returns a data frame with first 6 columns copied from the .bim file (in our example, ”geno.bim”), followed by the sample size N, the allele frequency (AF) of A2 allele (the effect allele, note that A1 allele in .bim is coded 0 and A2 allele is coded 1), effect size estimate BETA of A2 allele, standard error SE, Wald test P value PVAL, and an indicator for whether the GLMM is converged.<br>Alternatively, if your genotype information is saved in a GDS file ”geno.gds”, you can use:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">geno.file  <-  system.file("extdata",  "geno.gds",  package  =  "GMMAT")
glmm.wald(fixed  =  disease  ~  age  +  sex,  data  =  pheno,  kins  =  GRM,  id  =  "id",
    family  =  binomial(link  =  "logit"),  infile  =  geno.file,  snps  =  snps)
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span></span></code></pre>
<p>It returns a data frame with first 5 columns information extracted from the GDS file, followed by the sample size N, the allele frequency (AF) of ALT allele, effect size estimate BETA of ALT allele, standard error SE, Wald test P value PVAL, and an indicator for whether the GLMM is converged.</p>
<h5 id="5-2-3Meta-analysis"><a href="#5-2-3Meta-analysis" class="headerlink" title="5.2.3	Meta-analysis"></a>5.2.3	Meta-analysis</h5><p>Score test results from multiple studies can be combined in meta-analysis. We provide the function glmm.score.meta to perform meta-analysis on score test results. Generally, if each study performs score tests using genotypes in PLINK binary PED format or    GDS format, the score test output from glmm.score can be directly used as input files. Otherwise the meta-analysis function needs a tab or space delimited plain text file (or compressed files that can be recognized by the R function read.table) with at least 8 columns: SNP name, effect allele, reference allele, N, AF, SCORE, VAR and PVAL. Note that the SNP name, effect allele, reference allele can have customized column names in different input files, but the column names of N, AF, SCORE, VAR and PVAL should match exactly. Customized SNP and alleles column names can be specified using ”SNP”, ”A1” and ”A2”.   Note that we  do not define whether ”A1” or ”A2” is the effect allele:      it is your choice.  However,  your choice should be consistent across different studies:   for example, if you have two studies with the same allele column names ”Allele1” and ”Allele2”, and you want to define ”A1” as the effect allele, but the effect allele is ”Allele1” in the first study and ”Allele2” in the second study, you need A2 &#x3D; c(”Allele2”, ”Allele1”) for the reference allele column in each study, and A1 &#x3D; c(”Allele1”, ”Allele2”) for the effect allele column in each study. Note that in glmm.score output from analyzing PLINK binary PED format genotypes, the effect allele has column name ”A2”, and in glmm.score output from analyzing GDS format genotypes, the effect allele has column name ”ALT”. Thus if you have a result file from analyzing PLINK binary PED format genotypes in the third study, and another result file from analyzing GDS format genotypes in the fourth study, in addition to the aforementioned two studies, and you still want to define ”A1” as the effect allele, you need A2 &#x3D; c(”Allele2”, ”Allele1”, ”A1”, ”REF”) for the reference allele column in each study, and A1 &#x3D; c(”Allele1”, ”Allele2”, ”A2”, ”ALT”) for the effect allele column in each study.<br>Here is an example of meta-analyzing 3 score test result files:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">meta1.file  <-  system.file("extdata",  "meta1.txt",  package  =  "GMMAT")
meta2.file  <-  system.file("extdata",  "meta2.txt",  package  =  "GMMAT")
meta3.file  <-  system.file("extdata",  "meta3.txt",  package  =  "GMMAT")
glmm.score.meta(files  =  c(meta1.file,  meta2.file,  meta3.file), outfile  =  "glmm.score.meta.testoutfile.txt", SNP  =  rep("SNP",  3),  A1  =  rep("A1",  3),  A2  =  rep("A2",  3))

#### 5.3	Variant set tests
##### 5.3.1	Pooled analysis
Variant set tests (also known as gene-based tests or aggregate variant tests) in a single study (or a pooled analysis of multiple studies) can be performed using the function SMMAT. Currently only the GDS genotype format is supported. In addition to a glmmkin class object returned from the function glmmkin for the null GLMM and the GDS format genotype file, a group definition file with no header and 6 columns (variant set id, variant chromosome, variant position, variant reference allele, variant alternate
 
Note that each variant in the group definition file is matched by chromosome, position, reference allele and alternate allele with variants  from the GDS file.  One genetic variant   can be included in different groups with possibly different weights.  If no external weights  are needed in the analysis, simply replace the 6th column by all 1’s.
Four variant set tests are supported in the SMMAT framework: ”B” for the burden  test, ”S” for SKAT, ”O” for SKAT-O and ”E” for the efficient hybrid test of the burden test and SKAT. You can include one or more tests in a single analysis. If ”O” is selected, the burden test and SKAT results will be automatically included; if ”E” is selected, the burden test results will be automatically included. Therefore, the following example gives all four test results:
```&#123;r&#125;
group.file  <-  system.file("extdata",  "SetID.withweights.txt", package  =  "GMMAT")
geno.file  <-  system.file("extdata",  "geno.gds",  package  =  "GMMAT")
SMMAT(model0,  group.file  =  group.file,  geno.file  =  geno.file,	MAF.range  =  c(1e-7,  0.5),  miss.cutoff  =  1,  method  =  "davies",	tests  =  c("O",  "E"))
<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>
<p>It returns a data frame with first 8 columns showing the group (variant set) name, num- ber of variants in each group, minimum, mean, maximum missing rate of variants  in  each group, minimum, mean, maximum effect allele frequency of variants in each group, followed by variant set test results. For Burden, 3 columns will be included to show the burden test score, variance of the score, and its P value. For SKAT, the P value column will be included. For SKAT-O, 3 columns will be included to show SKAT-O P value, minimum P value in the search grid, and the value of the mixing parameter ρ at which the minimum P value is observed. For the efficient hybrid test, the P  value column will be included.</p>
<p>For a single study, intermediate files containing single variant scores and their covari- ance matrices for each variant set (based on the group definition file) can be saved for future use in re-analysis and&#x2F;or meta-analysis. For example, here we perform the burden test and save intermediate files:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">SMMAT(model0,  group.file  =  group.file,  geno.file  =  geno.file,	MAF.range  =  c(1e-7,  0.5),  miss.cutoff  =  1,  method  =  "davies",	tests  =  "B",  meta.file.prefix  =  "SMMAT.meta")
<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>
<p>In the example above, a space-delimited file ”SMMAT.meta.score.1” will be generated to save the single variant scores, and a binary file ”SMMAT.meta.var.1” will be generated to</p>
<p>save the covariance matrices for the variant sets. Note that the binary file is not human- readable, but can be used by SMMAT.meta in re-analysis and&#x2F;or meta-analysis.</p>
<h5 id="5-3-2Meta-analysis"><a href="#5-3-2Meta-analysis" class="headerlink" title="5.3.2	Meta-analysis"></a>5.3.2	Meta-analysis</h5><p>With intermediate files generated by SMMAT, the function SMMAT.meta can be used in re-analysis of single study results, and&#x2F;or meta-analysis to combine multiple studies. Here we show an example of rerunning SKAT using intermediate files generated above in the burden test:</p>
<pre class="line-numbers language-&#123;r&#125;"><code class="language-&#123;r&#125;">SMMAT.meta(meta.files.prefix  =  "SMMAT.meta",  n.files  =  1,	group.file  =  group.file,  MAF.range  =  c(1e-7,  0.5),	miss.cutoff  =  1,  method  =  "davies",  tests  =  "S")
<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>
<p>The first argument, ”meta.files.prefix”, is a vector of intermediate files’ prefix with length equal to the number of studies, and the second argument, ”n.files”, is a vector of integers showing how many sets of intermediate files each study has (also with length equal to the number of studies). In our above example of re-analysis, we have one set of intermedi- ate files (”SMMAT.meta.score.1” and ”SMMAT.meta.var.1”) with prefix ”SMMAT.meta”. The group definition file (passed to ”group.file”) should be the same as the one used to generate intermediate files by SMMAT (with possibly different weights allowed). In the above example, only SKAT is performed, but four variant set tests are supported in the SMMAT framework: ”B” for the burden test, ”S” for SKAT, ”O” for SKAT-O and ”E” for the efficient hybrid test of the burden test and SKAT. You can include one or more tests by passing a vector to the argument ”tests”. If ”O” is selected, the burden test and SKAT results will be automatically included; if ”E” is selected, the burden test results will be automatically included.</p>
<h2 id="6Output"><a href="#6Output" class="headerlink" title="6	Output"></a>6	Output</h2><p>The single variant score test function glmm.score generates a tab-delimited plain text output file. Here we show the header and the first five rows of the example output ”glmm.score.text.testoutfile.txt” from using a plain text genotype file ”geno.txt” in the function glmm.score:</p>
<pre><code>SNP	Allele1	Allele2	N	AF	SCORE	VAR	PVAL
SNP1	A	T	393	0.0254453	1.985	4.55635	0.352406
SNP2	A	C	400	0.5	3.51032	46.3328	0.60606 
SNP3	A	C	400	0.2075	-0.5334	30.6023	0.923185
SNP4	A	G	400	0.29875	-3.11494	40.5128	0.624567
SNP5	A	G	400	0.59375	-4.00135	42.2757	0.538287
...							
</code></pre>
<p>The first 3 columns are copied from the genotype file using ”infile.ncol.print” with names specified using ”infile.header.print”. Results are included in 5 columns: the sample size N, the allele frequency (AF) of the effect allele (Allele2 in this example, but it is the user’s choice: you can also define Allele1 as the effect allele in your coded genotype file), the score statistic SCORE of the effect allele, the variance of the score VAR, and score test  P value PVAL.</p>
<p>If you use a PLINK binary PED file ”geno.bed” as the genotype file, here are the header and the first 5 rows of the example output ”glmm.score.bed.testoutfile.txt” from glmm.score:</p>
<pre><code>CHR	SNP	cM	POS	A1	A2	N	AF
1	SNP1	0	1	T	A	393	0.974555
1	SNP2	0	2	A	C	400	0.5
1	SNP3	0	3	C	A	400	0.7925
1	SNP4	0	4	G	A	400	0.70125
1	SNP5	0	5	A	G	400	0.59375
...							
SCORE	VAR	PVAL
-1.985	4.55635	0.352406
3.51032	46.3328	0.60606
0.5334	30.6023	0.923185
3.11494	40.5128	0.624567
-4.00135	42.2757	0.538287
...	
</code></pre>
<p>The first 6 columns are copied from the .bim file (in our example, ”geno.bim”): the chromosome CHR, SNP name, genetic location cM, physical position POS, and alleles A1 and A2. Results are included in 5 columns: the sample size N, the allele frequency (AF) of A2 allele, the score statistic SCORE of A2 allele, the variance of the score VAR, and score test P value PVAL.</p>
<p>If you use a GDS genotype file ”geno.gds”, here are the header and the first 5 rows of the example output ”glmm.score.gds.testoutfile.txt” from glmm.score:</p>
<pre><code>SNP	CHR	POS	REF	ALT	N	MISSRATE	AF
SNP1	1	1	T	A	393	0.0175	0.974554707379135
SNP2	1	2	A	C	400	0	0.5
SNP3	1	3	C	A	400	0	0.7925
SNP4	1	4	G	A	400	0	0.70125
SNP5	1	5	A	G	400	0	0.59375
...							
SCORE	VAR	PVAL
-1.98499773964117	4.55635419833128	0.352406198841659
3.51031642022882	46.3327704279554	0.606059807621643
0.533400376138446	30.6022846771608	0.923185374786553
3.11494101139992	40.5127610067912	0.624566559783622
 
-4.00135050079485	42.2757210650549	0.538287231263494
...
</code></pre>
<p>The first 5 columns are extracted from the GDS file: SNP (”annotation&#x2F;id”), CHR (”chro- mosome”), POS (”position”), reference and alternate alleles (”allele”). Results are included in 6 columns: the sample size N (with non-missing genotypes), the genotype missing rate MISSRATE, the allele frequency (AF) of ALT allele, the score statistic SCORE of ALT allele, the variance of the score VAR, and score test P value PVAL.</p>
<p>The meta-analysis function glmm.score.meta generates a tab-delimited plain text output file. Here are the header and the first 5 rows of the example output from the meta-analysis ”glmm.score.meta.testoutfile.txt”:</p>
<pre><code>SNP	A1	A2	N	AF	SCORE	VAR	PVAL
L14	A	C	10000	0.65895	21.5371	445.72	0.30766609304268
L25	A	C	10000	0.78425	14.3376	387.091	0.466163476535903
L7	A	C	20000	0.50435	33.1136	1019.122	0.299608382095623
L9	A	C	30000	0.39875	-33.0641	904.842	0.271687891048334
L35	A	C	10000	0.78425	14.3376	387.091	0.466163476535903
...	
</code></pre>
<p>The first 3 columns are set by the function glmm.score.meta to denote SNP name and alleles (your choice of either A1 or A2 as the effect allele). N is the total sample size, AF is the effect allele frequency, SCORE is the summary score statistic of the effect allele, VAR is the variance of the summary score statistic, and PVAL is the meta-analysis P value.<br>In variant set tests SMMAT, if ”meta.file.prefix” is specified, space-delimited inter- mediate files for single variant scores and binary intermediate files for covariance matrices will be generated. Here are the header and the first 5 rows of the example intermediate file ”SMMAT.meta.score.1”:</p>
<pre><code>group  chr  pos  ref  alt  N  missrate  altfreq Set1 1 1 T A 393 0.0175 0.974554707379135
Set1	1	2	A	C	400	0	0.5
Set1	1	3	C	A	400	0	0.7925
Set1	1	4	G	A	400	0	0.70125
Set1	1	5	A	G	400	0	0.59375
...		
                    
SCORE VAR PVAL
-1.98499773963038 4.55635419833203 0.352406198844316
3.51031642023436 46.3327704279556 0.606059807621076
0.533400376147224 30.6022846771614 0.923185374785294
3.11494101140768 40.5127610067916 0.62456655978276
-4.00135050078827 42.2757210650552 0.538287231264163
...
</code></pre>
<p>The first 5 columns are copied from the group definition file, indicating the variant set (group) id, variant chromosome, variant position, variant reference allele, variant alternate allele, respectively. Results are included in 6 columns: the sample size N (with non- missing genotypes), the genotype missing rate missrate, the alt allele frequency altfreq, the score statistic SCORE of alt allele, the variance of the score VAR, and single variant score test P value PVAL.</p>
<h2 id="7Advanced-options"><a href="#7Advanced-options" class="headerlink" title="7	Advanced options"></a>7	Advanced options</h2><h4 id="7-1Alternative-model-fitting-algorithms"><a href="#7-1Alternative-model-fitting-algorithms" class="headerlink" title="7.1	Alternative model fitting algorithms"></a>7.1	Alternative model fitting algorithms</h4><p>By default we use the Average Information REML algorithm13,14 to fit the GLMM in glmmkin, which is computationally efficient and recommended in most cases. However, there are also alternative model fitting algorithms:</p>
<pre><code>method  =  &quot;REML&quot;,  method.optim  =  &quot;Brent&quot;
</code></pre>
<p>It maximizes the restricted likelihood using the derivative-free Brent method,15 but only works when there is one matrix for the covariance structure of the random effects.</p>
<pre><code>method  =  &quot;ML&quot;,  method.optim  =  &quot;Brent&quot;
</code></pre>
<p>It maximizes the likelihood using the Brent method.</p>
<pre><code>method  =  &quot;REML&quot;,  method.optim  =  &quot;Nelder-Mead&quot;
</code></pre>
<p>It maximizes the restricted likelihood using the Nelder-Mead method,16 however it is usually very slow in large samples.</p>
<pre><code>method  =  &quot;ML&quot;,  method.optim  =  &quot;Nelder-Mead&quot;
</code></pre>
<p>It maximizes the likelihood using the Nelder-Mead method.<br>Note that the default algorithm is</p>
<pre><code>method  =  &quot;REML&quot;,  method.optim  =  &quot;AI&quot;
</code></pre>
<p>A maximum likelihood version of Average Information algorithm is not available in<br>glmmkin.</p>
<h4 id="7-2Changing-model-fitting-parameters"><a href="#7-2Changing-model-fitting-parameters" class="headerlink" title="7.2	Changing model fitting parameters"></a>7.2	Changing model fitting parameters</h4><p>By default we set the maximum number of iteration to 500 and tolerance to declare convergence to 1e-5:</p>
<pre><code>maxiter  =  500,  tol  =  1e-5
</code></pre>
<p>These parameters can be changed. When using the Brent method for maximizing the likelihood (or restricted likelihood), we specify the search range of the ratio of the variance component parameter τ1 over the dispersion parameter  φ  to  be  between  1e-5  and  1e5, and we divide the search region evenly into 10 regions on the log scale:</p>
<pre><code>taumin  =  1e-5,  taumax  =  1e5,  tauregion  =  10
</code></pre>
<p>These parameters can also be changed, but they are only effective when using the Brent method.</p>
<h4 id="7-3Missing-genotypes"><a href="#7-3Missing-genotypes" class="headerlink" title="7.3	Missing genotypes"></a>7.3	Missing genotypes</h4><p>It is recommended to perform genotype quality control prior to analysis to impute missing genotypes or filter out SNPs with high missing rates. However, GMMAT does allow miss- ing genotypes, and imputes to the mean value by default. Alternatively, in glmm.score and glmm.wald, missing genotypes can be omitted from the analysis using</p>
<pre><code>missing.method  =  &quot;omit&quot;
</code></pre>
<p>In variant set tests using SMMAT, instead of imputing missing genotypes to the mean value, you can impute missing genotypes to 0 (homozygous reference allele) using</p>
<pre><code>missing.method  =  &quot;impute2zero&quot;
</code></pre>
<p>If using a plain text (or compressed .gz and .bz2) genotype file, missing genotypes should be coded as ”NA”. If you have missing genotypes coded in a different way, you can specify this in the argument ”infile.na”.</p>
<h4 id="7-4Reordered-genotypes"><a href="#7-4Reordered-genotypes" class="headerlink" title="7.4	Reordered genotypes"></a>7.4	Reordered genotypes</h4><p>The genotype file (either a plain text file, a PLINK binary PED file, or a GDS file) can include more individuals than in the phenotype and covariates data frame, and they can be in different orders. GMMAT handles this issue using an argument ”select” in both glmm.score and glmm.wald. For example, if the order of individuals in your genotype file is A, B, C, D, but you only have 3 unique individuals (with order C, A, B) in the fitted ”obj” (for glmm.score) or in the data frame ”data” (for glmm.wald), then you can specify</p>
<pre><code>select = c(2, 3, 1, 0)
</code></pre>
<p>to reflect the order of individuals. Note that since individual D is not included, its order is assigned to 0. The length of the vector must match the number of individuals in your genotype file. Also note that if there are observations with missing phenotype&#x2F;covariates in ”data”, ”select” for glmm.wald should match to ”data” before removing any missing values, while ”select” for glmm.score should match to ”obj” (in which missing values have been excluded).</p>
<p>In variant set tests, SMMAT will extract ID from ”null.obj” using ”id include” re- turned in the glmmkin fitted null model object. The ID will be matched to ”sample.id” in the GDS genotype file.</p>
<h4 id="7-5Parallel-computing"><a href="#7-5Parallel-computing" class="headerlink" title="7.5	Parallel computing"></a>7.5	Parallel computing</h4><p>Parallel computing can be enabled in glmm.score and SMMAT using the argument ”ncores” to specify how many cores you would like to use on a computing node. By default ”ncores” is 1, meaning that these functions will run in a single thread. Currently parallel computing is only implemented for GDS format genotype files.</p>
<p>If you  enable parallel computing and save  intermediate files,  you  will get multi-   ple sets of intermediate files. For example, if your ”ncores” is 12 and you specified ”meta.file.prefix” to ”study1”, then you will get 12 sets of (totaling 24) intermediate files ”study1.score.1”, ”study1.var.1”, ”study1.score.2”, ”study1.var.2”, …, ”study1.score.12”, ”study1.var.12”. Later in the meta-analysis to combine with 2 sets of intermediate files ”study2.score.1”, ”study2.var.1”, ”study2.score.2”, ”study2.var.2”, you will need to use</p>
<pre><code>meta.files.prefix  =  c(&quot;study1&quot;,  &quot;study2&quot;),  n.files  =  c(12,  2)
</code></pre>
<p>If your R is configured with Intel MKL and you would like to enable parallel comput- ing, it is recommended that you set the environmental variable ”MKL NUM THREADS” to 1 before running R to avoid hanging. Alternatively,  you  can do this at the beginning  of your R script by using</p>
<pre><code>Sys.setenv(MKL_NUM_THREADS  =  1)
</code></pre>
<h4 id="7-6Variant-filters"><a href="#7-6Variant-filters" class="headerlink" title="7.6	Variant filters"></a>7.6	Variant filters</h4><p>Variants can be filtered in glmm.score and SMMAT based on minor allele frequency (MAF) and missing rate filters. The argument ”MAF.range” specifies the minimum and maximum MAFs for a variant to be included in the analysis. By default the minimum MAF is 1 10−7 and the maximum MAF is 0.5, meaning that only monomorphic markers in the sample will be excluded (if your sample size is no more than 5 million). The argument ”miss.cutoff” specifies the maximum missing rate for a variant to be included  in the analysis.  By default it is set to 1, meaning that no variants  will be removed due to high genotype missing rates.</p>
<h4 id="7-7Internal-minor-allele-frequency-weights"><a href="#7-7Internal-minor-allele-frequency-weights" class="headerlink" title="7.7	Internal minor allele frequency weights"></a>7.7	Internal minor allele frequency weights</h4><p>Internal weights are calculated based on the minor allele frequency (NOT the effect allele frequency, therefore, variants with effect allele frequencies 0.01 and 0.99 have the same weights) as a beta probability density function. Internal weights are multiplied by the external weights given in the last column of the group definition file. To turn off internal weights, use</p>
<pre><code>MAF.weights.beta  =  c(1,  1)
</code></pre>
<p>to assign flat weights, as a beta distribution with parameters 1 and 1 is a uniform distri- bution on the interval between 0 and 1.</p>
<h4 id="7-8Allele-flipping"><a href="#7-8Allele-flipping" class="headerlink" title="7.8	Allele flipping"></a>7.8	Allele flipping</h4><p>In variant set tests SMMAT, by default the alt allele is used as the coding allele and variants in each variant set are matched strictly on chromosome, position, reference and alternate alleles.</p>
<p>The argument ”auto.flip” allows automatic allele flipping if a specified variant is not found in the genotype file, but a variant at the same chromosome and position with reference allele matching the alternate allele in the group definition file ”group.file”, and alternate allele matching the reference allele in the group definition file ”group.file”, to  be included in the analysis. Please use with caution for whole genome sequence data, as both ref&#x2F;alt and alt&#x2F;ref variants at the same position are not uncommon, and they are  likely two different variants, rather than allele flipping.</p>
<p>The argument ”use.minor.allele” allows using the minor allele instead of the alt allele as the coding allele in variant set tests. Note that this choice does not change ”S” for SKAT results, but ”B” for the burden test, ”O” for SKAT-O and ”E” for efficient hybrid test of the burden test and SKAT results will be affected. Generally the alt allele can either be the minor or the major allele. If in a variant set, different variants with alt allele</p>
<p>frequencies 0.001 and 0.998 are combined together in a burden test, the results would be difficult to interpret. We generally recommend turning on the ”use.minor.allele” option, unless you know the ancestry alleles explicitly and the specific scientific hypothesis clearly that you would like to test. Along with the MAF filter, this option is useful for combining rare mutations, assuming rare allele effects are in the same direction.</p>
<h4 id="7-9P-values-of-weighted-sum-of-chi-squares"><a href="#7-9P-values-of-weighted-sum-of-chi-squares" class="headerlink" title="7.9	P values of weighted sum of chi-squares"></a>7.9	P values of weighted sum of chi-squares</h4><p>In variant set tests SMMAT, you can use 3 methods in the ”method” argument to compute P values of weighted sum of chi-square distributions: ”davies”,17 ”kuonen”18 and ”liu”.19  By default ”davies” is used, if it returns an error message in the calculation, or a  P value greater than 1, or less than 1 10−5, ”kuonen” method will be used. If ”kuonen” method fails to compute the P value, ”liu” method will be used.</p>
<h4 id="7-10Heterogeneous-genetic-effects-in-variant-set-meta-analysis"><a href="#7-10Heterogeneous-genetic-effects-in-variant-set-meta-analysis" class="headerlink" title="7.10	Heterogeneous genetic effects in variant set meta-analysis"></a>7.10	Heterogeneous genetic effects in variant set meta-analysis</h4><p>Heterogeneous genetic effects20 are allowed in variant set tests meta-analysis function SMMAT.meta, by specifying groups using the ”cohort.group.idx” argument. By default all studies are assumed to share the same genetic effects in the meta-analysis, and this can be changed by assigning different group indices to studies. For example,</p>
<pre><code>cohort.group.idx = c(&quot;a&quot;,&quot;b&quot;,&quot;a&quot;,&quot;a&quot;,&quot;b&quot;)
</code></pre>
<p>means cohorts 1, 3, 4 are assumed to have homogeneous genetic effects, and cohorts 2, 5 are in another group with homogeneous genetic effects (but possibly heterogeneous with group ”a”).</p>
<h4 id="7-11Other-options"><a href="#7-11Other-options" class="headerlink" title="7.11	Other options"></a>7.11	Other options</h4><p>By default, genotypes are centered to the mean before the analysis in single variant tests.<br>You can turn this feature off by specifying</p>
<pre><code>center = FALSE
</code></pre>
<p>in both glmm.score and glmm.wald functions to use raw genotypes.</p>
<p>If your genotype file is a plain text (or a compressed .gz and .bz2 file), and you want to read in fewer lines than all lines included in the file, you can use the ”infile.nrow” argument to specify how many lines (including lines to be skipped using ”infile.nrow.skip”) you want to read in. By default the delimiter is assumed to be a tab, but you can change it using the ”infile.sep” argument. These options are implemented in glmm.score and glmm.wald. In the score test function glmm.score, by default 100 SNPs are tested in a batch. You can change it using the ”nperbatch” argument, but the computational time can increase substantially if it is either too small or too large, depending on the performance of your<br>computing system.</p>
<p>If you perform Wald tests glmm.wald and use a plain text (or a compressed .gz  and .bz2) file, and your SNPs are not in your first column, you can change ”snp.col” in glmm.wald to indicate which column is your SNP name.</p>
<p>In the variant set tests SMMAT, by default the group definition file ”group.file” should be tab delimited, but you can change it using the ”group.file.sep” argument. Also there is a ”Garbage.Collection” argument (default FALSE), if turned on, SMMAT will call the function gc for each variant set tested. It helps save memory footprint, but the computation speed might be slower.</p>

            


        </div>
    </div>
    <div id="post-footer" class="post-footer main-content-wrap">
        
            <div class="post-footer-tags">
                <span class="text-color-light text-small">标签</span><br/>
                
    <a class="tag tag--primary tag--small t-none-link" href="/all-tags/GMMAT/" rel="tag">GMMAT</a> <a class="tag tag--primary tag--small t-none-link" href="/all-tags/GWAS/" rel="tag">GWAS</a> <a class="tag tag--primary tag--small t-none-link" href="/all-tags/SMMAT/" rel="tag">SMMAT</a>

            </div>
        
        
            <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    data-tooltip="GWAS|EPACTS安装及使用"
                    aria-label="上一篇: GWAS|EPACTS安装及使用"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-12-miRNA%E6%95%B0%E6%8D%AE%E5%BA%93-miRNA%E9%9D%B6%E5%9F%BA%E5%9B%A0%E9%A2%84%E6%B5%8B/"
                    data-tooltip="miRNA数据库|miRNA靶基因预测"
                    aria-label="下一篇: miRNA数据库|miRNA靶基因预测"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


        
        
    </div>
</article>



                <footer id="footer" class="main-content-wrap">
    <span class="copyrights">
        Copyrights &copy; 2022 Dr.iuu. All Rights Reserved.
    </span>
</footer>

            </div>
            
                <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
                    <div class="post-actions-wrap">
    <nav>
        <ul class="post-actions post-action-nav">
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-15-GWAS-EPACTS%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    data-tooltip="GWAS|EPACTS安装及使用"
                    aria-label="上一篇: GWAS|EPACTS安装及使用"
                >
                    
                        <i class="fa fa-angle-left" aria-hidden="true"></i>
                        <span class="hide-xs hide-sm text-small icon-ml">上一篇</span>
                    </a>
            </li>
            <li class="post-action">
                
                    
                <a
                    class="post-action-btn btn btn--default tooltip--top"
                    href="/2020-06-12-miRNA%E6%95%B0%E6%8D%AE%E5%BA%93-miRNA%E9%9D%B6%E5%9F%BA%E5%9B%A0%E9%A2%84%E6%B5%8B/"
                    data-tooltip="miRNA数据库|miRNA靶基因预测"
                    aria-label="下一篇: miRNA数据库|miRNA靶基因预测"
                >
                    
                        <span class="hide-xs hide-sm text-small icon-mr">下一篇</span>
                        <i class="fa fa-angle-right" aria-hidden="true"></i>
                    </a>
            </li>
        </ul>
    </nav>
    <ul class="post-actions post-action-share">
        <li class="post-action hide-lg hide-md hide-sm">
            <a
                class="post-action-btn btn btn--default btn-open-shareoptions"
                href="#btn-open-shareoptions"
                aria-label="Diesen Beitrag teilen"
            >
                <i class="fa fa-share-alt" aria-hidden="true"></i>
            </a>
        </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Facebook"
                    aria-label="分享到 Facebook"
                >
                    <i class="fab fa-facebook" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Twitter"
                    aria-label="分享到 Twitter"
                >
                    <i class="fab fa-twitter" aria-hidden="true"></i>
                </a>
            </li>
        
            
            
            <li class="post-action hide-xs">
                <a
                    class="post-action-btn btn btn--default"
                    target="new" href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                    title="分享到 Google+"
                    aria-label="分享到 Google+"
                >
                    <i class="fab fa-google-plus" aria-hidden="true"></i>
                </a>
            </li>
        
        
        <li class="post-action">
            
                <a class="post-action-btn btn btn--default" href="#table-of-contents" aria-label="目录">
            
                <i class="fa fa-list" aria-hidden="true"></i>
            </a>
        </li>
    </ul>
</div>


                </div>
                
    <div id="share-options-bar" class="share-options-bar" data-behavior="4">
        <i id="btn-close-shareoptions" class="fa fa-times"></i>
        <ul class="share-options">
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://www.facebook.com/sharer/sharer.php?u=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                        aria-label="分享到 Facebook"
                    >
                        <i class="fab fa-facebook" aria-hidden="true"></i><span>分享到 Facebook</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://twitter.com/intent/tweet?text=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                        aria-label="分享到 Twitter"
                    >
                        <i class="fab fa-twitter" aria-hidden="true"></i><span>分享到 Twitter</span>
                    </a>
                </li>
            
                
                
                <li class="share-option">
                    <a
                        class="share-option-btn"
                        target="new"
                        href="https://plus.google.com/share?url=https://iuu19.github.io/2020-06-13-GWAS-GMMAT%E5%AE%89%E8%A3%85%E5%8F%8A%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B/"
                        aria-label="分享到 Google+"
                    >
                        <i class="fab fa-google-plus" aria-hidden="true"></i><span>分享到 Google+</span>
                    </a>
                </li>
            
        </ul>
    </div>


            
        </div>
        


    
        
    

<div id="about">
    <div id="about-card">
        <div id="about-btn-close">
            <i class="fa fa-times"></i>
        </div>
        
            <img id="about-card-picture" src="/assets/images/coveriuu.jpg" alt="作者的图片"/>
        
            <h4 id="about-card-name">Dr.iuu</h4>
        
            <div id="about-card-bio"><p>NCMC-SCMC-SWCH</p>
</div>
        
        
            <div id="about-card-job">
                <i class="fa fa-briefcase"></i>
                <br/>
                <p>Bioinfo-CHD</p>

            </div>
        
        
            <div id="about-card-location">
                <i class="fa fa-map-marker-alt"></i>
                <br/>
                Pudong, Shanghai
            </div>
        
    </div>
</div>

        
        
<div id="cover" style="background-image:url('/assets/images/cover2.jpg');"></div>
        <!--SCRIPTS-->

<script src="/assets/js/script-3nlvflbbvekvxx0g6h2clfqhhxs61cvvbofv7nmctigkdbbhu5htlvmrqg80.min.js"></script>

<!--SCRIPTS END-->





    </body>
</html>
